<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<div class="figure">
<img src="../figures/bandnorm_intro.png" alt=""><p class="caption">The introduction for BandNorm</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are two possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1  38.000000</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 105.000000</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 116.000000</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 133.000000</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 112.000000</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony converged after 7 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                  PC1       PC2        PC3         PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1    1.0293516 49.473692 -76.618811 -36.8752564 -42.022535 -23.354049</span>
<span class="co">#&gt; Astro_2   -9.1977209 32.347768  -2.756897   8.3822718  -8.160682   4.438291</span>
<span class="co">#&gt; Astro_3   -0.5292853 14.812305  -2.973394   1.6324264  11.047545 -10.023949</span>
<span class="co">#&gt; Astro_4   -6.0115586 37.463300 -29.961596  11.2777084 -13.797429  -6.104187</span>
<span class="co">#&gt; Astro_5   12.0999009 18.577941 -23.574044  13.2213510 -26.996703 -10.876395</span>
<span class="co">#&gt; Astro_6  -20.6892354 13.743482 -24.919865  -0.7956708  -3.135400   7.802125</span>
<span class="co">#&gt; Astro_7  -49.4006359 29.103575 -20.027626   6.8908420   1.295028  -1.965032</span>
<span class="co">#&gt; Astro_8  -11.7727162 34.881938  -6.166616   8.6403040  -6.384149   4.617832</span>
<span class="co">#&gt; Astro_9  -11.5708831  8.350641 -22.892308  18.7070499 -15.263220  -5.301028</span>
<span class="co">#&gt; Astro_10 -16.0734039  4.329660 -18.211048   3.1339731 -23.705511  -9.755443</span>
<span class="co">#&gt;                 PC7        PC8       PC9        PC10       PC11       PC12</span>
<span class="co">#&gt; Astro_1    5.463078 -39.908755 37.371895  -7.5543198  38.568930 -33.102037</span>
<span class="co">#&gt; Astro_2   -3.914980  -3.740787  9.324096  14.9302331  22.664567 -24.724161</span>
<span class="co">#&gt; Astro_3  -27.728462  22.269925  5.494555  15.0396810  -3.799887  -8.524067</span>
<span class="co">#&gt; Astro_4   -3.742008   7.242514  5.243161  -1.9170383 -17.179946 -18.222599</span>
<span class="co">#&gt; Astro_5   -4.774827   7.453127 46.319538  29.1508380  81.142666 -46.266262</span>
<span class="co">#&gt; Astro_6  -12.451910  -3.223360 -2.009652   1.1047054   6.289144 -43.766538</span>
<span class="co">#&gt; Astro_7   -7.223646  -1.338891 -7.333724  -6.9135170  -6.937292  -5.093652</span>
<span class="co">#&gt; Astro_8  -20.610865   6.190320 -1.715928   0.5955311  17.998392   9.602011</span>
<span class="co">#&gt; Astro_9   18.812482  27.737164 48.252583  11.8094384  41.092434 -67.756942</span>
<span class="co">#&gt; Astro_10  11.236069  -2.918244 28.993287 -16.9185304  11.773731 -29.898823</span>
<span class="co">#&gt;                PC13       PC14       PC15        PC16       PC17        PC18</span>
<span class="co">#&gt; Astro_1   36.450373 -16.727650  26.892309  -8.1290435  51.809195 -44.4848454</span>
<span class="co">#&gt; Astro_2   -1.020508   7.944943  -3.754481  10.9843335 -19.575093 -14.7476760</span>
<span class="co">#&gt; Astro_3   28.377543 -24.561802 -11.678825 -13.8629673  32.477351  -7.4452104</span>
<span class="co">#&gt; Astro_4    7.638106 -28.233967 -35.049459 -36.6979262  23.466988  -0.3285857</span>
<span class="co">#&gt; Astro_5   38.215933  16.699635 -10.407043  -4.8891076  13.237492  11.9011582</span>
<span class="co">#&gt; Astro_6   27.994952   1.503671   9.652660 -16.8868315   1.422629  -9.8344336</span>
<span class="co">#&gt; Astro_7   19.795840  -9.758172  -5.624113   4.4242897   7.287895 -28.1444404</span>
<span class="co">#&gt; Astro_8    3.389298 -26.686585  -8.361308 -22.2581458   9.467595   0.5166268</span>
<span class="co">#&gt; Astro_9  -19.296436  -9.323479  32.195526  45.5151545  -6.140340 -43.9776395</span>
<span class="co">#&gt; Astro_10  17.931475 -22.066403  -6.880684   0.3976595  20.064442 -22.3763206</span>
<span class="co">#&gt;                PC19        PC20         PC21       PC22        PC23        PC24</span>
<span class="co">#&gt; Astro_1   36.600347  26.8185878 -147.0620917  62.975980  13.1925808  -43.660426</span>
<span class="co">#&gt; Astro_2    8.171562  -1.4955112    9.0129350 -15.638932  -0.1001894    2.150610</span>
<span class="co">#&gt; Astro_3   20.396139   2.5873515  -14.4104323   5.577496 -15.7555393    4.359450</span>
<span class="co">#&gt; Astro_4  -27.350005  -0.3935630   -1.0886763 -13.563974 -35.2976197   -2.517836</span>
<span class="co">#&gt; Astro_5  100.353701   4.8132207   90.9908739   1.436840 137.9149138 -108.072497</span>
<span class="co">#&gt; Astro_6  -18.436873  -6.1207879    4.8147881  43.919609  -7.9468088   11.584170</span>
<span class="co">#&gt; Astro_7   -2.912850   4.4504741   -4.2811323  10.185427 -16.8892275   -2.059864</span>
<span class="co">#&gt; Astro_8   20.856135   0.3562213  -47.2415088 -10.193786 -28.5701192   53.105555</span>
<span class="co">#&gt; Astro_9  -61.168698 -34.7447627   15.1963450 -43.678426  68.1156788    6.403451</span>
<span class="co">#&gt; Astro_10  -2.811194 -10.6610313   -0.5190964 -25.226746  -4.7663421    3.187728</span>
<span class="co">#&gt;                PC25         PC26       PC27          PC28      PC29       PC30</span>
<span class="co">#&gt; Astro_1   16.878322  -73.4069402  11.565298 -132.12895549  15.99669 -39.139799</span>
<span class="co">#&gt; Astro_2   -1.671599    6.8910566 -11.619943    0.07659627 -10.85001  13.462678</span>
<span class="co">#&gt; Astro_3    5.002134   11.1562218 -18.838594    3.52801178  43.71871   3.217179</span>
<span class="co">#&gt; Astro_4   -2.993162    5.3338359  35.343259    7.07005100  16.53636 -13.925318</span>
<span class="co">#&gt; Astro_5   47.480065   41.6587233 -52.832806   69.27468765 -53.44459  11.938828</span>
<span class="co">#&gt; Astro_6    6.021035    0.7363682  24.592123  -10.46160398  24.57937  -1.887627</span>
<span class="co">#&gt; Astro_7   37.133145   -1.3407220  22.447725    7.72297864  32.12957  -3.287794</span>
<span class="co">#&gt; Astro_8    4.822512   39.3224695 -32.338640    8.05203637  12.73663 -24.432273</span>
<span class="co">#&gt; Astro_9    9.823496 -147.7889985  -7.790911   65.90312086 -20.40180   3.733545</span>
<span class="co">#&gt; Astro_10 -41.053756   -0.6003121  11.323381    2.61211428  13.78707  -4.367368</span>
<span class="co">#&gt;                PC31       PC32          PC33       PC34       PC35       PC36</span>
<span class="co">#&gt; Astro_1   15.170835 -61.384365   9.340159071 -23.285969  59.585590 -59.685187</span>
<span class="co">#&gt; Astro_2    2.619909 -13.081421   8.394563749   8.427431 -11.150123  -4.734391</span>
<span class="co">#&gt; Astro_3    9.782181 -17.532229 -71.892748279   5.310214  -5.183289  -6.431404</span>
<span class="co">#&gt; Astro_4  -26.545923  17.482512  -0.008463749  29.012026  14.868218  15.483310</span>
<span class="co">#&gt; Astro_5  -16.089578  -4.580247   6.651398091 -42.639237  60.222377  15.201723</span>
<span class="co">#&gt; Astro_6   -2.602373  39.545451  14.782254702  -4.059741  21.021221  10.925223</span>
<span class="co">#&gt; Astro_7  -21.034836  -1.929582 -31.663202878   5.625456 -37.651852  -8.797673</span>
<span class="co">#&gt; Astro_8    2.112254   8.790524  41.240476002 -15.072993  64.533421  42.162179</span>
<span class="co">#&gt; Astro_9   54.508165 -24.035543  -6.492675989 -13.395225 -61.438521 -13.621271</span>
<span class="co">#&gt; Astro_10 -38.982159  38.013844 -28.381169515  24.180290  12.499797  14.878211</span>
<span class="co">#&gt;                PC37       PC38       PC39       PC40       PC41        PC42</span>
<span class="co">#&gt; Astro_1   13.247817  -16.85256  43.202536  -7.055817  -1.683316  -0.2070641</span>
<span class="co">#&gt; Astro_2    5.001783   18.77908   5.697223   5.230481   3.327796  17.1015400</span>
<span class="co">#&gt; Astro_3  -10.349191   64.58069  -9.286422 -23.030223   6.024418   9.9562026</span>
<span class="co">#&gt; Astro_4  -20.831042  -23.26558  31.801104  10.089743 -53.422976 -11.1093791</span>
<span class="co">#&gt; Astro_5    0.224684  -26.96889  14.546407 -37.383181  71.694498  20.1771640</span>
<span class="co">#&gt; Astro_6  -17.038125   28.90068  40.873825  44.149871   2.019554  19.0590047</span>
<span class="co">#&gt; Astro_7  -22.703862   47.85184 -22.433550   1.141341  41.644556 -52.8785130</span>
<span class="co">#&gt; Astro_8   24.853904 -112.21119  17.176909 -16.520592  -1.972054 -26.5114899</span>
<span class="co">#&gt; Astro_9   -1.320368   80.83622 -67.838042  25.109245  -9.967477  23.8448185</span>
<span class="co">#&gt; Astro_10  11.156463  -33.49567 -53.305006 -60.669860 -21.026879  10.2927822</span>
<span class="co">#&gt;                 PC43        PC44       PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -6.5215128 -15.1906129  25.597990  -2.942280 -15.483955  18.799563</span>
<span class="co">#&gt; Astro_2   31.4258317  -2.1848247  -6.086869 -24.308698   8.063508  -4.890081</span>
<span class="co">#&gt; Astro_3    0.8390653  33.7975958  41.669168  10.413949  54.169226  75.182113</span>
<span class="co">#&gt; Astro_4   28.4241608  -2.4405875  18.469685  53.508480  37.046838  -9.282998</span>
<span class="co">#&gt; Astro_5  -16.0046799  -0.5111173  -6.981688 -13.377257 -49.037175 -14.344441</span>
<span class="co">#&gt; Astro_6  -44.5623456   9.1873870 -47.659106  15.241128 -28.906248  22.249599</span>
<span class="co">#&gt; Astro_7  -13.6391401  -6.1774895   7.447094 -54.666682 -11.993722   7.998112</span>
<span class="co">#&gt; Astro_8   46.9426615  27.3817462 -53.221615  23.026799  21.412688 -25.393771</span>
<span class="co">#&gt; Astro_9  -25.8759005 -13.3802244  36.020301  17.674731  20.597259 -40.797061</span>
<span class="co">#&gt; Astro_10  21.0630570  17.7644054  18.374701   2.760565 -35.820752 -10.865968</span>
<span class="co">#&gt;                 PC49       PC50</span>
<span class="co">#&gt; Astro_1   38.5229171 -19.136755</span>
<span class="co">#&gt; Astro_2   -4.1254963  28.762265</span>
<span class="co">#&gt; Astro_3   -4.3219212  -2.166093</span>
<span class="co">#&gt; Astro_4   35.3323296  30.507097</span>
<span class="co">#&gt; Astro_5   -5.5908174  -4.588561</span>
<span class="co">#&gt; Astro_6    0.1494455 -11.449717</span>
<span class="co">#&gt; Astro_7  -15.5481144  38.296103</span>
<span class="co">#&gt; Astro_8    5.4483902  31.735607</span>
<span class="co">#&gt; Astro_9   35.4435382 -12.440492</span>
<span class="co">#&gt; Astro_10 -49.6520784 -82.831661</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
