<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD_tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; Warning: package 'rmarkdown' was built under R version 4.0.5</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony 10/10</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                  PC1       PC2        PC3       PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1  -19.6818675 33.187558 -24.291312 34.305986 -37.328502 -17.326726</span>
<span class="co">#&gt; Astro_2   -6.9413515 35.764457  -0.881772 13.842615  -8.433046  11.788085</span>
<span class="co">#&gt; Astro_3   -0.1522657 11.731713  -1.567521  2.776508  11.146740 -12.273138</span>
<span class="co">#&gt; Astro_4  -11.4269339 24.421817 -24.699096 12.630375 -11.461560 -19.269284</span>
<span class="co">#&gt; Astro_5   10.9686354 14.232116 -23.579580 10.386387 -23.045573 -14.536422</span>
<span class="co">#&gt; Astro_6  -23.1276963 13.196159 -19.509804 -0.738982  -3.575348  11.037074</span>
<span class="co">#&gt; Astro_7  -24.3950214 21.607768 -13.830574  8.450608  -2.419186  -2.942963</span>
<span class="co">#&gt; Astro_8  -12.8734154 36.446388  -5.606941  7.560606  -1.447784   4.944132</span>
<span class="co">#&gt; Astro_9  -12.6697465  9.948373 -22.331075 17.613220 -10.338764  -4.950630</span>
<span class="co">#&gt; Astro_10 -18.4595052  3.857227 -12.850553  3.136981 -24.144070  -6.476602</span>
<span class="co">#&gt;                 PC7        PC8        PC9       PC10       PC11       PC12</span>
<span class="co">#&gt; Astro_1   -3.174965 -45.142899 38.1858949  -1.310533 17.4545481 -23.973717</span>
<span class="co">#&gt; Astro_2   -3.114486  -1.966325 12.0148420  16.080430 19.4947920 -30.928776</span>
<span class="co">#&gt; Astro_3  -20.962141  26.375546  1.0220099  10.104739 -4.4285838  -6.151360</span>
<span class="co">#&gt; Astro_4    9.714641   3.194137  4.7535728 -10.556344  2.6782999 -15.394851</span>
<span class="co">#&gt; Astro_5    2.106940  12.486421 41.3317700  22.283793 81.0174869 -42.992193</span>
<span class="co">#&gt; Astro_6  -11.478945  -3.656145  0.9432855   4.087796  8.9836023 -45.498311</span>
<span class="co">#&gt; Astro_7    1.283122   2.550774 -3.0860071  -9.269331  0.8813164 -16.760036</span>
<span class="co">#&gt; Astro_8  -21.742754   7.500742  0.1119756   1.091117 18.5508798   9.283238</span>
<span class="co">#&gt; Astro_9   17.701556  29.060677 50.0758019  12.301419 41.6392893 -68.062144</span>
<span class="co">#&gt; Astro_10  12.153053  -3.311661 31.9030157 -13.937689 14.3380999 -31.615839</span>
<span class="co">#&gt;                 PC13       PC14       PC15        PC16        PC17       PC18</span>
<span class="co">#&gt; Astro_1   41.9913391 -40.315977  12.005285 -34.2698820  62.3687102 -48.712968</span>
<span class="co">#&gt; Astro_2    0.4929860  -4.876995 -15.080028   1.4619842 -15.7442708 -14.193378</span>
<span class="co">#&gt; Astro_3   17.8029350 -13.042382 -13.040820  -2.6092264  20.2778343  -2.682610</span>
<span class="co">#&gt; Astro_4   -6.2407974   7.291412 -16.712736  -3.0916003  -0.1070502   1.580848</span>
<span class="co">#&gt; Astro_5   25.7541686  32.271291 -10.616557   9.3623690   0.6137995  17.118863</span>
<span class="co">#&gt; Astro_6   27.7983094  -0.392700   7.627107 -16.3108803   0.9139409 -12.429116</span>
<span class="co">#&gt; Astro_7   17.7175670  -8.857574 -13.545722  -0.1547511  10.1359683 -19.565539</span>
<span class="co">#&gt; Astro_8    0.5981962 -28.107482  -4.788695 -20.5537922  10.3528223  -1.354872</span>
<span class="co">#&gt; Astro_9  -22.1027050 -10.760005  35.777556  47.2319585  -5.2599078 -45.848556</span>
<span class="co">#&gt; Astro_10  17.7511224 -24.062487  -9.018490   0.8658103  19.6204282 -24.955344</span>
<span class="co">#&gt;                PC19        PC20        PC21       PC22        PC23        PC24</span>
<span class="co">#&gt; Astro_1   23.625974  17.6605511 -163.783008  59.797498   0.7301939  -34.822957</span>
<span class="co">#&gt; Astro_2   -4.814722  -5.9885370    5.767593 -16.329984 -10.5933142   18.147501</span>
<span class="co">#&gt; Astro_3   16.313200   4.1354337    2.209153  -2.182894 -20.1453510   10.251953</span>
<span class="co">#&gt; Astro_4   -2.911575  18.6703911    9.336044 -12.284077 -14.0280746  -16.445838</span>
<span class="co">#&gt; Astro_5   95.797062   6.4276278  113.142337  -7.337412 131.7953384 -102.850942</span>
<span class="co">#&gt; Astro_6  -18.198631  -4.9939682    2.629506  45.355615  -2.0375692   10.791805</span>
<span class="co">#&gt; Astro_7   -9.348555  -0.6934357   -8.678324  11.438277 -25.6017120    7.771993</span>
<span class="co">#&gt; Astro_8   20.975432  -2.3895087  -47.449594 -14.347009 -28.0862617   52.385958</span>
<span class="co">#&gt; Astro_9  -61.052777 -37.4811980   14.994089 -47.847431  68.6077214    5.665649</span>
<span class="co">#&gt; Astro_10  -2.741766  -9.6476862   -2.608466 -23.840050   0.9570935    2.503927</span>
<span class="co">#&gt;                PC25        PC26       PC27        PC28       PC29         PC30</span>
<span class="co">#&gt; Astro_1   11.215157  -77.584368  29.681514 -143.306304  16.857011 -38.00863995</span>
<span class="co">#&gt; Astro_2   -5.590577    3.084717  -3.372871    1.533399   1.974656  22.00529798</span>
<span class="co">#&gt; Astro_3    3.483674   19.009360 -20.093098   21.033444  43.453755   6.90860784</span>
<span class="co">#&gt; Astro_4    2.280373    5.383140  19.351155   18.369493   1.434371  -9.17083647</span>
<span class="co">#&gt; Astro_5   45.429895   53.100323 -52.534766   91.301842 -54.592389  15.92679637</span>
<span class="co">#&gt; Astro_6    3.635530   -3.698877  30.294998  -10.915871  21.423324   2.44838313</span>
<span class="co">#&gt; Astro_7   37.695662   -4.995751  17.011352   12.665976  29.152033 -14.76670572</span>
<span class="co">#&gt; Astro_8    5.563171   36.607953 -34.494820   12.071759  10.908332 -28.21795362</span>
<span class="co">#&gt; Astro_9   10.563340 -150.510090  -9.915328   69.965334 -22.254300  -0.08238563</span>
<span class="co">#&gt; Astro_10 -43.488906   -4.933268  17.113636    2.237104  10.700541  -0.06850146</span>
<span class="co">#&gt;                 PC31       PC32        PC33       PC34       PC35       PC36</span>
<span class="co">#&gt; Astro_1   11.8738139 -63.128476  16.6029856 -11.865479  57.000630 -71.917909</span>
<span class="co">#&gt; Astro_2  -17.7354614   8.042276  10.7771190   9.709260  -9.305294   2.812745</span>
<span class="co">#&gt; Astro_3    3.4370045  -6.392208 -77.1591353   8.385250  -9.672752   2.330486</span>
<span class="co">#&gt; Astro_4  -14.8573259   3.390100  -9.2146524   5.995918  14.522816   7.328910</span>
<span class="co">#&gt; Astro_5  -21.2889828   8.966508   0.4062215 -39.595214  52.704800  25.979615</span>
<span class="co">#&gt; Astro_6   -0.3523226  40.594552  14.9858348  -8.923712  13.643289   6.666905</span>
<span class="co">#&gt; Astro_7  -47.6772669   4.389424 -32.7900052 -12.873335 -50.048431  -4.285490</span>
<span class="co">#&gt; Astro_8    4.9481780   5.714122  40.3087348 -15.165899  63.522566  40.643218</span>
<span class="co">#&gt; Astro_9   57.3574100 -27.110912  -7.4264464 -13.503395 -62.479762 -15.149021</span>
<span class="co">#&gt; Astro_10 -36.8266178  39.230523 -28.1701091  19.497967   5.065160  10.774765</span>
<span class="co">#&gt;                PC37       PC38       PC39       PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1   14.855016  -10.11556  44.987885  -1.603123 -10.913944   3.532696</span>
<span class="co">#&gt; Astro_2    8.209055   17.99842  11.958522  -1.160660 -14.047774  27.748590</span>
<span class="co">#&gt; Astro_3  -16.826524   61.64229  -9.844178 -21.411507   2.488139   1.793023</span>
<span class="co">#&gt; Astro_4  -26.533055  -27.99969  41.022934   2.619541 -35.245890 -17.649277</span>
<span class="co">#&gt; Astro_5   -7.481602  -28.87420  13.593954 -35.437497  68.248616  10.552585</span>
<span class="co">#&gt; Astro_6  -14.343387   38.67769  45.255607  37.858962   2.488781  30.377203</span>
<span class="co">#&gt; Astro_7  -17.255262   24.47769 -27.132180   2.208254  28.914535 -40.765960</span>
<span class="co">#&gt; Astro_8   31.977432 -117.06676  17.100593 -18.650597  -1.376020 -25.872966</span>
<span class="co">#&gt; Astro_9    5.844894   75.98707 -67.906012  22.940683  -9.384698  24.491932</span>
<span class="co">#&gt; Astro_10  13.834084  -23.70211 -49.007339 -66.883225 -20.692720  21.584268</span>
<span class="co">#&gt;                PC43       PC44       PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -3.423037 -11.714530  32.019640  11.102977 -15.999976   9.615514</span>
<span class="co">#&gt; Astro_2   38.422306  -3.887788  -8.935167 -25.460414  -9.317209 -16.214077</span>
<span class="co">#&gt; Astro_3    2.075867  37.528452  36.880078  16.896689  60.666430  75.294983</span>
<span class="co">#&gt; Astro_4   29.313772  -2.721654  15.410464  48.510178  51.695060  -2.683999</span>
<span class="co">#&gt; Astro_5  -15.124398   5.037238 -13.448577  -6.741581 -40.602004 -13.539482</span>
<span class="co">#&gt; Astro_6  -35.855526   6.740143 -51.189316   2.473525 -26.301169  12.621379</span>
<span class="co">#&gt; Astro_7   -4.242339 -23.805673  17.121075 -70.197528 -13.171656  15.313285</span>
<span class="co">#&gt; Astro_8   44.946541  26.593256 -51.743957  29.035974  22.815002 -29.596931</span>
<span class="co">#&gt; Astro_9  -27.879987 -14.152819  37.484664  23.676329  22.011163 -45.019834</span>
<span class="co">#&gt; Astro_10  29.750325  15.361822  14.822085  -9.924490 -33.250847 -20.503981</span>
<span class="co">#&gt;                 PC49       PC50</span>
<span class="co">#&gt; Astro_1   39.7779297 -20.706846</span>
<span class="co">#&gt; Astro_2  -19.5857938  17.066738</span>
<span class="co">#&gt; Astro_3   -3.7441160   2.527529</span>
<span class="co">#&gt; Astro_4   40.2784077  42.525293</span>
<span class="co">#&gt; Astro_5   -5.6143615   2.438562</span>
<span class="co">#&gt; Astro_6  -12.0540448 -11.198558</span>
<span class="co">#&gt; Astro_7    0.6882549  32.307024</span>
<span class="co">#&gt; Astro_8   10.0850350  30.966134</span>
<span class="co">#&gt; Astro_9   40.0871823 -13.207744</span>
<span class="co">#&gt; Astro_10 -61.8649769 -82.609289</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
