<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<div class="figure">
<img src="../figures/bandnorm_intro.png" alt=""><p class="caption">The introduction for BandNorm</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony converged after 6 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2        PC3       PC4         PC5        PC6</span>
<span class="co">#&gt; Astro_1  -21.590636 18.560367 -24.188752 28.610665 -31.2769299 -25.602400</span>
<span class="co">#&gt; Astro_2   -4.971023 42.860076  -7.986992  4.310964  -3.3341473  13.405687</span>
<span class="co">#&gt; Astro_3  -12.990328 13.436494   2.065068  9.229056   0.4472658 -34.795037</span>
<span class="co">#&gt; Astro_4   -8.649083 26.206090 -24.079058 11.213386 -10.8162402 -19.658327</span>
<span class="co">#&gt; Astro_5   14.306702 18.365927 -33.160930  4.140978 -17.3612312 -17.497638</span>
<span class="co">#&gt; Astro_6  -19.443862 15.170204 -18.016069 -1.410261  -2.8907658  10.920523</span>
<span class="co">#&gt; Astro_7  -26.107893 24.600856 -16.972531  1.221498  -8.9488508  -7.127529</span>
<span class="co">#&gt; Astro_8   -7.650069 27.188388  -3.679888 -3.695466   5.0032029  14.411574</span>
<span class="co">#&gt; Astro_9   -2.327826 19.380931 -30.033330  5.492537  -4.0796230  10.203688</span>
<span class="co">#&gt; Astro_10 -14.829330  5.839627 -11.110047  2.692948 -23.5522299  -6.688826</span>
<span class="co">#&gt;                 PC7         PC8        PC9       PC10      PC11       PC12</span>
<span class="co">#&gt; Astro_1    7.386865 -47.1103701  41.016478  -6.223400 25.504455 -26.657076</span>
<span class="co">#&gt; Astro_2   -8.333668  -0.8105483  12.971964  21.763254  5.253582 -28.136500</span>
<span class="co">#&gt; Astro_3  -40.088141 -15.3701565 -22.486957 -43.278274  0.137275 -37.519846</span>
<span class="co">#&gt; Astro_4   10.056041   3.4605838   2.097282 -10.577418  2.984861 -11.699438</span>
<span class="co">#&gt; Astro_5    2.822982   8.6348606  45.690144  25.763821 77.673605 -44.608557</span>
<span class="co">#&gt; Astro_6  -11.051822  -3.4265881  -2.803915   4.242302  9.332474 -40.575812</span>
<span class="co">#&gt; Astro_7   -3.007822  -1.3181194   1.247636 -10.556318  3.054452 -15.475170</span>
<span class="co">#&gt; Astro_8  -14.106386  23.1469730  -3.104324   2.143086 15.856261  -2.049197</span>
<span class="co">#&gt; Astro_9   15.459186  43.0601343  48.214407  21.861221 27.001378 -78.279520</span>
<span class="co">#&gt; Astro_10  12.588644  -3.2161487  28.111004 -13.693695 15.073818 -26.502156</span>
<span class="co">#&gt;                PC13       PC14       PC15       PC16         PC17       PC18</span>
<span class="co">#&gt; Astro_1   35.957434 -10.734839  25.156732  -9.764040  43.20432729 -42.956327</span>
<span class="co">#&gt; Astro_2   11.874227 -10.491648 -41.860157  -1.193108   8.66094770 -11.715536</span>
<span class="co">#&gt; Astro_3   13.635812   3.048385   9.649868  12.681772  35.42315277  -2.274114</span>
<span class="co">#&gt; Astro_4   -7.081535   9.410192 -16.700414  -0.564813  -0.09408939   1.259175</span>
<span class="co">#&gt; Astro_5   38.640058  35.373267 -34.253074  16.244519  23.30620512  20.972149</span>
<span class="co">#&gt; Astro_6   27.467159   1.261276   7.363917 -14.165561   1.24041162 -12.656055</span>
<span class="co">#&gt; Astro_7   19.553525  -3.668719  -5.365759   3.396505  10.99534518 -24.808170</span>
<span class="co">#&gt; Astro_8  -15.374003 -27.912352 -10.131826 -14.435481 -15.15187407   2.457861</span>
<span class="co">#&gt; Astro_9  -19.380681 -37.452626  -6.788385  35.137567  11.26713517 -43.935919</span>
<span class="co">#&gt; Astro_10  17.411185 -22.383016  -9.024475   3.141243  20.00664183 -25.367698</span>
<span class="co">#&gt;                PC19       PC20        PC21       PC22        PC23         PC24</span>
<span class="co">#&gt; Astro_1   34.078733  29.095098 -125.110026  63.358905  22.6285701  -49.1695484</span>
<span class="co">#&gt; Astro_2   -1.261360  -5.800269   -7.407561   2.168155 -22.6976494   17.4980892</span>
<span class="co">#&gt; Astro_3   51.929645  13.281582  -21.867524   5.593873  13.4733940   24.3373754</span>
<span class="co">#&gt; Astro_4   -1.495245  18.625042   10.813951  -9.969829 -14.1988295  -19.9250084</span>
<span class="co">#&gt; Astro_5  115.488513  18.051316   86.974870  21.217143 128.3773743 -104.3882977</span>
<span class="co">#&gt; Astro_6  -15.848040  -4.550650    3.068606  48.139212  -2.5670923    7.8468878</span>
<span class="co">#&gt; Astro_7   -6.324085   1.740016   -7.499318  16.156906 -22.3324920   -4.0425590</span>
<span class="co">#&gt; Astro_8  -16.935582 -24.622586    7.345238 -39.012418 -22.7268236   40.3232254</span>
<span class="co">#&gt; Astro_9  -90.680134 -60.167093    5.917338 -48.037433  48.1880422    6.7246464</span>
<span class="co">#&gt; Astro_10   0.258950  -8.947292   -3.176563 -20.867133   0.8141688   -0.7962204</span>
<span class="co">#&gt;                PC25        PC26       PC27        PC28        PC29         PC30</span>
<span class="co">#&gt; Astro_1    9.551919  -66.623998  24.038794 -126.269806   5.5002602 -26.18217438</span>
<span class="co">#&gt; Astro_2  -10.108615   18.779908   7.079220  -18.377870   3.7257699   9.85187370</span>
<span class="co">#&gt; Astro_3    1.372278   11.002065 -44.381960   26.239374  29.6213553   9.66559866</span>
<span class="co">#&gt; Astro_4    2.635084   12.415691  20.577740   20.328892   0.6771578 -11.14691095</span>
<span class="co">#&gt; Astro_5   46.236828   62.334378 -54.258416   60.775287 -50.5738696   8.21670989</span>
<span class="co">#&gt; Astro_6    4.815607    4.601172  31.099089   -9.358538  21.8140813   0.01171336</span>
<span class="co">#&gt; Astro_7   30.103244    1.599981  23.432027   10.341346  20.5349303 -23.44783753</span>
<span class="co">#&gt; Astro_8    4.949766   16.810694 -30.754312   51.667727   9.5609725 -22.13046654</span>
<span class="co">#&gt; Astro_9   12.167585 -171.807187  -1.615163   62.585362  -8.1516726 -13.48765086</span>
<span class="co">#&gt; Astro_10 -42.013022    3.036955  17.604853    3.256095  10.8652840  -2.73796857</span>
<span class="co">#&gt;                PC31       PC32       PC33       PC34        PC35       PC36</span>
<span class="co">#&gt; Astro_1   15.304086 -59.619236   2.955959 -15.364271  44.9958088 -66.762761</span>
<span class="co">#&gt; Astro_2   -8.017976   3.331875   9.687362  23.694065  -3.3636608   6.754429</span>
<span class="co">#&gt; Astro_3   19.684051 -17.113675 -57.706688  -7.428971   1.8387909  19.055637</span>
<span class="co">#&gt; Astro_4  -16.209902   1.602227 -12.952615   6.537337  10.7343231   3.353123</span>
<span class="co">#&gt; Astro_5  -16.364413  -3.998495  -1.456841 -43.880710  74.1390621  23.680273</span>
<span class="co">#&gt; Astro_6   -3.515392  39.086358  11.011791  -9.807929   8.8244274   1.407635</span>
<span class="co">#&gt; Astro_7  -30.898886   2.710658 -38.120117   3.900612 -50.9113250   2.460639</span>
<span class="co">#&gt; Astro_8    6.169376  27.667213  38.248636  -8.555413  38.9955651  44.973300</span>
<span class="co">#&gt; Astro_9   59.957776 -20.945851   2.321965  -6.844107 -54.3048395 -17.891862</span>
<span class="co">#&gt; Astro_10 -39.589573  36.746887 -32.042337  17.884666   0.4441527   4.633317</span>
<span class="co">#&gt;                PC37      PC38        PC39        PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1    2.745975  12.73982  38.0100703  -3.1128845   3.047620  11.524201</span>
<span class="co">#&gt; Astro_2    8.843665  23.28807  15.0099704   0.3650376 -12.664928  23.049839</span>
<span class="co">#&gt; Astro_3    4.533454  59.86152 -11.3368237 -31.7772507   3.018579   8.498137</span>
<span class="co">#&gt; Astro_4  -24.753434 -24.38000  40.8093909   0.9542982 -38.092721 -18.323550</span>
<span class="co">#&gt; Astro_5   -4.123858 -35.34449  29.5016385 -42.6772527  74.507529   6.831696</span>
<span class="co">#&gt; Astro_6  -11.767012  42.00513  45.2594895  36.1986676  -1.383576  29.595877</span>
<span class="co">#&gt; Astro_7  -23.675376  33.88801 -35.9937893   1.2296458  28.514053 -39.577104</span>
<span class="co">#&gt; Astro_8   22.472335 -94.55283  -0.3874293  -9.3919149   4.447043 -21.438597</span>
<span class="co">#&gt; Astro_9   11.653565  69.69688 -63.0981727  26.8636044 -11.535982  17.165256</span>
<span class="co">#&gt; Astro_10  16.766341 -20.76630 -48.6022052 -68.7789728 -24.216154  20.674140</span>
<span class="co">#&gt;                PC43       PC44       PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1  -19.264238 -20.016476  32.934968  -6.024707 -25.019187  25.813394</span>
<span class="co">#&gt; Astro_2   38.059816   2.999517  -9.331552 -23.423601  -1.511303  -5.894349</span>
<span class="co">#&gt; Astro_3   -7.633953  33.142150  23.670026  20.429195  42.898922  74.366083</span>
<span class="co">#&gt; Astro_4   33.145211   1.847414  17.661218  48.905391  56.555404   2.679288</span>
<span class="co">#&gt; Astro_5  -16.502902   5.590063 -10.244694  -7.693653 -43.964111   3.161473</span>
<span class="co">#&gt; Astro_6  -30.447392  10.755350 -48.221816   2.580513 -21.273880  18.987550</span>
<span class="co">#&gt; Astro_7   -6.420239  -7.826226  12.761391 -61.021672  -2.751353  10.084615</span>
<span class="co">#&gt; Astro_8   40.443489  20.920875 -57.706712  19.908409  35.837637 -33.125522</span>
<span class="co">#&gt; Astro_9  -17.907778 -10.855657  33.653629  30.827154  41.014387 -48.290657</span>
<span class="co">#&gt; Astro_10  35.637445  19.399288  17.957439  -9.906010 -27.580241 -14.132299</span>
<span class="co">#&gt;                PC49       PC50</span>
<span class="co">#&gt; Astro_1   28.976721 -25.188735</span>
<span class="co">#&gt; Astro_2  -22.042254  19.179869</span>
<span class="co">#&gt; Astro_3    6.246591   9.679576</span>
<span class="co">#&gt; Astro_4   44.537135  42.328597</span>
<span class="co">#&gt; Astro_5   -8.734697  -1.202317</span>
<span class="co">#&gt; Astro_6   -6.702543 -11.859677</span>
<span class="co">#&gt; Astro_7    3.127422  38.708469</span>
<span class="co">#&gt; Astro_8    8.378052  28.668286</span>
<span class="co">#&gt; Astro_9   45.936236 -14.517698</span>
<span class="co">#&gt; Astro_10 -56.010662 -82.965138</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
