<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD_tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html" class="external-link">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="format-of-input">Format of input<a class="anchor" aria-label="anchor" href="#format-of-input"></a>
</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; Loading required package: viridis</span>
<span class="co">#&gt; Warning: package 'viridis' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridisLite</span>
<span class="co">#&gt; Warning: package 'viridisLite' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: sparsesvd</span>
<span class="co">#&gt; Warning: package 'sparsesvd' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: GenomicInteractions</span>
<span class="co">#&gt; Loading required package: InteractionSet</span>
<span class="co">#&gt; Warning: package 'InteractionSet' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Warning: package 'GenomicRanges' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Warning: package 'BiocGenerics' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine, intersect, setdiff, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Warning: package 'S4Vectors' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:Matrix':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand</span>
<span class="co">#&gt; The following objects are masked from 'package:data.table':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, second</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, rename</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Warning: package 'IRanges' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'IRanges'</span>
<span class="co">#&gt; The following object is masked from 'package:data.table':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     shift</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse, desc, slice</span>
<span class="co">#&gt; The following object is masked from 'package:grDevices':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     windows</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Warning: package 'GenomeInfoDb' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: SummarizedExperiment</span>
<span class="co">#&gt; Warning: package 'SummarizedExperiment' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: MatrixGenerics</span>
<span class="co">#&gt; Warning: package 'MatrixGenerics' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'MatrixGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Warning: package 'Biobase' was built under R version 4.0.3</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biobase'</span>
<span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     rowMedians</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'SummarizedExperiment'</span>
<span class="co">#&gt; The following object is masked from 'package:SeuratObject':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Assays</span>
<span class="co">#&gt; The following object is masked from 'package:Seurat':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Assays</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="download-existing-single-cell-hi-c-data">Download existing single-cell Hi-C data<a class="anchor" aria-label="anchor" href="#download-existing-single-cell-hi-c-data"></a>
</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173" class="external-link">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z" class="external-link">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/" class="external-link">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z" class="external-link">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/" class="external-link">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># download_schic("Li2019", cell_type = "2i", cell_path = getwd(), summary_path = getwd())</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="use-bandnorm">Use BandNorm<a class="anchor" aria-label="anchor" href="#use-bandnorm"></a>
</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z" class="external-link">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span>, chrs <span class="op">=</span> <span class="st">"chr1"</span><span class="op">)</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony converged after 3 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;              [,1]        [,2]      [,3]       [,4]       [,5]        [,6]</span>
<span class="co">#&gt; Astro_1  260.3526   8.1135225 -37.99541 -27.637209 35.5155997  37.7551698</span>
<span class="co">#&gt; Astro_2  257.6080  -3.6042183 -44.97101  -2.874494 10.9749787   7.7033996</span>
<span class="co">#&gt; Astro_3  263.1855 -10.2760030 -14.89914  -4.794485  3.4979843 -14.9298880</span>
<span class="co">#&gt; Astro_4  257.3574  -4.3393372 -38.03208 -33.188010 10.0472560  10.3663865</span>
<span class="co">#&gt; Astro_5  253.7632 -21.9796554 -18.92307 -26.852030 11.7774452  23.7761340</span>
<span class="co">#&gt; Astro_6  255.0481   6.6770804 -28.01846 -28.661405 -3.4785882   3.0962115</span>
<span class="co">#&gt; Astro_7  263.0181  15.0558768 -32.85678 -15.017966  3.6887475   0.6919548</span>
<span class="co">#&gt; Astro_8  261.3980  -6.0394693 -46.63089  -7.240393  3.1012269   1.3377341</span>
<span class="co">#&gt; Astro_9  258.4566  -5.9404006 -20.04996 -23.811089 13.1282406  10.2811432</span>
<span class="co">#&gt; Astro_10 262.4341   0.9125993 -18.87508 -22.205064  0.5082808  23.4561662</span>
<span class="co">#&gt;                [,7]       [,8]        [,9]     [,10]       [,11]      [,12]</span>
<span class="co">#&gt; Astro_1   28.967413   3.773816 -56.7300982 43.027889  -7.2126966 -31.752076</span>
<span class="co">#&gt; Astro_2   -5.727039  10.756033  -7.8405881 12.720999  13.5040421 -18.652449</span>
<span class="co">#&gt; Astro_3   23.517734  24.264418  15.7123085  5.508611   4.2839173  -7.785308</span>
<span class="co">#&gt; Astro_4   21.206200  -2.132022   0.4309741  5.170276 -14.6706684   3.163833</span>
<span class="co">#&gt; Astro_5   26.605780  -1.682126   1.0757176 45.385754  16.1063905 -95.402760</span>
<span class="co">#&gt; Astro_6   -9.485467  19.074127  -6.4871828  1.257115  -0.5425305  -2.042503</span>
<span class="co">#&gt; Astro_7    7.830537  10.782299  -2.1790887 -3.859459 -10.3167749   8.503863</span>
<span class="co">#&gt; Astro_8  -11.238297  25.553017  13.1000179 -1.258949   1.0122390 -21.935382</span>
<span class="co">#&gt; Astro_9   -1.339360 -14.015318  35.1956871 48.225797  12.3488145 -44.868737</span>
<span class="co">#&gt; Astro_10   7.961125  -4.559699  -6.1356634 32.506359 -18.3962130  -8.052886</span>
<span class="co">#&gt;               [,13]       [,14]      [,15]      [,16]      [,17]      [,18]</span>
<span class="co">#&gt; Astro_1  25.5633041  47.9398495  22.478343  27.188739 -17.641651  47.957222</span>
<span class="co">#&gt; Astro_2  27.4616831   7.0380738  -2.215954  -9.449302   9.252975 -21.939760</span>
<span class="co">#&gt; Astro_3   7.0575463  23.8576762  -3.670740   1.379607  13.325759  10.715544</span>
<span class="co">#&gt; Astro_4  16.5184554   0.9389263  -5.080709 -19.610403  -3.991777  -2.111688</span>
<span class="co">#&gt; Astro_5  44.9230402  31.8834440 -50.774116   5.351053  29.076329 -11.439970</span>
<span class="co">#&gt; Astro_6  46.8716792  35.9103409   4.174828   3.446433 -18.666068  -1.316567</span>
<span class="co">#&gt; Astro_7   8.2137701  26.2113260   4.844540 -11.848266   1.978565   9.397178</span>
<span class="co">#&gt; Astro_8   0.9325309  -3.9272734  39.201864  -7.862755 -20.689068 -10.270790</span>
<span class="co">#&gt; Astro_9  78.6731736 -26.0037345  22.067146  32.500969  47.603801 -24.287294</span>
<span class="co">#&gt; Astro_10 33.3363916  26.0556176  27.831414 -13.340627  -2.357419  17.038993</span>
<span class="co">#&gt;               [,19]        [,20]       [,21]       [,22]      [,23]       [,24]</span>
<span class="co">#&gt; Astro_1  -52.262475  -45.0162052 -33.7416459  174.551434  72.239557  -16.155404</span>
<span class="co">#&gt; Astro_2  -17.208324   -0.2116591   2.2275211   -6.360190  -8.934007    7.498682</span>
<span class="co">#&gt; Astro_3   -6.201621  -35.0432312 -17.2175763    5.773774  10.779726    7.895256</span>
<span class="co">#&gt; Astro_4    1.505483   10.4462011 -16.6801691  -10.450704  -7.604244   25.629045</span>
<span class="co">#&gt; Astro_5   13.871873 -117.9575051 -21.6247078 -102.136654   8.563661 -147.542568</span>
<span class="co">#&gt; Astro_6  -11.998566   27.3764515   7.3103715   -3.101594  50.498649   16.208089</span>
<span class="co">#&gt; Astro_7  -22.617429    8.0282563   0.1499101    3.737578  17.949162   28.357815</span>
<span class="co">#&gt; Astro_8   -6.278820    6.1563417  19.9056478   40.313335 -28.522353   28.879956</span>
<span class="co">#&gt; Astro_9  -50.846390   88.2602466  55.5924631  -22.190134 -61.477880  -67.710048</span>
<span class="co">#&gt; Astro_10 -24.645220   11.7535037  11.5168801    1.178594 -18.640846   12.993880</span>
<span class="co">#&gt;                [,25]      [,26]       [,27]      [,28]        [,29]      [,30]</span>
<span class="co">#&gt; Astro_1   -55.042277  20.529774  81.6833577 -14.006380 -152.9556685   9.702145</span>
<span class="co">#&gt; Astro_2     1.659384  -2.935077  -7.6750711   5.194189   -0.9210722  -6.059692</span>
<span class="co">#&gt; Astro_3    -8.537257  10.512533 -16.8470436  33.011104   13.0741382  35.953096</span>
<span class="co">#&gt; Astro_4   -17.186685   3.461941 -11.3528451 -19.661874   19.1088363   7.731867</span>
<span class="co">#&gt; Astro_5  -123.116104  54.704392 -50.2149655  68.765068   81.6844453 -63.256249</span>
<span class="co">#&gt; Astro_6    12.252912   4.247862  -2.6202268 -31.481901  -10.1413407  30.486520</span>
<span class="co">#&gt; Astro_7    -6.847864  36.487597 -11.2550944 -23.374599   11.8233896  18.786469</span>
<span class="co">#&gt; Astro_8    35.196933  14.487016  -0.8782535  40.717700   25.2400467  13.244034</span>
<span class="co">#&gt; Astro_9   -13.393167  21.346784 186.1971617  16.250007   82.8392727 -19.525716</span>
<span class="co">#&gt; Astro_10    3.013840 -42.589724  -1.3326694 -18.204384    3.0854205  19.265504</span>
<span class="co">#&gt;               [,31]       [,32]      [,33]      [,34]       [,35]      [,36]</span>
<span class="co">#&gt; Astro_1   42.662657  25.4670451  82.027913 -17.092693  26.9200210  73.630118</span>
<span class="co">#&gt; Astro_2  -11.422312  -0.8734051   2.477016 -11.995369 -13.0049463  -9.488378</span>
<span class="co">#&gt; Astro_3   -2.419629  17.5747169  21.855586  76.702736   7.4504649   5.813165</span>
<span class="co">#&gt; Astro_4   15.239538 -18.3586703 -12.167009   4.871560 -15.5756698  22.489379</span>
<span class="co">#&gt; Astro_5  -10.631646  -7.0786479   9.170275  -5.909235  54.9328361  68.462031</span>
<span class="co">#&gt; Astro_6    3.592436  -7.6340952 -52.536336 -19.703626  -2.1901094  22.010548</span>
<span class="co">#&gt; Astro_7   28.784114 -26.5766528   3.142350  34.330293  -0.3827319 -56.225730</span>
<span class="co">#&gt; Astro_8   38.566491  16.1438966  -5.448540 -56.535512  18.6230539  62.994826</span>
<span class="co">#&gt; Astro_9    9.861795  68.0596352  27.368291  -1.833168  21.1641710 -59.568140</span>
<span class="co">#&gt; Astro_10   6.643161 -43.3034033 -51.852029  23.173428 -29.6596145  13.716246</span>
<span class="co">#&gt;               [,37]      [,38]      [,39]      [,40]       [,41]       [,42]</span>
<span class="co">#&gt; Astro_1  -76.487700 -10.434532  -11.75673 -53.308771   9.4383957   1.5908169</span>
<span class="co">#&gt; Astro_2    3.772680  -7.469229   28.21647  -9.081047  -0.7593826  10.1042145</span>
<span class="co">#&gt; Astro_3    1.075691  14.860347   61.52607   2.290494  27.2869867 -11.1039271</span>
<span class="co">#&gt; Astro_4   22.994920  25.496448  -27.19201 -38.745578  -3.9972197  47.8042120</span>
<span class="co">#&gt; Astro_5   21.232105   4.798170  -31.17273 -21.035657  43.1657705 -77.6513207</span>
<span class="co">#&gt; Astro_6   24.907350  11.950930   38.08774 -43.651911 -39.4565051  12.8459988</span>
<span class="co">#&gt; Astro_7    3.647619  19.878727   43.94684  35.965018  -9.4068593 -30.8054842</span>
<span class="co">#&gt; Astro_8   30.391384 -36.721498 -119.53660 -11.654340  15.1845487   0.8869246</span>
<span class="co">#&gt; Astro_9  -22.815026  -6.894744   75.94917  68.979935 -26.1733785   8.5960743</span>
<span class="co">#&gt; Astro_10  24.929773 -14.661624  -23.20090  51.934605  65.0899455  35.9871547</span>
<span class="co">#&gt;               [,43]      [,44]      [,45]     [,46]      [,47]      [,48]</span>
<span class="co">#&gt; Astro_1   -7.898727  -9.739691  10.134096  34.97171  -6.851681  27.011539</span>
<span class="co">#&gt; Astro_2  -30.000613  33.349661  -2.559901 -10.05259  26.129130   9.623988</span>
<span class="co">#&gt; Astro_3   -5.849908  -3.535801 -39.435809  39.33388 -13.352501 -51.297440</span>
<span class="co">#&gt; Astro_4   15.997035  23.629368  -3.686791  16.85451 -56.374371 -34.393722</span>
<span class="co">#&gt; Astro_5  -15.101235 -21.351504  -6.455596 -11.14534   9.488932  52.425610</span>
<span class="co">#&gt; Astro_6  -32.833728 -41.778814 -14.162631 -51.49212 -13.797670  48.035272</span>
<span class="co">#&gt; Astro_7   39.353948 -11.188075  13.123469  13.07211  71.250235   9.060424</span>
<span class="co">#&gt; Astro_8   24.960013  51.152417 -29.345156 -54.53558 -35.476197 -33.973106</span>
<span class="co">#&gt; Astro_9  -25.460816 -22.626667  10.063097  34.11381 -27.494558 -33.193473</span>
<span class="co">#&gt; Astro_10 -23.864329  23.918758 -22.362002  14.49189   1.206429  54.400720</span>
<span class="co">#&gt;               [,49]       [,50]</span>
<span class="co">#&gt; Astro_1  -19.897417  41.7590572</span>
<span class="co">#&gt; Astro_2    9.707357 -18.2709970</span>
<span class="co">#&gt; Astro_3  -84.611610  -2.2426752</span>
<span class="co">#&gt; Astro_4   -1.499401  37.4954592</span>
<span class="co">#&gt; Astro_5    3.023368  -3.6162898</span>
<span class="co">#&gt; Astro_6  -17.146731 -15.7506927</span>
<span class="co">#&gt; Astro_7  -22.210508   0.2327628</span>
<span class="co">#&gt; Astro_8   39.038464  21.1447653</span>
<span class="co">#&gt; Astro_9   54.492503  51.1732646</span>
<span class="co">#&gt; Astro_10  15.964921 -65.5154282</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
