<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<div class="figure">
<img src="../figures/bandnorm_intro.png" alt=""><p class="caption">The introduction for BandNorm</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1  38.000000</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 105.000000</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 116.000000</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 133.000000</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 112.000000</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony 10/10</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2         PC3        PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1  -18.858459 26.929972 -20.4148901  36.182105 -39.228443 -24.531952</span>
<span class="co">#&gt; Astro_2    9.353232 52.998638 -48.9166137 -43.402489 -12.285634   6.207424</span>
<span class="co">#&gt; Astro_3   -1.061167  4.010516   2.7440664   4.259482  12.826951 -19.415611</span>
<span class="co">#&gt; Astro_4   -3.325972 27.762970 -25.0860251  12.016578  -1.493958 -10.346122</span>
<span class="co">#&gt; Astro_5    3.090067  9.594368 -18.4525407   9.968954 -22.816327 -22.905345</span>
<span class="co">#&gt; Astro_6  -16.760000 16.935415 -21.6631976  -2.952196  -4.242104  13.963459</span>
<span class="co">#&gt; Astro_7  -24.339050 23.813101 -12.3716902   0.323278  -2.713555  -4.073692</span>
<span class="co">#&gt; Astro_8   -1.886558 36.937931   0.7605835   5.327121   6.799016  21.817579</span>
<span class="co">#&gt; Astro_9  -13.563187 10.616172 -15.3081622  11.276743  -9.482299   4.814173</span>
<span class="co">#&gt; Astro_10 -12.172483  7.464188 -14.6613860   1.289550 -24.598008  -3.479463</span>
<span class="co">#&gt;                   PC7         PC8       PC9       PC10       PC11         PC12</span>
<span class="co">#&gt; Astro_1    6.89674155 -50.4354034 39.085906  -5.185349 36.3274530 -20.65201710</span>
<span class="co">#&gt; Astro_2   -0.02433428   5.8892741 14.011301  10.541066 29.5347082 -37.19502958</span>
<span class="co">#&gt; Astro_3  -13.13837943  22.1595580  1.617257   6.315454 12.3372380  -2.34592952</span>
<span class="co">#&gt; Astro_4    9.54532640  -2.5109600  8.052191 -11.111286  0.2768933 -15.67678305</span>
<span class="co">#&gt; Astro_5    1.10791010  -1.5621463 35.749721  16.571760 79.6845360 -32.96762174</span>
<span class="co">#&gt; Astro_6   -8.03329120   0.1150059 -2.399883   1.755151  7.4556109 -42.43299304</span>
<span class="co">#&gt; Astro_7    0.27742091   5.2740444 -7.225785  -8.534154 -2.9774127  -5.07780285</span>
<span class="co">#&gt; Astro_8  -18.08441738  10.4160150  1.699783   4.409333 25.1525562   0.05563992</span>
<span class="co">#&gt; Astro_9   13.77679037  32.8550425 38.254876  12.718636 28.9562033 -66.59740691</span>
<span class="co">#&gt; Astro_10  15.54753056   0.1697525 28.696494 -16.250197 12.8383921 -28.56413904</span>
<span class="co">#&gt;                 PC13        PC14         PC15        PC16       PC17       PC18</span>
<span class="co">#&gt; Astro_1   29.0235253 -16.1228042  25.49005835 -12.5789967  44.899279 -46.164126</span>
<span class="co">#&gt; Astro_2    0.1097689  -1.2883049  -9.95346496  11.3575421  -9.014140 -13.663738</span>
<span class="co">#&gt; Astro_3    5.3226086   9.6825464  -0.46658322  16.9526803   6.134275  -0.408486</span>
<span class="co">#&gt; Astro_4   -0.2593494  11.8795904 -11.96344070  -0.1654546   2.627561  -7.196075</span>
<span class="co">#&gt; Astro_5   32.8096889  46.5378728   0.06812889  12.6024599  -1.317309  15.753106</span>
<span class="co">#&gt; Astro_6   17.1608340   2.0167053   1.39050410 -15.3827184  -5.186769  -6.253107</span>
<span class="co">#&gt; Astro_7    9.1860603  -0.3819316 -13.46750260   6.0776282   6.316227 -16.300933</span>
<span class="co">#&gt; Astro_8   -4.8720606 -33.3303006  -0.23547609 -16.4916979  -4.777160 -14.517968</span>
<span class="co">#&gt; Astro_9  -25.2027491 -26.3086608  27.44005067  31.4491042 -13.276322 -49.152121</span>
<span class="co">#&gt; Astro_10   7.4409798 -21.5428860 -14.93069115   1.9109951  13.567090 -19.037459</span>
<span class="co">#&gt;                PC19       PC20         PC21       PC22        PC23        PC24</span>
<span class="co">#&gt; Astro_1   45.918689  29.216478 -157.8607638  60.416330  24.6123784  -53.472277</span>
<span class="co">#&gt; Astro_2   -1.005361  -4.076626    0.4495948 -10.335220  -7.1644299   12.757067</span>
<span class="co">#&gt; Astro_3   35.929114  13.379761    9.7637731  -2.925745   0.5249199   -7.065342</span>
<span class="co">#&gt; Astro_4   -8.506592  17.446869    9.2301192   1.041819 -16.7160931  -25.786372</span>
<span class="co">#&gt; Astro_5  117.952610  28.221927   83.1243590  12.283861 120.2728541 -101.717302</span>
<span class="co">#&gt; Astro_6  -26.538322 -11.242531   19.8188406  36.391431  -7.8236844   13.107631</span>
<span class="co">#&gt; Astro_7   -6.866999  -4.082582   10.3350550   5.339154 -21.1964140   -5.475874</span>
<span class="co">#&gt; Astro_8   -3.872958 -19.787418  -42.2116318 -19.410544 -20.8510066   26.586252</span>
<span class="co">#&gt; Astro_9  -88.214430 -50.006152    2.0267440 -56.923734  40.0615554    9.411762</span>
<span class="co">#&gt; Astro_10 -10.843120 -15.682185   14.1597587 -32.314861  -4.7122785    4.514952</span>
<span class="co">#&gt;                PC25         PC26       PC27        PC28        PC29       PC30</span>
<span class="co">#&gt; Astro_1   15.532773  -75.8381814  18.475225 -135.562428  -2.5122717 -35.264694</span>
<span class="co">#&gt; Astro_2    2.547951   -7.8538146  -6.018598    2.465334  11.2974315  20.576822</span>
<span class="co">#&gt; Astro_3    6.627157   22.8166114 -28.592999   29.932438  24.8345226   9.863498</span>
<span class="co">#&gt; Astro_4    8.462077   10.5845987  21.909247    2.234805  -0.7287303  -8.108983</span>
<span class="co">#&gt; Astro_5   42.218611   67.3292542 -59.302138   51.013764 -50.5242654  15.642933</span>
<span class="co">#&gt; Astro_6    0.538839    9.4530633  36.044164    7.403809  18.5261607   4.195849</span>
<span class="co">#&gt; Astro_7   33.501183   15.2357173  27.181508   27.597959   9.5725385 -21.058564</span>
<span class="co">#&gt; Astro_8   19.553850    0.4208499 -38.444406    8.928798   5.0485710 -32.489481</span>
<span class="co">#&gt; Astro_9    8.181547 -166.7539319  -6.629677   52.768679  -8.0966863  -6.052393</span>
<span class="co">#&gt; Astro_10 -46.410278    8.1044025  22.740974   19.745603   7.6972070   1.651165</span>
<span class="co">#&gt;                PC31       PC32       PC33        PC34       PC35       PC36</span>
<span class="co">#&gt; Astro_1   17.459999 -78.051983  15.349586 -30.3848281  60.466971 -87.330169</span>
<span class="co">#&gt; Astro_2  -22.249129   6.851965   8.857795  -0.6720799  -5.393287   1.623244</span>
<span class="co">#&gt; Astro_3    9.420120 -19.633046 -78.107916  -7.9751806  -8.796425 -12.141150</span>
<span class="co">#&gt; Astro_4  -26.951953   8.180778  -3.200784  12.5121955 -10.666999  39.906091</span>
<span class="co">#&gt; Astro_5  -16.400210  -7.795099  -5.540964 -46.4802889  70.058534  19.037051</span>
<span class="co">#&gt; Astro_6  -14.744711  56.782829  17.008017  -0.8723065  11.181576  12.356657</span>
<span class="co">#&gt; Astro_7  -35.406023   0.768624 -33.410925  -0.3672865 -67.658930  -7.082017</span>
<span class="co">#&gt; Astro_8    7.944586   2.220515  54.564446 -23.8914466  33.445501  53.588893</span>
<span class="co">#&gt; Astro_9   59.936663 -24.798919  -1.766434  -9.4590759 -58.409470 -22.589748</span>
<span class="co">#&gt; Astro_10 -51.032975  55.045635 -26.068281  27.4514978   2.193711  16.891209</span>
<span class="co">#&gt;                PC37       PC38       PC39       PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1   19.353550  -18.34551  60.548052  -7.199720  -1.298858   7.423820</span>
<span class="co">#&gt; Astro_2    5.309627   11.26816  15.402118  -4.710591  -9.085915  20.010125</span>
<span class="co">#&gt; Astro_3  -12.443091   54.84251   4.245136 -25.399493  11.592938   5.333743</span>
<span class="co">#&gt; Astro_4  -37.312523  -13.06087  40.075186  14.536630 -29.967013   4.069661</span>
<span class="co">#&gt; Astro_5  -10.580103  -38.36581  27.102354 -42.145426  73.686690   6.113492</span>
<span class="co">#&gt; Astro_6  -11.344049   31.92000  50.953104  41.603049 -12.047837  32.034897</span>
<span class="co">#&gt; Astro_7  -17.037545   35.54484 -29.544880  10.625745  31.433122 -40.553142</span>
<span class="co">#&gt; Astro_8   25.832367 -105.26400  15.323855  -5.366660  16.770508  -5.485172</span>
<span class="co">#&gt; Astro_9    5.217775   66.69333 -65.486138  27.405244 -12.314153  16.414392</span>
<span class="co">#&gt; Astro_10  16.584758  -30.02364 -43.385890 -63.013413 -34.749776  23.682299</span>
<span class="co">#&gt;                PC43       PC44       PC45        PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1    4.555896 -12.240405  24.601881   8.1597814  -5.757584   4.399773</span>
<span class="co">#&gt; Astro_2   34.741702 -12.056682  -6.166504 -34.6861347  -8.287352 -11.662032</span>
<span class="co">#&gt; Astro_3   10.123044  37.375478  29.435024  14.4679534  72.163631  69.466921</span>
<span class="co">#&gt; Astro_4    6.011482   9.175086  28.671946  47.3276484  25.859523  -2.927588</span>
<span class="co">#&gt; Astro_5  -11.809295   6.608502 -10.870861 -11.2756559 -41.726562  11.275346</span>
<span class="co">#&gt; Astro_6  -28.380423  12.451239 -62.173402  12.2960358 -26.378125   1.459014</span>
<span class="co">#&gt; Astro_7    2.061523 -16.479264   6.534307 -71.4391439   9.844556   8.174128</span>
<span class="co">#&gt; Astro_8   33.209900  35.525292 -42.654864  24.1824009  25.470194 -43.165527</span>
<span class="co">#&gt; Astro_9  -13.172479  -9.871281  33.050274  27.2141019  43.301760 -40.113758</span>
<span class="co">#&gt; Astro_10  36.654124  21.240205   4.308096  -0.3324222 -33.800059 -31.385282</span>
<span class="co">#&gt;                PC49       PC50</span>
<span class="co">#&gt; Astro_1   45.875700 -15.295847</span>
<span class="co">#&gt; Astro_2  -18.950920  19.800772</span>
<span class="co">#&gt; Astro_3    1.862580   8.979707</span>
<span class="co">#&gt; Astro_4   25.214726  20.439967</span>
<span class="co">#&gt; Astro_5  -10.282589   4.402969</span>
<span class="co">#&gt; Astro_6  -11.596792 -13.054838</span>
<span class="co">#&gt; Astro_7    3.892795  45.457285</span>
<span class="co">#&gt; Astro_8    9.302509  16.565572</span>
<span class="co">#&gt; Astro_9   44.387168  -8.890063</span>
<span class="co">#&gt; Astro_10 -61.692156 -84.844219</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
