<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; Warning: package 'rmarkdown' was built under R version 4.0.5</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony converged after 7 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                  PC1       PC2         PC3        PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1  -17.4176728 28.640209 -22.3764360  33.601768 -36.261823 -16.973427</span>
<span class="co">#&gt; Astro_2  -14.7534909 33.127959   2.3909745  10.772931  -6.082757   9.483162</span>
<span class="co">#&gt; Astro_3    0.3782542  5.393258   0.5081808   1.628431  16.223578 -11.763774</span>
<span class="co">#&gt; Astro_4   -7.7004284 29.141338 -27.9868303   7.443200  -9.996266 -15.573331</span>
<span class="co">#&gt; Astro_5    3.7816457  7.339258 -21.8853389  11.354060 -21.937231 -22.419036</span>
<span class="co">#&gt; Astro_6  -18.5732357 16.028582 -23.2775843  -3.618943  -4.376977   9.494114</span>
<span class="co">#&gt; Astro_7  -22.1545077 27.699069 -21.5546806 -10.407249  -2.965215  -5.007962</span>
<span class="co">#&gt; Astro_8    1.6836471 42.949661   1.4034796   4.772469  -2.159732  12.415489</span>
<span class="co">#&gt; Astro_9  -12.8622171  8.368211 -18.8144326  12.623711  -8.613994   5.274317</span>
<span class="co">#&gt; Astro_10 -15.0835371  4.808163 -13.4615228   3.922271 -24.966316  -9.238266</span>
<span class="co">#&gt;                 PC7          PC8         PC9       PC10      PC11       PC12</span>
<span class="co">#&gt; Astro_1    4.534953 -47.69303387 38.90785998  -2.899750 31.747502 -27.346440</span>
<span class="co">#&gt; Astro_2   -9.047229  -8.29995771  6.76355227  13.395959  4.558958 -18.162244</span>
<span class="co">#&gt; Astro_3  -15.552695  25.10160328  1.44743315   8.737697  7.837773  -8.958316</span>
<span class="co">#&gt; Astro_4    8.078645   2.12817997  2.11804209 -13.195984 -6.381440 -13.024414</span>
<span class="co">#&gt; Astro_5    2.700738  -0.08388356 39.68634520  15.901929 80.375063 -35.045770</span>
<span class="co">#&gt; Astro_6   -9.935260  -3.64532886  1.87974430   1.529228  7.709511 -43.860775</span>
<span class="co">#&gt; Astro_7   -2.582141   3.00539622 -4.32968189  -7.268729 -2.456077 -10.742317</span>
<span class="co">#&gt; Astro_8  -20.701587  12.28003206 -0.03921554   0.266245 26.395019   2.352171</span>
<span class="co">#&gt; Astro_9   15.341119  34.32173517 42.18143564  12.047269 29.634253 -68.671247</span>
<span class="co">#&gt; Astro_10  14.501111  -3.83329039 33.60773521 -16.710056 14.249362 -28.886806</span>
<span class="co">#&gt;                PC13       PC14        PC15       PC16        PC17        PC18</span>
<span class="co">#&gt; Astro_1   30.971073 -18.454055  19.5738964 -13.081318  43.2884793 -45.6068819</span>
<span class="co">#&gt; Astro_2    5.716159  -2.429419 -13.9076734  -4.439544 -10.2444372 -14.2955180</span>
<span class="co">#&gt; Astro_3    7.224886   7.505230  -6.3758673  16.562470   4.6358074   0.1926291</span>
<span class="co">#&gt; Astro_4   -4.632777   3.929068 -22.1965774  -8.354175   0.8384303   4.2150682</span>
<span class="co">#&gt; Astro_5   30.578500  45.883706   0.5375785  14.219731  -0.6034274  17.1297287</span>
<span class="co">#&gt; Astro_6   26.872823  -1.453893   4.8572276 -15.032970   1.9171376 -11.5165644</span>
<span class="co">#&gt; Astro_7   16.577513  -2.768065 -10.5878811   6.682498  12.1134071 -21.1849821</span>
<span class="co">#&gt; Astro_8    4.237907 -55.335834  -0.6390919  -7.359417 -11.8734215   1.3451980</span>
<span class="co">#&gt; Astro_9  -27.397196 -26.981318  27.9325912  33.054401 -12.5178399 -47.7909862</span>
<span class="co">#&gt; Astro_10  16.745274 -24.548985 -11.0625769   3.250906  20.1912689 -24.4246697</span>
<span class="co">#&gt;                 PC19       PC20        PC21       PC22        PC23        PC24</span>
<span class="co">#&gt; Astro_1   35.5577932  18.406417 -141.501378  57.317615  18.5763607  -41.519853</span>
<span class="co">#&gt; Astro_2   -1.6151287   1.301126  -10.670435  -1.808620 -28.9321327   19.595912</span>
<span class="co">#&gt; Astro_3   25.6629597   2.417551   26.187710  -6.178041  -5.6708654    4.740579</span>
<span class="co">#&gt; Astro_4  -12.6565705  15.574520   13.385498  -7.722992 -31.3169884   -8.380131</span>
<span class="co">#&gt; Astro_5  115.5409406  26.066239   85.001252  15.221265 124.6045462 -103.663869</span>
<span class="co">#&gt; Astro_6  -21.4155905  -3.446328    3.445456  47.993474  -5.9039535    8.763484</span>
<span class="co">#&gt; Astro_7   -6.5031516  -1.732311    2.819872  12.358947 -22.3743707   -6.096222</span>
<span class="co">#&gt; Astro_8    0.4345102 -21.252375  -37.077112 -25.179668 -32.1514520   32.550400</span>
<span class="co">#&gt; Astro_9  -90.5973573 -52.111460    3.780557 -53.933148  44.3544681    7.445945</span>
<span class="co">#&gt; Astro_10  -3.8968947  -6.331250   -3.185879 -20.700933   0.2786888   -2.298422</span>
<span class="co">#&gt;                PC25         PC26       PC27        PC28       PC29       PC30</span>
<span class="co">#&gt; Astro_1   13.047944  -68.4572855  16.524973 -126.150035   4.104362 -33.476547</span>
<span class="co">#&gt; Astro_2  -12.247049   21.6369737   5.127722  -25.462233   5.569868  13.021824</span>
<span class="co">#&gt; Astro_3    4.047332   30.1625066 -30.545980   39.441670  31.290534  11.819228</span>
<span class="co">#&gt; Astro_4   -2.778717   17.9984987  27.167413   15.980864  10.882714 -12.880600</span>
<span class="co">#&gt; Astro_5   43.981409   62.9056379 -58.050849   55.566255 -49.613485  12.444214</span>
<span class="co">#&gt; Astro_6    3.211131   -0.5734338  33.590888   -9.216997  23.663390   0.903483</span>
<span class="co">#&gt; Astro_7   33.940842   10.4066871  24.935409   18.656923  15.055483 -25.274789</span>
<span class="co">#&gt; Astro_8   13.886598   -2.4820052 -31.211242   24.669472  14.129846 -43.437984</span>
<span class="co">#&gt; Astro_9    9.941331 -171.1972288  -5.371510   57.205826  -7.154668  -9.284975</span>
<span class="co">#&gt; Astro_10 -42.890368   -3.9991989  19.884859    3.340287  11.047972  -1.234239</span>
<span class="co">#&gt;                 PC31       PC32       PC33        PC34       PC35       PC36</span>
<span class="co">#&gt; Astro_1    7.8315359 -56.602154  24.497247 -19.4089401  60.278980 -67.998705</span>
<span class="co">#&gt; Astro_2   -9.6897095   3.776238   9.646572  20.2709350  -6.269714   5.524498</span>
<span class="co">#&gt; Astro_3   -0.2682587   1.787032 -69.060631   2.8623180  -9.133313   6.985820</span>
<span class="co">#&gt; Astro_4  -26.1529702  22.653694  -7.571578  17.8785156  16.030386  22.082895</span>
<span class="co">#&gt; Astro_5  -17.9126108  -3.130256  -2.040178 -48.5017290  71.239777  22.163127</span>
<span class="co">#&gt; Astro_6   -4.3600697  43.788125  12.908247  -6.8101252  14.203896  10.387055</span>
<span class="co">#&gt; Astro_7  -30.3918830  -1.734462 -35.141855   0.3409764 -63.843551  -1.857040</span>
<span class="co">#&gt; Astro_8   15.0763230  13.645525  47.082241 -25.0419800  58.337868  33.809758</span>
<span class="co">#&gt; Astro_9   58.4562564 -20.180832   1.689658 -11.4873817 -57.197585 -19.484217</span>
<span class="co">#&gt; Astro_10 -38.1470857  37.994621 -31.232474  18.7377728   4.579268  11.168639</span>
<span class="co">#&gt;                PC37       PC38      PC39         PC40        PC41       PC42</span>
<span class="co">#&gt; Astro_1   18.832755  -20.43063  45.68468  -3.45560886  -0.8481624   4.674073</span>
<span class="co">#&gt; Astro_2    7.116087   21.66101  15.20692  -0.04128949 -13.4600045  22.587491</span>
<span class="co">#&gt; Astro_3  -13.074406   52.55659 -10.38646 -21.53285353  12.0090238   2.597177</span>
<span class="co">#&gt; Astro_4  -26.083111  -30.29398  39.43702   5.63135475 -46.6266591 -17.258294</span>
<span class="co">#&gt; Astro_5   -6.539265  -37.11084  28.80854 -44.03089507  73.9194949   7.158959</span>
<span class="co">#&gt; Astro_6  -16.047342   30.38081  46.15957  34.77702740  -9.5105976  31.726090</span>
<span class="co">#&gt; Astro_7  -19.127264   35.07010 -37.00469   7.84570539  33.9036946 -41.370074</span>
<span class="co">#&gt; Astro_8   35.468314 -110.02991  13.93903 -12.45468526   2.6458097 -19.831518</span>
<span class="co">#&gt; Astro_9    9.254088   67.96276 -63.74380  25.51111595 -12.0844988  17.456624</span>
<span class="co">#&gt; Astro_10  11.498606  -32.18451 -47.31483 -71.28493278 -32.1278260  23.741770</span>
<span class="co">#&gt;                 PC43       PC44       PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -6.4226492 -10.372188  16.603325   6.090436 -28.813363   3.233874</span>
<span class="co">#&gt; Astro_2   34.7514282   1.551253  -9.348577 -24.994177  -3.259521  -4.157615</span>
<span class="co">#&gt; Astro_3   -0.6019272  39.150295  21.323536  12.296393  49.513398  68.069723</span>
<span class="co">#&gt; Astro_4   24.7815347   3.649895  10.389494  54.093853  36.200929  -1.137512</span>
<span class="co">#&gt; Astro_5  -20.3329561   4.982379 -10.616189  -8.526580 -46.263991   3.229538</span>
<span class="co">#&gt; Astro_6  -29.8460638  10.292486 -51.787219   9.351135 -26.192349   8.690574</span>
<span class="co">#&gt; Astro_7   -1.8744299 -16.453084  12.585293 -73.353553   3.003089  12.713421</span>
<span class="co">#&gt; Astro_8   40.7490304  34.437886 -54.984755  26.284004  32.227146 -39.097879</span>
<span class="co">#&gt; Astro_9  -21.6904418 -11.501838  33.361109  29.969348  38.792047 -48.119415</span>
<span class="co">#&gt; Astro_10  37.4505878  18.351100  15.454004  -3.424597 -29.551979 -24.744740</span>
<span class="co">#&gt;                PC49        PC50</span>
<span class="co">#&gt; Astro_1   32.837366 -24.9876279</span>
<span class="co">#&gt; Astro_2  -25.494445  21.0696629</span>
<span class="co">#&gt; Astro_3  -11.155537  -0.5350497</span>
<span class="co">#&gt; Astro_4   33.584708  34.7346347</span>
<span class="co">#&gt; Astro_5  -11.504925   0.3825573</span>
<span class="co">#&gt; Astro_6  -10.190548  -7.5309290</span>
<span class="co">#&gt; Astro_7    4.133794  44.4430464</span>
<span class="co">#&gt; Astro_8   22.746427  25.4800060</span>
<span class="co">#&gt; Astro_9   43.197499 -12.9081923</span>
<span class="co">#&gt; Astro_10 -58.576217 -76.0753316</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
