<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony 10/10</span>
<span class="co">#&gt; Harmony converged after 10 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2          PC3        PC4         PC5        PC6</span>
<span class="co">#&gt; Astro_1  -23.073491 30.297083 -19.73875216 30.8619717 -28.7363811 -18.540817</span>
<span class="co">#&gt; Astro_2   -8.599776 33.214895   0.81589421 14.6415949 -14.0820704   8.741318</span>
<span class="co">#&gt; Astro_3   -2.691967  3.040100  -1.28871168  7.1321228   8.0866667 -20.544098</span>
<span class="co">#&gt; Astro_4   -8.652986 26.576039 -28.11022567 12.7413986 -14.2926579 -17.951178</span>
<span class="co">#&gt; Astro_5    7.577649 11.212912 -18.89976662  7.0645847 -14.1848900 -15.828268</span>
<span class="co">#&gt; Astro_6  -22.427896 13.791919 -18.24245899  0.3619986  -3.1304498   8.108719</span>
<span class="co">#&gt; Astro_7  -27.774476 21.083890 -11.44540599  7.3578402  -7.3724091  -4.896353</span>
<span class="co">#&gt; Astro_8   -5.353543 41.562932   0.09699079  6.7034916  -7.7967877  12.112307</span>
<span class="co">#&gt; Astro_9   -9.075179 12.206922 -15.74899606  8.4253811  -0.9032461  11.838654</span>
<span class="co">#&gt; Astro_10 -17.221057  5.166105 -11.77236566  4.4967479 -23.7653079  -8.946312</span>
<span class="co">#&gt;                 PC7        PC8        PC9       PC10        PC11        PC12</span>
<span class="co">#&gt; Astro_1   -7.594958 -48.952807 29.6482347  -3.744406  7.92448941 -13.7973016</span>
<span class="co">#&gt; Astro_2   -1.853509  -8.448316 16.1858075  13.984738 25.45669970 -28.2914159</span>
<span class="co">#&gt; Astro_3  -15.216373  15.164415  8.9070569   4.619891 14.49909282  -8.1125923</span>
<span class="co">#&gt; Astro_4    9.765944   1.199330  6.2791143 -13.586999  0.04168374 -15.5893117</span>
<span class="co">#&gt; Astro_5   -2.338662   8.664926 32.6100345  19.821059 71.46057021 -32.6989156</span>
<span class="co">#&gt; Astro_6   -8.970303  -3.933349  2.5162345   2.250781  7.91776932 -43.6534070</span>
<span class="co">#&gt; Astro_7   -1.713211  -2.684308 -0.5325713  -9.860737 -1.49540831  -9.4517237</span>
<span class="co">#&gt; Astro_8  -17.913010  14.082511 -0.1689852   1.991545 22.22124199   0.8840939</span>
<span class="co">#&gt; Astro_9   10.387961  43.059185 35.1804789  15.970227 20.85669841 -66.3647118</span>
<span class="co">#&gt; Astro_10  14.316778  -4.014122 33.1067919 -16.180209 13.44523518 -30.3914637</span>
<span class="co">#&gt;                  PC13       PC14       PC15       PC16       PC17       PC18</span>
<span class="co">#&gt; Astro_1   42.13303590 -28.369023  13.503273 -29.231787  54.932172 -40.773997</span>
<span class="co">#&gt; Astro_2   -0.05960855   2.840941  -4.195210   5.416869 -17.168185 -18.126766</span>
<span class="co">#&gt; Astro_3   16.71457445   4.399337   6.980905  10.195392  15.493917  -6.914996</span>
<span class="co">#&gt; Astro_4   -6.01355953   5.039857 -17.033548  -7.806587   2.509568   2.168870</span>
<span class="co">#&gt; Astro_5   25.98607836  44.568138  -8.977355  14.684230  -7.130040  25.294624</span>
<span class="co">#&gt; Astro_6   26.95955966  -1.726479   5.588204 -14.894379   2.782293 -12.149518</span>
<span class="co">#&gt; Astro_7   19.25882785  -4.147610  -6.341807  -1.079372  14.414553 -23.373176</span>
<span class="co">#&gt; Astro_8   -8.63315684 -45.762486  -7.094981 -15.057877  -7.597148  -4.583326</span>
<span class="co">#&gt; Astro_9  -32.04120247 -28.247469  18.468230  33.618639 -19.104535 -39.654218</span>
<span class="co">#&gt; Astro_10  17.59909352 -24.983972 -11.173427   2.360632  21.532606 -25.457641</span>
<span class="co">#&gt;                PC19       PC20        PC21       PC22         PC23       PC24</span>
<span class="co">#&gt; Astro_1   34.765551  29.578458 -165.079427  71.372411 -14.89429804 -23.653269</span>
<span class="co">#&gt; Astro_2   -1.380197  -6.785663    4.129635 -11.093370   0.08486817   4.875333</span>
<span class="co">#&gt; Astro_3   33.108159  12.231927   -8.041653   8.695959  -0.43895922  -4.655246</span>
<span class="co">#&gt; Astro_4  -11.965406  13.646340    9.698028  -9.932636 -21.89754514 -12.073904</span>
<span class="co">#&gt; Astro_5  107.419525  18.892250  111.589833   4.485616 115.79325729 -91.335939</span>
<span class="co">#&gt; Astro_6  -20.069542  -1.654682    3.673820  48.546284  -2.87176456   5.482120</span>
<span class="co">#&gt; Astro_7   -9.776675  -4.761593   -5.679960  15.934803 -20.62667700  -3.920921</span>
<span class="co">#&gt; Astro_8    1.394895 -14.862392  -38.732064 -28.755109 -24.22492689  27.136466</span>
<span class="co">#&gt; Astro_9  -98.672480 -59.305569   30.470947 -64.714346  35.78702136  19.615467</span>
<span class="co">#&gt; Astro_10  -4.023878  -5.293881   -2.619474 -21.498159  -1.16358604  -3.243238</span>
<span class="co">#&gt;                PC25        PC26        PC27        PC28       PC29        PC30</span>
<span class="co">#&gt; Astro_1    3.837417  -48.835899  26.6527927 -155.999182  20.663068 -34.8613475</span>
<span class="co">#&gt; Astro_2   -7.071452    3.978576  -0.4681554   -6.067916  -8.885078  15.6817043</span>
<span class="co">#&gt; Astro_3    6.034805   12.515972 -28.3606001    7.524584  33.784116   6.6589079</span>
<span class="co">#&gt; Astro_4   -1.507713   11.309608  25.3871040   14.619842   5.869870 -10.3161214</span>
<span class="co">#&gt; Astro_5   38.001837   82.385978 -55.8727118   78.285114 -50.486500  19.2811822</span>
<span class="co">#&gt; Astro_6    5.039564   -3.372054  36.5937207  -11.722839  20.311115   0.8804704</span>
<span class="co">#&gt; Astro_7   32.260240    6.879909  28.0274277    6.451650  16.212719 -24.2186809</span>
<span class="co">#&gt; Astro_8   14.876740    1.322895 -38.6528971   26.227909  13.587061 -36.2119457</span>
<span class="co">#&gt; Astro_9    4.029583 -151.863731  -3.2405932   80.056143  -8.154873  -2.4377931</span>
<span class="co">#&gt; Astro_10 -41.549931   -4.446960  22.8930497    1.613867  10.258092  -2.3111014</span>
<span class="co">#&gt;                 PC31       PC32        PC33       PC34         PC35       PC36</span>
<span class="co">#&gt; Astro_1    9.2395486 -62.635136  12.1200153 -10.121555  62.92279283 -65.488539</span>
<span class="co">#&gt; Astro_2   -4.6423686   3.592134  17.6677754  12.105104  -3.92546280  -1.164524</span>
<span class="co">#&gt; Astro_3   12.4275261 -16.768447 -70.9232432  -6.535977   8.01493401  -7.661328</span>
<span class="co">#&gt; Astro_4  -22.6151289  17.039062  -3.2481276  15.326745  23.52664626  15.593609</span>
<span class="co">#&gt; Astro_5  -24.2251718   9.499109  -4.2218197 -38.179885  59.01565918  32.515445</span>
<span class="co">#&gt; Astro_6   -0.3113551  39.932408  10.7926362  -8.885007   8.30110282   6.677443</span>
<span class="co">#&gt; Astro_7  -31.1474265   2.795211 -26.3408942   1.556208 -52.42576566  -3.434696</span>
<span class="co">#&gt; Astro_8   14.0836468   5.771078  55.2595294 -25.409898  61.53443052  26.123637</span>
<span class="co">#&gt; Astro_9   52.2019048  -7.637311  -0.4463137  -1.277904 -69.46954557  -9.235945</span>
<span class="co">#&gt; Astro_10 -36.9734959  39.061988 -31.3483889  19.751079  -0.07246345  12.773080</span>
<span class="co">#&gt;                PC37       PC38       PC39       PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1   12.869634  -14.94331  54.251228  -6.917953  -6.666643   1.403420</span>
<span class="co">#&gt; Astro_2    8.764531   14.88371   9.345336  -1.482249  -9.525483  26.986607</span>
<span class="co">#&gt; Astro_3  -14.451378   47.59206  -2.338277 -29.630470  11.267376   2.685711</span>
<span class="co">#&gt; Astro_4  -26.612622  -37.56086  39.886870   2.409798 -46.591720 -19.269073</span>
<span class="co">#&gt; Astro_5   -9.499134  -34.05019  23.301035 -40.985219  72.645749   8.400509</span>
<span class="co">#&gt; Astro_6  -16.405769   36.92129  43.774285  37.191720  -2.655285  30.821027</span>
<span class="co">#&gt; Astro_7  -18.301087   29.38353 -36.069457   6.936953  31.350557 -42.441836</span>
<span class="co">#&gt; Astro_8   29.283570 -114.63814   9.410252 -13.164703 -10.809752 -28.123344</span>
<span class="co">#&gt; Astro_9    6.302795   71.02910 -69.293763  28.552398 -13.318218  18.726829</span>
<span class="co">#&gt; Astro_10  11.852726  -25.56234 -50.245964 -67.186048 -25.632001  22.626312</span>
<span class="co">#&gt;                PC43       PC44       PC45        PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -1.379679 -10.167157  26.783558   4.8844893 -16.604971  24.952965</span>
<span class="co">#&gt; Astro_2   34.242398   2.016852 -14.753445 -23.7456940  -9.391538 -18.802521</span>
<span class="co">#&gt; Astro_3   -2.955026  38.688319  33.931751  17.3080243  50.757099  75.341107</span>
<span class="co">#&gt; Astro_4   26.343542   2.114936   9.774017  56.9896720  39.092636  -5.464513</span>
<span class="co">#&gt; Astro_5  -13.062010   6.520175 -18.703894 -13.1328473 -41.476398   2.202484</span>
<span class="co">#&gt; Astro_6  -31.414367   1.987061 -53.512493   0.1023383 -19.947890  11.481558</span>
<span class="co">#&gt; Astro_7   -9.782255 -14.282588  10.054903 -67.9604108  -9.487772  12.949448</span>
<span class="co">#&gt; Astro_8   41.278687  37.593404 -54.952740  22.6392559  44.107209 -39.270917</span>
<span class="co">#&gt; Astro_9  -14.434996  -9.991141  25.248281  25.3177202  43.610909 -49.231889</span>
<span class="co">#&gt; Astro_10  33.252830  11.521956  12.918566 -11.6908434 -28.285087 -21.765664</span>
<span class="co">#&gt;                PC49       PC50</span>
<span class="co">#&gt; Astro_1   26.146699 -18.417859</span>
<span class="co">#&gt; Astro_2  -20.592183  12.895531</span>
<span class="co">#&gt; Astro_3   -4.945454  -6.459438</span>
<span class="co">#&gt; Astro_4   34.928297  32.080658</span>
<span class="co">#&gt; Astro_5  -19.585681   4.616851</span>
<span class="co">#&gt; Astro_6   -7.074786  -4.430888</span>
<span class="co">#&gt; Astro_7   -2.217498  31.089373</span>
<span class="co">#&gt; Astro_8   28.013394  30.859145</span>
<span class="co">#&gt; Astro_9   35.194498  -8.639278</span>
<span class="co">#&gt; Astro_10 -57.301790 -77.378248</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="1248"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="1248"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
