<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<div class="figure">
<img src="../figures/bandnorm_intro.png" alt=""><p class="caption">The introduction for BandNorm</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony 10/10</span>
<span class="co">#&gt; Harmony converged after 10 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                  PC1       PC2         PC3        PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1  -19.7082464 28.811616 -23.8949498  35.150585 -34.167979 -16.093015</span>
<span class="co">#&gt; Astro_2   -7.3267574 46.272562   4.4755768  16.842213 -13.951679  -2.484006</span>
<span class="co">#&gt; Astro_3   -0.3239463  8.682601  -1.2777052   4.923611  11.560447 -10.780792</span>
<span class="co">#&gt; Astro_4   -8.7088714 26.736962 -26.4223117  11.835827  -9.729075 -15.327933</span>
<span class="co">#&gt; Astro_5   11.5349369 17.157246 -20.4018298  14.351155 -26.069305 -25.380617</span>
<span class="co">#&gt; Astro_6  -21.3484236 13.865803 -19.3961148  -0.153196  -3.861859   9.290331</span>
<span class="co">#&gt; Astro_7    6.6332584 47.349138 -71.6731422 -65.113668  -8.434199  -9.947602</span>
<span class="co">#&gt; Astro_8   -1.0163779 44.994779  -0.9627146  11.857249 -10.516306  12.779721</span>
<span class="co">#&gt; Astro_9   -4.6578099 24.170456 -16.3786659  19.835603 -18.212833  -9.650617</span>
<span class="co">#&gt; Astro_10 -16.8051802  4.346302 -12.5563555   3.848788 -24.587289  -8.691638</span>
<span class="co">#&gt;                 PC7        PC8        PC9        PC10       PC11       PC12</span>
<span class="co">#&gt; Astro_1    3.971630 -45.824813  35.491455  -4.9164623 25.6828846 -26.705808</span>
<span class="co">#&gt; Astro_2  -35.064483 -39.457115 -16.017228  -5.7350189 19.1075119 -34.027850</span>
<span class="co">#&gt; Astro_3  -16.555027  24.815195  -3.795087   6.0263446  2.0859114  -7.651319</span>
<span class="co">#&gt; Astro_4   11.243274   6.373361   1.860960 -12.2390168 -2.6482597 -16.277114</span>
<span class="co">#&gt; Astro_5   -9.038704 -12.613895  26.538324   5.9862251 92.8851315 -48.960787</span>
<span class="co">#&gt; Astro_6   -8.845734  -4.565472   2.119168   2.4071755  8.5886062 -45.276414</span>
<span class="co">#&gt; Astro_7   -2.601860  -5.739887  -1.728622  -5.6909678 -0.1082682 -14.329732</span>
<span class="co">#&gt; Astro_8  -20.371635  13.336647  -3.368872  -0.7644421 26.3609992  -1.502431</span>
<span class="co">#&gt; Astro_9  -17.140215  -4.368754  14.319022 -11.4978225 45.1122917 -86.255199</span>
<span class="co">#&gt; Astro_10  14.826843  -4.608527  33.284809 -15.6667030 14.4296093 -31.341040</span>
<span class="co">#&gt;                PC13         PC14        PC15       PC16        PC17        PC18</span>
<span class="co">#&gt; Astro_1   33.267260 -25.17568128  19.4643960 -19.799950  44.0249134 -47.0121746</span>
<span class="co">#&gt; Astro_2    0.933620   1.59852290 -13.0564159  -2.688189   6.2808227  -3.4470677</span>
<span class="co">#&gt; Astro_3   11.216990   0.03414993  -6.0808427   9.230157   8.8755025  -0.6561208</span>
<span class="co">#&gt; Astro_4   -8.627166   4.95422553 -22.4326127  -4.734458  -2.7430644   3.0204413</span>
<span class="co">#&gt; Astro_5   21.635421  48.97062554  -0.3859812  19.714740   0.7697081  24.0448713</span>
<span class="co">#&gt; Astro_6   29.148818  -1.55913277   5.3884866 -16.337634   2.7421274 -12.9429822</span>
<span class="co">#&gt; Astro_7   19.872486 -16.58236610 -20.5080704   3.879476 -16.6457480 -17.5526323</span>
<span class="co">#&gt; Astro_8   -3.787355 -47.98449497  -4.2642034 -14.888026  -6.4390501   3.4524723</span>
<span class="co">#&gt; Astro_9  -31.842175 -22.70490753  28.8000539  33.451974   9.3860604 -34.5335164</span>
<span class="co">#&gt; Astro_10  19.293212 -24.95818732 -10.8435122   1.124084  21.6804394 -25.6096973</span>
<span class="co">#&gt;                PC19       PC20        PC21       PC22       PC23       PC24</span>
<span class="co">#&gt; Astro_1   25.569480  20.680785 -154.874315  58.762475   5.682463 -28.422498</span>
<span class="co">#&gt; Astro_2    7.291372   4.825205   -7.973925  -0.998890  -9.769785  34.216144</span>
<span class="co">#&gt; Astro_3   18.965097   6.531660   10.037759  -2.129291 -16.481520  15.518390</span>
<span class="co">#&gt; Astro_4  -15.883141  13.301597   18.833884 -15.862836 -25.588426  -6.900209</span>
<span class="co">#&gt; Astro_5  112.596791  20.670186  103.386677   5.009961 141.164295 -92.156122</span>
<span class="co">#&gt; Astro_6  -21.535031  -3.230735    3.795656  49.127652  -6.068639   9.357072</span>
<span class="co">#&gt; Astro_7   -1.443606  13.896255  -25.314012  15.063107  -8.786138   2.371008</span>
<span class="co">#&gt; Astro_8    5.967188 -19.764046  -42.337378 -27.961891 -32.674232  34.287198</span>
<span class="co">#&gt; Astro_9  -78.827472 -46.470264    2.896760 -50.534104  64.648074  25.632585</span>
<span class="co">#&gt; Astro_10  -5.149690  -7.271300   -2.123971 -19.580544  -2.290375   0.176057</span>
<span class="co">#&gt;                 PC25        PC26         PC27         PC28       PC29</span>
<span class="co">#&gt; Astro_1   12.1935111  -79.343250  18.59919747 -135.9605035  17.238159</span>
<span class="co">#&gt; Astro_2  -11.9290241   26.261261  -0.05216965   -9.3614280   1.157036</span>
<span class="co">#&gt; Astro_3    4.3691953   19.747991 -29.12762396   26.9947170  43.825815</span>
<span class="co">#&gt; Astro_4    0.8225943    9.486687  23.09186494   25.2270435   8.684373</span>
<span class="co">#&gt; Astro_5   45.3509199   60.317126 -63.58004701   84.7414570 -54.101696</span>
<span class="co">#&gt; Astro_6    5.8265579   -2.823348  33.93480500  -13.3936443  24.188855</span>
<span class="co">#&gt; Astro_7   34.3245959    4.499158  21.00952192    9.7421743   8.629582</span>
<span class="co">#&gt; Astro_8   14.1380930   -2.104559 -37.84794820   24.8639094  12.425118</span>
<span class="co">#&gt; Astro_9    9.3413389 -163.316411 -10.63670051   71.9553682 -11.118146</span>
<span class="co">#&gt; Astro_10 -40.9998584   -4.359693  20.58503209   -0.9146004  12.996311</span>
<span class="co">#&gt;                PC30       PC31       PC32       PC33       PC34        PC35</span>
<span class="co">#&gt; Astro_1  -28.364969   2.630491 -60.837972  22.722263 -25.021677  56.6723521</span>
<span class="co">#&gt; Astro_2   16.446365  -6.876204   1.341228  10.067612  14.936994  -7.5624446</span>
<span class="co">#&gt; Astro_3   14.317662  -2.799851  -4.811118 -71.345234  -3.606042 -11.2324967</span>
<span class="co">#&gt; Astro_4   -6.012161 -27.119416  16.934662  -6.771020  11.673170   8.4588643</span>
<span class="co">#&gt; Astro_5   20.117123 -22.021135   1.627635   2.515616 -53.830017  62.8973717</span>
<span class="co">#&gt; Astro_6    1.596179  -4.174825  43.462861  11.241279  -8.308725   8.6126161</span>
<span class="co">#&gt; Astro_7  -40.262873 -42.746742  -5.571561 -46.634139  20.029894 -50.6331918</span>
<span class="co">#&gt; Astro_8  -39.446797  19.105656   4.413553  49.473658 -24.435097  60.3847720</span>
<span class="co">#&gt; Astro_9   -6.486986  62.271737 -23.502344   1.848401 -16.519224 -55.9514609</span>
<span class="co">#&gt; Astro_10  -1.180698 -39.810292  41.042705 -32.305885  19.570680   0.1739368</span>
<span class="co">#&gt;                 PC36       PC37       PC38       PC39       PC40       PC41</span>
<span class="co">#&gt; Astro_1  -74.1424197  14.923780  -17.69893  50.095454  -6.247827  -4.736382</span>
<span class="co">#&gt; Astro_2   13.1454219   3.984598   19.26058  12.311948  -1.830040 -11.536834</span>
<span class="co">#&gt; Astro_3    0.6412143 -17.820441   55.59081  -5.967786 -25.415644   8.214722</span>
<span class="co">#&gt; Astro_4   14.9913135 -28.381364  -29.57139  35.912263   4.680885 -44.278178</span>
<span class="co">#&gt; Astro_5   27.4239359  -8.475901  -38.67301  22.964764 -43.573559  75.059885</span>
<span class="co">#&gt; Astro_6    8.7730816 -16.719097   38.66338  39.939968  37.769685  -3.573403</span>
<span class="co">#&gt; Astro_7    4.7352076 -10.671704   37.09157 -32.083872   3.914934  22.502308</span>
<span class="co">#&gt; Astro_8   30.4848800  33.005323 -117.29667  12.649075 -15.670954   2.116625</span>
<span class="co">#&gt; Astro_9   -9.5644615   5.636209   64.13661 -65.415749  22.857269 -10.927915</span>
<span class="co">#&gt; Astro_10  12.2812031  11.442395  -23.60050 -54.243049 -67.187197 -26.279287</span>
<span class="co">#&gt;                PC42        PC43       PC44       PC45        PC46       PC47</span>
<span class="co">#&gt; Astro_1    5.679983  -3.6159279 -18.873884  31.027150  -0.3648084 -28.501443</span>
<span class="co">#&gt; Astro_2   21.751747  35.1477764   1.064156 -12.516591 -26.1315392  -5.250772</span>
<span class="co">#&gt; Astro_3    3.292416   0.7089085  32.031181  35.717836   7.0587053  50.221097</span>
<span class="co">#&gt; Astro_4  -18.303450  30.4140343  -4.819627  12.679055  45.6234534  44.376601</span>
<span class="co">#&gt; Astro_5    9.317038 -18.1368048   1.956061 -14.056205 -12.9302934 -51.293981</span>
<span class="co">#&gt; Astro_6   29.758175 -33.7533853   4.429936 -52.924096  -2.2852806 -26.187753</span>
<span class="co">#&gt; Astro_7  -35.792229 -10.4774835  -8.064994  -1.171811 -65.6235758  -5.489606</span>
<span class="co">#&gt; Astro_8  -26.454784  50.0747148  28.412574 -55.906298  25.5403307  42.912852</span>
<span class="co">#&gt; Astro_9   15.832520 -21.7671366 -10.613418  30.016549  31.0852055  36.249807</span>
<span class="co">#&gt; Astro_10  20.944158  31.9308996  12.987480  13.190013 -14.8923618 -32.437163</span>
<span class="co">#&gt;                PC48       PC49        PC50</span>
<span class="co">#&gt; Astro_1    7.531487  31.487318 -24.3067588</span>
<span class="co">#&gt; Astro_2   -4.489506 -25.912060  19.9586798</span>
<span class="co">#&gt; Astro_3   74.976222 -10.861318   0.3823273</span>
<span class="co">#&gt; Astro_4   -5.838995  35.383907  40.2005409</span>
<span class="co">#&gt; Astro_5   -5.210229 -12.862788  -1.3742454</span>
<span class="co">#&gt; Astro_6   13.801770 -10.776746  -9.1290176</span>
<span class="co">#&gt; Astro_7   15.886520   1.530984  28.8335352</span>
<span class="co">#&gt; Astro_8  -40.203908  23.215867  32.1765928</span>
<span class="co">#&gt; Astro_9  -46.718874  42.224517 -14.4806991</span>
<span class="co">#&gt; Astro_10 -19.004853 -60.122392 -80.1495754</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
