<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD_tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html" class="external-link">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="format-of-input">Format of input<a class="anchor" aria-label="anchor" href="#format-of-input"></a>
</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: furrr</span>
<span class="co">#&gt; Warning: package 'furrr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: future</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'rmarkdown'</span>
<span class="co">#&gt; The following object is masked from 'package:future':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     run</span>
<span class="co">#&gt; Loading required package: viridis</span>
<span class="co">#&gt; Warning: package 'viridis' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridisLite</span>
<span class="co">#&gt; Warning: package 'viridisLite' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: sparsesvd</span>
<span class="co">#&gt; Warning: package 'sparsesvd' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: GenomicInteractions</span>
<span class="co">#&gt; Loading required package: InteractionSet</span>
<span class="co">#&gt; Warning: package 'InteractionSet' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Warning: package 'GenomicRanges' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Warning: package 'BiocGenerics' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine, intersect, setdiff, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Warning: package 'S4Vectors' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:future':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     values</span>
<span class="co">#&gt; The following object is masked from 'package:Matrix':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand</span>
<span class="co">#&gt; The following objects are masked from 'package:data.table':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, second</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, rename</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Warning: package 'IRanges' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'IRanges'</span>
<span class="co">#&gt; The following object is masked from 'package:data.table':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     shift</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse, desc, slice</span>
<span class="co">#&gt; The following object is masked from 'package:grDevices':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     windows</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Warning: package 'GenomeInfoDb' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: SummarizedExperiment</span>
<span class="co">#&gt; Warning: package 'SummarizedExperiment' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: MatrixGenerics</span>
<span class="co">#&gt; Warning: package 'MatrixGenerics' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'MatrixGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Warning: package 'Biobase' was built under R version 4.0.3</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biobase'</span>
<span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     rowMedians</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'SummarizedExperiment'</span>
<span class="co">#&gt; The following object is masked from 'package:SeuratObject':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Assays</span>
<span class="co">#&gt; The following object is masked from 'package:Seurat':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Assays</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="download-existing-single-cell-hi-c-data">Download existing single-cell Hi-C data<a class="anchor" aria-label="anchor" href="#download-existing-single-cell-hi-c-data"></a>
</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173" class="external-link">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z" class="external-link">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/" class="external-link">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z" class="external-link">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/" class="external-link">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># download_schic("Li2019", cell_type = "2i", cell_path = getwd(), summary_path = getwd())</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="use-bandnorm">Use BandNorm<a class="anchor" aria-label="anchor" href="#use-bandnorm"></a>
</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z" class="external-link">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span>, chrs <span class="op">=</span> <span class="st">"chr1"</span><span class="op">)</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony converged after 3 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;              [,1]       [,2]      [,3]         [,4]      [,5]       [,6]</span>
<span class="co">#&gt; Astro_1  260.4890   8.767428 -38.15255 -24.67058474 41.438432  42.359819</span>
<span class="co">#&gt; Astro_2  257.7442  -2.951022 -45.12881   0.09234945 16.898551  12.308446</span>
<span class="co">#&gt; Astro_3  263.3217  -9.622970 -15.05707  -1.82768494  9.421532 -10.324896</span>
<span class="co">#&gt; Astro_4  257.4937  -3.686170 -38.18991 -30.22117139 15.970829  14.971427</span>
<span class="co">#&gt; Astro_5  253.8995 -21.326499 -19.08085 -23.88545629 17.700467  28.380758</span>
<span class="co">#&gt; Astro_6  255.1843   7.330198 -28.17633 -25.69458359  2.444970   7.701229</span>
<span class="co">#&gt; Astro_7  263.1543  15.709052 -33.01460 -12.05113123  9.612310   5.296989</span>
<span class="co">#&gt; Astro_8  261.5342  -5.386315 -46.78872  -4.27357902  9.024755   5.942738</span>
<span class="co">#&gt; Astro_9  258.5931  -5.285896 -20.20657 -20.84444116 19.050879  14.885779</span>
<span class="co">#&gt; Astro_10 262.5704   1.565764 -19.03290 -19.23823636  6.431833  28.061190</span>
<span class="co">#&gt;                [,7]       [,8]        [,9]      [,10]        [,11]      [,12]</span>
<span class="co">#&gt; Astro_1   27.375286   2.595551 -56.6927552 42.0278246  -6.64794030 -30.997075</span>
<span class="co">#&gt; Astro_2   -7.319173   9.577958  -7.8033543 11.7208199  14.06890106 -17.897096</span>
<span class="co">#&gt; Astro_3   21.925646  23.086416  15.7495192  4.5084385   4.84878002  -7.029905</span>
<span class="co">#&gt; Astro_4   19.614074  -3.310085   0.4682037  4.1700977 -14.10580821   3.919195</span>
<span class="co">#&gt; Astro_5   25.013790  -2.860101   1.1129512 44.3856683  16.67119525 -94.647483</span>
<span class="co">#&gt; Astro_6  -11.077577  17.896088  -6.4499603  0.2569391   0.02233005  -1.287126</span>
<span class="co">#&gt; Astro_7    6.238411   9.604235  -2.1418579 -4.8596355  -9.75191614   9.259222</span>
<span class="co">#&gt; Astro_8  -12.830410  24.374966  13.1372461 -2.2591202   1.57709507 -21.180022</span>
<span class="co">#&gt; Astro_9   -2.931585 -15.193808  35.2331178 47.2257586  12.91352783 -44.113965</span>
<span class="co">#&gt; Astro_10   6.369004  -5.737758  -6.0984339 31.5061841 -17.83135495  -7.297526</span>
<span class="co">#&gt;               [,13]      [,14]      [,15]      [,16]      [,17]      [,18]</span>
<span class="co">#&gt; Astro_1  24.8389123  48.014163  22.594232  26.879686 -17.764963  47.575686</span>
<span class="co">#&gt; Astro_2  26.7372741   7.112525  -2.100140  -9.758681   9.129671 -22.321543</span>
<span class="co">#&gt; Astro_3   6.3331554  23.932153  -3.554945   1.070171  13.202461  10.333723</span>
<span class="co">#&gt; Astro_4  15.7940490   1.013382  -4.964899 -19.919792  -4.115079  -2.493478</span>
<span class="co">#&gt; Astro_5  44.1986970  31.957885 -50.658311   5.041710  28.953036 -11.821712</span>
<span class="co">#&gt; Astro_6  46.1472793  35.984804   4.290632   3.137027 -18.789369  -1.698368</span>
<span class="co">#&gt; Astro_7   7.4893642  26.285780   4.960351 -12.157652   1.855262   9.015391</span>
<span class="co">#&gt; Astro_8   0.2081309  -3.852817  39.317673  -8.172145 -20.812369 -10.652579</span>
<span class="co">#&gt; Astro_9  77.9487476 -25.929523  22.183102  32.192150  47.480473 -24.668663</span>
<span class="co">#&gt; Astro_10 32.6119878  26.130073  27.947224 -13.650015  -2.480721  16.657205</span>
<span class="co">#&gt;               [,19]         [,20]       [,21]        [,22]      [,23]</span>
<span class="co">#&gt; Astro_1  -52.165887  -44.77027195 -33.5171701  174.2001968  72.226768</span>
<span class="co">#&gt; Astro_2  -17.111597    0.03440498   2.4519609   -6.7115828  -8.946803</span>
<span class="co">#&gt; Astro_3   -6.104868  -34.79714807 -16.9931502    5.4223607  10.766928</span>
<span class="co">#&gt; Astro_4    1.602215   10.69226879 -16.4557316  -10.8021012  -7.617040</span>
<span class="co">#&gt; Astro_5   13.968589 -117.71146709 -21.4002871 -102.4880111   8.550867</span>
<span class="co">#&gt; Astro_6  -11.901826   27.62252463   7.5348045   -3.4529971  50.485852</span>
<span class="co">#&gt; Astro_7  -22.520698    8.27432244   0.3743479    3.3861833  17.936365</span>
<span class="co">#&gt; Astro_8   -6.182087    6.40240844  20.1300829   39.9619402 -28.535149</span>
<span class="co">#&gt; Astro_9  -50.749904   88.50609366  55.8169814  -22.5412733 -61.490664</span>
<span class="co">#&gt; Astro_10 -24.548488   11.99957044  11.7413168    0.8271987 -18.653642</span>
<span class="co">#&gt;                [,24]       [,25]      [,26]      [,27]     [,28]       [,29]</span>
<span class="co">#&gt; Astro_1   -16.386022  -54.869325  20.560727  81.544077 -13.76737 -153.072692</span>
<span class="co">#&gt; Astro_2     7.268235    1.832398  -2.904150  -7.814636   5.43311   -1.038148</span>
<span class="co">#&gt; Astro_3     7.664850   -8.364236  10.543454 -16.986662  33.25000   12.957056</span>
<span class="co">#&gt; Astro_4    25.398605  -17.013670   3.492867 -11.492420 -19.42296   18.991759</span>
<span class="co">#&gt; Astro_5  -147.772999 -122.943107  54.735317 -50.354511  69.00397   81.567382</span>
<span class="co">#&gt; Astro_6    15.977662   12.425930   4.278786  -2.759817 -31.24299  -10.258419</span>
<span class="co">#&gt; Astro_7    28.127373   -6.674850  36.518524 -11.394666 -23.13568   11.706313</span>
<span class="co">#&gt; Astro_8    28.649519   35.369948  14.517941  -1.017830  40.95661   25.122970</span>
<span class="co">#&gt; Astro_9   -67.940815  -13.220251  21.377759 186.058095  16.48910   82.722282</span>
<span class="co">#&gt; Astro_10   12.763440    3.186854 -42.558798  -1.472243 -17.96547    2.968344</span>
<span class="co">#&gt;               [,30]      [,31]       [,32]      [,33]      [,34]       [,35]</span>
<span class="co">#&gt; Astro_1    9.590189  42.816171  25.5879791  82.152135 -17.152270  26.9323154</span>
<span class="co">#&gt; Astro_2   -6.171868 -11.269152  -0.7524294   2.601524 -12.055115 -12.9924918</span>
<span class="co">#&gt; Astro_3   35.840878  -2.266545  17.6956974  21.980148  76.642959   7.4629514</span>
<span class="co">#&gt; Astro_4    7.619683  15.392685 -18.2376936 -12.042492   4.811808 -15.5632095</span>
<span class="co">#&gt; Astro_5  -63.368410 -10.478491  -6.9576844   9.294764  -5.968970  54.9452854</span>
<span class="co">#&gt; Astro_6   30.374323   3.745559  -7.5131172 -52.411802 -19.763387  -2.1776394</span>
<span class="co">#&gt; Astro_7   18.674288  28.937264 -26.4556766   3.266864  34.270544  -0.3702734</span>
<span class="co">#&gt; Astro_8   13.131849  38.719632  16.2648726  -5.324021 -56.595264  18.6355156</span>
<span class="co">#&gt; Astro_9  -19.637507  10.015595  68.1805446  27.492296  -1.892617  21.1763405</span>
<span class="co">#&gt; Astro_10  19.153321   6.796307 -43.1824271 -51.727511  23.113677 -29.6471542</span>
<span class="co">#&gt;               [,36]       [,37]      [,38]      [,39]      [,40]       [,41]</span>
<span class="co">#&gt; Astro_1   73.577289 -76.6473786 -10.249177  -11.69798 -53.204571   9.6541863</span>
<span class="co">#&gt; Astro_2   -9.541354   3.6128059  -7.283889   28.27541  -8.976800  -0.5436889</span>
<span class="co">#&gt; Astro_3    5.760162   0.9157821  15.045677   61.58504   2.394746  27.5026545</span>
<span class="co">#&gt; Astro_4   22.436398  22.8350392  25.681786  -27.13307 -38.641330  -3.7815305</span>
<span class="co">#&gt; Astro_5   68.409064  21.0722508   4.983493  -31.11380 -20.931421  43.3814474</span>
<span class="co">#&gt; Astro_6   21.957560  24.7474594  12.136265   38.14670 -43.547662 -39.2408240</span>
<span class="co">#&gt; Astro_7  -56.278708   3.4877407  20.064065   44.00578  36.069266  -9.1911690</span>
<span class="co">#&gt; Astro_8   62.941845  30.2315027 -36.536162 -119.47765 -11.550092  15.4002350</span>
<span class="co">#&gt; Astro_9  -59.620860 -22.9745626  -6.709363   76.00778  69.084106 -25.9574979</span>
<span class="co">#&gt; Astro_10  13.663266  24.7698930 -14.476287  -23.14196  52.038852  65.3056339</span>
<span class="co">#&gt;                [,42]      [,43]      [,44]      [,45]     [,46]      [,47]</span>
<span class="co">#&gt; Astro_1    1.4042453  -8.010776  -9.936818  10.086427  34.95841  -6.878220</span>
<span class="co">#&gt; Astro_2    9.9176614 -30.112688  33.152526  -2.607484 -10.06566  26.102611</span>
<span class="co">#&gt; Astro_3  -11.2904710  -5.961986  -3.732932 -39.483373  39.32086 -13.379015</span>
<span class="co">#&gt; Astro_4   47.6176604  15.884958  23.432234  -3.734371  16.84145 -56.400890</span>
<span class="co">#&gt; Astro_5  -77.8378575 -15.213299 -21.548621  -6.503176 -11.15840   9.462415</span>
<span class="co">#&gt; Astro_6   12.6594503 -32.945805 -41.975947 -14.210205 -51.50516 -13.824187</span>
<span class="co">#&gt; Astro_7  -30.9920359  39.241872 -11.385210  13.075889  13.05906  71.223717</span>
<span class="co">#&gt; Astro_8    0.7003749  24.847937  50.955284 -29.392734 -54.54864 -35.502715</span>
<span class="co">#&gt; Astro_9    8.4094760 -25.572851 -22.823801  10.015358  34.10033 -27.521115</span>
<span class="co">#&gt; Astro_10  35.8006038 -23.976404  23.721624 -22.409581  14.47883   1.179911</span>
<span class="co">#&gt;               [,48]      [,49]       [,50]</span>
<span class="co">#&gt; Astro_1   27.144996 -19.977632  41.7302522</span>
<span class="co">#&gt; Astro_2    9.757486   9.627174 -18.2998709</span>
<span class="co">#&gt; Astro_3  -51.163939 -84.691785  -2.2715623</span>
<span class="co">#&gt; Astro_4  -34.260223  -1.579583  37.4665829</span>
<span class="co">#&gt; Astro_5   52.559095   2.943192  -3.6451595</span>
<span class="co">#&gt; Astro_6   48.168772 -17.226910 -15.7795729</span>
<span class="co">#&gt; Astro_7    9.193921 -22.290690   0.2038873</span>
<span class="co">#&gt; Astro_8  -33.839609  38.958284  21.1158886</span>
<span class="co">#&gt; Astro_9  -33.060039  54.412258  51.1445118</span>
<span class="co">#&gt; Astro_10  54.534217  15.884740 -65.5443043</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
