<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD_tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; Warning: package 'rmarkdown' was built under R version 4.0.5</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony 10/10</span>
<span class="co">#&gt; Harmony converged after 10 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2        PC3       PC4        PC5         PC6</span>
<span class="co">#&gt; Astro_1  -18.598110 29.931080 -24.382399 31.116336 -40.118579 -21.7019210</span>
<span class="co">#&gt; Astro_2  -13.670918 36.083452  -3.312267  6.930737  -9.009162   6.3464861</span>
<span class="co">#&gt; Astro_3   -2.267128  4.597544   2.897464  3.377565  12.693628 -15.4582799</span>
<span class="co">#&gt; Astro_4   -7.187638 28.130342 -25.996973  9.657781 -12.472984 -14.0028934</span>
<span class="co">#&gt; Astro_5    1.628802  8.360079 -23.223005 13.253111 -24.817775 -26.7349927</span>
<span class="co">#&gt; Astro_6  -20.061033 14.267743 -20.536668  1.234884  -4.438121   7.3750693</span>
<span class="co">#&gt; Astro_7  -26.900891 23.071582  -8.286288  2.590130  -2.842186   0.2114248</span>
<span class="co">#&gt; Astro_8   -1.107634 43.001823  -4.118322 -2.020583  -2.913827  14.4833077</span>
<span class="co">#&gt; Astro_9  -10.996912 12.592310 -26.308491  7.604737 -12.041808   1.4774103</span>
<span class="co">#&gt; Astro_10 -15.471287  4.827477 -13.715952  5.268014 -25.010593 -10.1452987</span>
<span class="co">#&gt;                 PC7        PC8        PC9        PC10      PC11       PC12</span>
<span class="co">#&gt; Astro_1    4.460227 -52.100513 40.2474960  -6.5515215 37.317000 -27.162293</span>
<span class="co">#&gt; Astro_2   -7.407541  -8.174406  7.0753176  12.1423323 11.933554 -18.978569</span>
<span class="co">#&gt; Astro_3  -15.356983  20.370090  2.8735243   5.2442983 13.856275  -8.737645</span>
<span class="co">#&gt; Astro_4   10.894640   4.463943  2.8272319 -13.7644010  2.907668 -17.859815</span>
<span class="co">#&gt; Astro_5    2.952286  -1.792666 39.5549378  14.3352078 84.920551 -32.838864</span>
<span class="co">#&gt; Astro_6   -8.898990  -4.130747  1.3905495   0.8574634  8.598461 -44.111732</span>
<span class="co">#&gt; Astro_7   -1.840967   2.606614 -6.5318506  -9.3327071 -2.039107 -10.357310</span>
<span class="co">#&gt; Astro_8  -17.574318  11.489550 -0.2931915   0.5600867 24.944239   1.166866</span>
<span class="co">#&gt; Astro_9   16.646614  34.493576 42.5938448  10.7620046 36.837644 -69.246289</span>
<span class="co">#&gt; Astro_10  14.756432  -3.866637 32.3606157 -17.1826607 14.062444 -30.227001</span>
<span class="co">#&gt;                PC13        PC14       PC15        PC16        PC17       PC18</span>
<span class="co">#&gt; Astro_1   35.475167 -17.8441865  31.093670 -14.5614458  46.8598862 -49.489388</span>
<span class="co">#&gt; Astro_2    3.732014   0.5568262  -6.336249   0.8622607 -10.2712421 -15.806322</span>
<span class="co">#&gt; Astro_3   10.502538   9.1290884   5.634217  15.2370060   6.6189743  -3.797738</span>
<span class="co">#&gt; Astro_4  -11.257326   9.9240360 -15.740841  -2.8328200  -5.5238469   3.630193</span>
<span class="co">#&gt; Astro_5   30.232807  47.9757347   8.794098  17.5248834   1.3960024  15.270531</span>
<span class="co">#&gt; Astro_6   26.243766   0.3979917   6.999083 -15.5129058  -0.1274073 -11.690016</span>
<span class="co">#&gt; Astro_7   13.108140   0.4956446  -7.808197   3.9769618   5.7202350 -20.204446</span>
<span class="co">#&gt; Astro_8   -4.015004 -43.1880686  -5.824666 -21.9046330  -8.0297394  -3.715574</span>
<span class="co">#&gt; Astro_9  -28.286495 -25.0738512  35.459946  38.5349062 -11.7070867 -49.076144</span>
<span class="co">#&gt; Astro_10  16.175552 -23.1416778  -9.513600   1.7511113  18.4756010 -24.237886</span>
<span class="co">#&gt;                 PC19       PC20        PC21       PC22       PC23       PC24</span>
<span class="co">#&gt; Astro_1   39.4165478  24.680260 -158.567288  61.766255  17.998826 -43.281226</span>
<span class="co">#&gt; Astro_2    0.9824895   3.274970  -10.570522  -6.868671 -23.768942  30.192397</span>
<span class="co">#&gt; Astro_3   28.0204018   6.850144   13.156922  -3.509714  -5.139016   2.895744</span>
<span class="co">#&gt; Astro_4  -16.0350655   8.270592   27.079930 -20.272976 -22.422512  -7.440697</span>
<span class="co">#&gt; Astro_5  118.7523443  30.320110   80.174146  12.452443 126.448437 -88.869054</span>
<span class="co">#&gt; Astro_6  -17.3645302  -1.256421    4.016735  47.114303  -1.410502   6.991088</span>
<span class="co">#&gt; Astro_7  -14.8838890 -10.230062   15.273690   3.798401 -25.429257   3.683419</span>
<span class="co">#&gt; Astro_8   -2.7854051 -17.413308  -37.636104 -23.203178 -29.014660  33.578798</span>
<span class="co">#&gt; Astro_9  -86.8926044 -48.965071    1.486148 -58.063612  48.838538  18.450623</span>
<span class="co">#&gt; Astro_10  -1.8175842  -5.881182   -1.216682 -22.072054   1.690142  -1.315146</span>
<span class="co">#&gt;                PC25        PC26       PC27        PC28       PC29       PC30</span>
<span class="co">#&gt; Astro_1   13.275856  -75.573154  13.570530 -140.979724   5.727072 -33.404555</span>
<span class="co">#&gt; Astro_2  -13.810191   21.399419   3.177459  -12.656734   2.592716  15.532955</span>
<span class="co">#&gt; Astro_3    3.443207   24.605286 -33.378943   26.871663  30.746832  12.667440</span>
<span class="co">#&gt; Astro_4   -3.009014   18.087493  20.170282   31.347615   2.768800  -5.696529</span>
<span class="co">#&gt; Astro_5   40.311243   65.800138 -57.668802   64.648362 -52.043389  13.761692</span>
<span class="co">#&gt; Astro_6    5.102028   -2.080999  37.222388   -7.593822  21.344240   1.416606</span>
<span class="co">#&gt; Astro_7   29.780438   18.722245  22.822204   25.700187  13.581398 -18.215850</span>
<span class="co">#&gt; Astro_8   11.442772   -3.138051 -31.041168   22.986442  16.198882 -40.829396</span>
<span class="co">#&gt; Astro_9    8.753905 -172.107216  -7.446399   69.200780  -9.240530  -7.393721</span>
<span class="co">#&gt; Astro_10 -42.034477   -3.314376  23.923047    5.525915  10.524619  -1.069653</span>
<span class="co">#&gt;                PC31       PC32        PC33       PC34      PC35       PC36</span>
<span class="co">#&gt; Astro_1   13.580232 -69.607471  16.6579039 -26.495668  65.49716 -78.252944</span>
<span class="co">#&gt; Astro_2   -6.525119   3.483396   8.1538417  19.066489 -10.20887   7.762151</span>
<span class="co">#&gt; Astro_3    5.810426  -9.658983 -76.1098192  -2.781780  -4.87731  -3.005881</span>
<span class="co">#&gt; Astro_4  -28.487199  23.962197  -5.9613303  17.252775  15.17926  18.719909</span>
<span class="co">#&gt; Astro_5  -11.679693  -4.382173  -4.0230368 -47.109811  67.49965  25.917323</span>
<span class="co">#&gt; Astro_6   -4.221079  41.352804  14.1815273  -7.919299  13.86098   3.646989</span>
<span class="co">#&gt; Astro_7  -38.228441  10.121056 -31.3917566   5.180027 -64.89057   1.322206</span>
<span class="co">#&gt; Astro_8   12.758204  14.914500  44.5473871 -25.783057  54.17184  33.915650</span>
<span class="co">#&gt; Astro_9   62.347971 -21.511858  -0.2814753 -13.167109 -60.75524 -16.874267</span>
<span class="co">#&gt; Astro_10 -40.601569  39.919305 -28.9582106  20.420831   5.33959   7.718047</span>
<span class="co">#&gt;                PC37       PC38       PC39       PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1   19.234383  -19.72138  48.978159  -8.438157  -2.704423   8.304591</span>
<span class="co">#&gt; Astro_2    7.249256   20.57278  11.611439   2.210857 -14.957570  22.795593</span>
<span class="co">#&gt; Astro_3  -12.026184   53.14093  -8.680514 -26.336300  10.112864   7.037904</span>
<span class="co">#&gt; Astro_4  -23.695526  -34.09582  34.570352   4.923201 -47.910495 -15.061011</span>
<span class="co">#&gt; Astro_5   -6.889219  -38.01094  25.105337 -41.485029  71.087053   6.614945</span>
<span class="co">#&gt; Astro_6  -15.249171   34.83729  49.644906  34.994742  -3.293604  26.009546</span>
<span class="co">#&gt; Astro_7  -15.893129   34.73427 -42.180391  10.182665  30.075029 -38.179932</span>
<span class="co">#&gt; Astro_8   35.996855 -111.34210  14.768940 -12.423909   2.819454 -18.334980</span>
<span class="co">#&gt; Astro_9    8.840987   66.99017 -67.030869  27.755628 -14.035714  17.645652</span>
<span class="co">#&gt; Astro_10  12.966523  -27.58261 -44.618766 -69.782635 -26.383727  17.267455</span>
<span class="co">#&gt;                PC43       PC44        PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -4.145073  -8.444849  21.9471602   6.047150 -27.199368   5.215858</span>
<span class="co">#&gt; Astro_2   36.674543   1.459047  -7.8698662 -26.422441  -3.071296  -5.414492</span>
<span class="co">#&gt; Astro_3    1.882710  42.746501  24.2629300  12.988262  50.515300  67.692793</span>
<span class="co">#&gt; Astro_4   28.768820   6.959527   1.8817834  55.178998  34.769948 -13.564325</span>
<span class="co">#&gt; Astro_5  -18.638189   5.949651  -7.4544784  -8.575980 -44.524834   4.749983</span>
<span class="co">#&gt; Astro_6  -30.365114   9.156468 -47.9319598   7.531936 -20.798874  12.920457</span>
<span class="co">#&gt; Astro_7   -5.100947 -10.305991   0.5733604 -72.181916 -10.194963   5.410968</span>
<span class="co">#&gt; Astro_8   46.220330  30.644549 -53.6890401  25.336719  28.601953 -37.886459</span>
<span class="co">#&gt; Astro_9  -19.931964 -12.170762  36.1928984  28.388143  39.232422 -48.170223</span>
<span class="co">#&gt; Astro_10  35.212835  17.806082  18.0071919  -4.895024 -27.807602 -20.226128</span>
<span class="co">#&gt;                PC49        PC50</span>
<span class="co">#&gt; Astro_1   31.145673 -28.4404935</span>
<span class="co">#&gt; Astro_2  -21.514625  20.2684588</span>
<span class="co">#&gt; Astro_3  -14.452685  -5.0696845</span>
<span class="co">#&gt; Astro_4   27.367058  29.6196348</span>
<span class="co">#&gt; Astro_5   -6.657970   0.3094712</span>
<span class="co">#&gt; Astro_6   -2.914457  -8.7122316</span>
<span class="co">#&gt; Astro_7  -11.781670  32.2319840</span>
<span class="co">#&gt; Astro_8   22.288657  26.9777669</span>
<span class="co">#&gt; Astro_9   48.155778 -13.0457456</span>
<span class="co">#&gt; Astro_10 -52.804567 -80.1829432</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
