<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony converged after 9 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2        PC3        PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1   -8.969767 40.089997 -28.931637  34.992240 -39.727489 -13.834137</span>
<span class="co">#&gt; Astro_2  -11.149834 28.972019   4.487580  12.676573  -8.142019   4.915454</span>
<span class="co">#&gt; Astro_3    7.843229 16.650007  -5.810262   2.784114  12.278465  -9.976391</span>
<span class="co">#&gt; Astro_4   -3.016996 30.720151 -24.922868  10.046986  -9.421887 -12.622245</span>
<span class="co">#&gt; Astro_5    3.907945 12.487845 -20.874442  12.820043 -20.464178 -21.453012</span>
<span class="co">#&gt; Astro_6  -21.010006 11.039105 -21.222518  -1.751442  -3.719947   7.901075</span>
<span class="co">#&gt; Astro_7   -7.835735 43.243026 -59.021362 -50.280241  -6.679413  -6.637038</span>
<span class="co">#&gt; Astro_8   -1.781730 38.863262   3.141645   4.435578  -1.791360  19.843036</span>
<span class="co">#&gt; Astro_9  -12.762278 13.494298 -17.721576  14.137408  -7.130942   6.259576</span>
<span class="co">#&gt; Astro_10 -16.575015  1.699037 -14.595547   2.226926 -24.218402  -9.564946</span>
<span class="co">#&gt;                   PC7         PC8         PC9       PC10      PC11       PC12</span>
<span class="co">#&gt; Astro_1  -12.83539906 -37.9608516 46.56590161   9.149491 22.310021 -31.050737</span>
<span class="co">#&gt; Astro_2   -2.48862663  -5.2389110 10.82925483  15.683157 21.444405 -22.811844</span>
<span class="co">#&gt; Astro_3  -32.32547287  32.7863676  8.38208879  18.650401 -1.345968 -12.274705</span>
<span class="co">#&gt; Astro_4    7.24835712   0.4163443  4.38981009 -11.781630 -3.691702 -12.857437</span>
<span class="co">#&gt; Astro_5   -1.61129688  -5.2075153 35.63609544  14.630270 72.023757 -30.380654</span>
<span class="co">#&gt; Astro_6   -9.51057743  -3.5666296 -0.05624249   1.677075  7.634661 -43.094923</span>
<span class="co">#&gt; Astro_7   -0.02719537   8.7929496 -0.39305517 -10.336975  5.465266 -19.992044</span>
<span class="co">#&gt; Astro_8  -17.21089576   9.7657032 -2.70551041   0.980966 22.695435   4.662534</span>
<span class="co">#&gt; Astro_9   11.06804511  29.1926017 38.12876102  10.760006 21.294910 -63.993976</span>
<span class="co">#&gt; Astro_10  14.11059322  -3.2459321 30.89214337 -16.264277 13.130992 -29.209376</span>
<span class="co">#&gt;                 PC13       PC14       PC15        PC16        PC17       PC18</span>
<span class="co">#&gt; Astro_1   50.0683721 -48.679357   2.709258 -43.8063948  70.5724082 -44.324435</span>
<span class="co">#&gt; Astro_2    0.8093972   3.316843  -8.768100   8.0312260 -17.1094020 -18.005916</span>
<span class="co">#&gt; Astro_3   25.5384337 -20.605961 -21.217415 -12.2731665  30.9379320   1.186832</span>
<span class="co">#&gt; Astro_4   -0.4483242   2.796980 -19.625066  -8.3572138   0.2179652   3.306631</span>
<span class="co">#&gt; Astro_5   29.4449143  46.112032  -2.084978  15.1289936  -3.5077524  17.063767</span>
<span class="co">#&gt; Astro_6   29.1814861  -1.878138   5.333888 -17.8333261   3.3540220 -13.154539</span>
<span class="co">#&gt; Astro_7   15.5519509  -4.968552 -10.612055   8.2976932  15.0844900 -17.122974</span>
<span class="co">#&gt; Astro_8   -6.1441559 -36.732788  -2.587488 -20.1836019 -11.0251059  -4.734321</span>
<span class="co">#&gt; Astro_9  -28.5826503 -26.700640  25.310196  33.9983979 -15.4827308 -47.843133</span>
<span class="co">#&gt; Astro_10  19.1329053 -25.506039 -11.174037  -0.5115198  21.8699584 -25.745521</span>
<span class="co">#&gt;                PC19        PC20        PC21       PC22        PC23       PC24</span>
<span class="co">#&gt; Astro_1   37.847480  11.2590974 -143.977204  54.722655  23.9479466 -53.114603</span>
<span class="co">#&gt; Astro_2    5.947750   2.8062685    5.186568 -10.454239  -0.9533249  -2.999064</span>
<span class="co">#&gt; Astro_3   28.809402  -2.7981324   20.692018  -7.021770  -1.5235340  -5.565258</span>
<span class="co">#&gt; Astro_4  -10.016390  16.0037480   17.909384  -4.582412 -18.6778238 -19.696351</span>
<span class="co">#&gt; Astro_5  110.167720  28.8167605   76.085319  21.069063 108.5570653 -93.498523</span>
<span class="co">#&gt; Astro_6  -20.295016  -1.5998678    4.070317  48.281956  -5.9785940   4.588927</span>
<span class="co">#&gt; Astro_7   -7.824522  -0.6100963  -10.068323  13.654578 -24.4345209   7.156349</span>
<span class="co">#&gt; Astro_8   -3.581616 -17.9575650  -36.849244 -19.794655 -22.6363731  31.505191</span>
<span class="co">#&gt; Astro_9  -95.991335 -49.3892609   -5.047858 -48.117776  28.3158106  17.645708</span>
<span class="co">#&gt; Astro_10  -4.725806  -6.0999794   -1.133030 -20.904735  -2.6925834  -3.827886</span>
<span class="co">#&gt;                PC25         PC26        PC27        PC28       PC29       PC30</span>
<span class="co">#&gt; Astro_1   19.635879  -68.1173459  20.1119609 -126.715210   5.821741 -36.960491</span>
<span class="co">#&gt; Astro_2   -3.324246    6.6714705  -3.4379881   -3.248832 -11.556256  11.862835</span>
<span class="co">#&gt; Astro_3   10.086341   29.8555971 -27.2275315   36.313074  33.827127   8.121110</span>
<span class="co">#&gt; Astro_4   -2.899802    9.1859898  28.7167286   20.109869   8.032473 -13.511212</span>
<span class="co">#&gt; Astro_5   38.814285   73.5777402 -53.0826380   46.819428 -45.132075  11.356351</span>
<span class="co">#&gt; Astro_6    5.087801   -0.7547032  32.0117319  -10.566756  22.932628  -3.280320</span>
<span class="co">#&gt; Astro_7   36.116235   -6.4786687  19.0388520   11.253640  31.807671 -14.348020</span>
<span class="co">#&gt; Astro_8    6.770830   -4.6231999 -30.7392429   21.994521  14.039229 -38.694441</span>
<span class="co">#&gt; Astro_9    4.768791 -160.5037960  -0.4061018   48.534578  -2.700025 -10.337689</span>
<span class="co">#&gt; Astro_10 -41.932996   -2.1514100  18.6983116    2.576428  12.283319  -5.608106</span>
<span class="co">#&gt;                PC31       PC32        PC33       PC34       PC35        PC36</span>
<span class="co">#&gt; Astro_1    5.840268 -61.972722  17.3909078 -16.306514  62.354259 -67.1113778</span>
<span class="co">#&gt; Astro_2    3.321981 -11.017646   4.5591255   3.519164 -18.443013  -7.0293428</span>
<span class="co">#&gt; Astro_3   -1.258084  -4.561217 -75.9530108   5.312339  -6.309850   7.3445454</span>
<span class="co">#&gt; Astro_4  -21.823839  23.761915 -15.0647680  14.610203  10.206458  21.6801790</span>
<span class="co">#&gt; Astro_5  -21.901600  -1.125251  -0.8814987 -40.906556  70.476032  23.1577474</span>
<span class="co">#&gt; Astro_6   -1.177030  42.081144   9.8660816 -10.651484  12.024423   9.2331653</span>
<span class="co">#&gt; Astro_7  -42.868212   5.145955 -29.3802662  -8.219392 -47.765647   0.5277358</span>
<span class="co">#&gt; Astro_8   15.684163  17.341847  40.7085124 -23.613141  52.091666  35.1228656</span>
<span class="co">#&gt; Astro_9   54.450629 -18.145862   2.8940363  -3.891396 -57.986711 -18.4838542</span>
<span class="co">#&gt; Astro_10 -37.419240  40.549001 -33.2421674  17.635568   3.654836  13.1950112</span>
<span class="co">#&gt;                PC37       PC38      PC39        PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1   14.772855  -13.39803  43.95249  -5.6461724  -1.287442   9.943955</span>
<span class="co">#&gt; Astro_2    4.982810   19.52708  10.26729   0.4722274  -3.125572  25.021891</span>
<span class="co">#&gt; Astro_3  -16.626702   59.61133 -11.19303 -24.1944250  11.081123   7.381702</span>
<span class="co">#&gt; Astro_4  -24.760199  -14.82105  36.63306   9.0699140 -38.437518  -6.309851</span>
<span class="co">#&gt; Astro_5   -8.069993  -39.97963  34.85295 -40.6217331  68.614970   4.858845</span>
<span class="co">#&gt; Astro_6  -17.279905   30.01016  44.36415  38.7987490  -4.190558  27.722072</span>
<span class="co">#&gt; Astro_7  -17.118873   24.71473 -23.85650   0.3684970  27.069261 -39.895820</span>
<span class="co">#&gt; Astro_8   38.899105 -101.99607  12.60670  -9.7854013  10.762517 -12.347851</span>
<span class="co">#&gt; Astro_9    7.728531   65.07066 -57.71304  28.9283231 -17.389348  15.151434</span>
<span class="co">#&gt; Astro_10  10.834696  -32.25103 -49.78838 -66.0562572 -27.114737  18.832764</span>
<span class="co">#&gt;                 PC43       PC44       PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -6.0805997 -11.311189  24.186428   7.051189 -26.014140   5.489425</span>
<span class="co">#&gt; Astro_2   41.5782998  -4.207760  -9.732243 -32.405172  13.544583 -14.669421</span>
<span class="co">#&gt; Astro_3   -0.2624106  38.360898  29.574113  13.447558  52.324093  71.623277</span>
<span class="co">#&gt; Astro_4   17.1223036   1.538473  14.612408  44.958658  27.071162   3.040288</span>
<span class="co">#&gt; Astro_5  -10.8234279   7.388717 -14.571108 -11.987744 -45.021274   5.448354</span>
<span class="co">#&gt; Astro_6  -34.0874419   6.374269 -51.737890   5.542779 -22.428837  11.272716</span>
<span class="co">#&gt; Astro_7   -2.5680533 -26.044433  14.122567 -73.070322  -4.468048  17.095225</span>
<span class="co">#&gt; Astro_8   40.6430911  28.037896 -54.131688  18.888297  22.326421 -33.614856</span>
<span class="co">#&gt; Astro_9  -12.1726716  -9.089355  29.353754  26.506861  40.025029 -45.932441</span>
<span class="co">#&gt; Astro_10  31.4761753  14.978342  14.293342  -6.801287 -29.480495 -21.884060</span>
<span class="co">#&gt;                PC49       PC50</span>
<span class="co">#&gt; Astro_1   36.800584 -21.602143</span>
<span class="co">#&gt; Astro_2   -8.265024  32.512026</span>
<span class="co">#&gt; Astro_3   -6.896702   2.728311</span>
<span class="co">#&gt; Astro_4   33.736715  26.121716</span>
<span class="co">#&gt; Astro_5  -15.792139   1.535368</span>
<span class="co">#&gt; Astro_6   -3.562260  -7.017650</span>
<span class="co">#&gt; Astro_7   -3.337298  36.045073</span>
<span class="co">#&gt; Astro_8   16.912190  17.928788</span>
<span class="co">#&gt; Astro_9   38.878708 -11.751123</span>
<span class="co">#&gt; Astro_10 -53.479679 -78.411554</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
