<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<div class="figure">
<img src="../figures/bandnorm_intro.png" alt=""><p class="caption">The introduction for BandNorm</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony converged after 9 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2        PC3         PC4         PC5        PC6</span>
<span class="co">#&gt; Astro_1  -17.410140 25.494663 -19.480286  34.2690571 -37.1122025 -25.464748</span>
<span class="co">#&gt; Astro_2    9.561101 55.027226 -54.792366 -55.7406411 -14.2623791   3.614309</span>
<span class="co">#&gt; Astro_3    1.683535  9.208540  -1.867107   1.1854228  11.9501638 -25.697850</span>
<span class="co">#&gt; Astro_4   -6.200497 27.988144 -27.294043   6.7155598  -9.9836926 -17.407310</span>
<span class="co">#&gt; Astro_5    3.205684  9.184638 -19.671400  11.4220857 -21.3390385 -19.489509</span>
<span class="co">#&gt; Astro_6  -15.649914 15.356238 -20.622328  -4.7440280  -1.9930015  13.157134</span>
<span class="co">#&gt; Astro_7  -23.536566 21.844362 -10.611313  -1.2447859  -0.3913154  -5.241657</span>
<span class="co">#&gt; Astro_8  -35.818404 30.597797 -13.914001   9.8165486 -19.8764394 -16.117398</span>
<span class="co">#&gt; Astro_9  -13.437842 10.195098 -16.587119  12.6951651  -7.9965350   8.231271</span>
<span class="co">#&gt; Astro_10 -10.784257  5.765060 -13.677581  -0.6267457 -22.5840728  -4.499106</span>
<span class="co">#&gt;                 PC7          PC8        PC9       PC10      PC11       PC12</span>
<span class="co">#&gt; Astro_1    2.638351 -50.70572513 38.0415231  -2.636112 35.960952 -20.099967</span>
<span class="co">#&gt; Astro_2   -2.377189   2.21201053 17.2342928  10.401020 26.867821 -34.015630</span>
<span class="co">#&gt; Astro_3  -15.126349  22.63693506 -0.4792958   9.134261 10.450195  -6.011320</span>
<span class="co">#&gt; Astro_4    6.399372   2.77994557 -1.2614367 -12.361952 -3.406752 -11.396054</span>
<span class="co">#&gt; Astro_5    2.642024  -1.78732203 37.7057810  17.181958 78.678345 -34.014077</span>
<span class="co">#&gt; Astro_6  -12.425329  -0.52965010 -3.6592940   4.331817  6.492571 -41.650122</span>
<span class="co">#&gt; Astro_7   -3.999190   4.97635786 -8.3065713  -6.000965 -3.622839  -4.035006</span>
<span class="co">#&gt; Astro_8   -9.894281  14.27877609 16.3305854 -34.498939 -4.516919   3.925841</span>
<span class="co">#&gt; Astro_9   15.301209  32.63923507 40.2093633  13.324488 27.958856 -67.649659</span>
<span class="co">#&gt; Astro_10  11.434423   0.03816165 27.4101712 -13.567914 12.613922 -27.898073</span>
<span class="co">#&gt;                PC13        PC14        PC15       PC16        PC17       PC18</span>
<span class="co">#&gt; Astro_1   34.893955 -15.1060105  29.3979125 -11.810200  45.0775294 -46.857373</span>
<span class="co">#&gt; Astro_2    2.958462  -0.5706396  -4.5273624  11.125619  -3.3634642 -14.342152</span>
<span class="co">#&gt; Astro_3   15.366001   9.2143969  -0.8872121  15.846444  12.0555174  -4.138300</span>
<span class="co">#&gt; Astro_4   -8.028570   7.7732926 -20.6506159  -5.404709  -0.7584566   5.991618</span>
<span class="co">#&gt; Astro_5   31.885293  43.3238932  -4.4413090  12.608177  -0.2297105  15.885575</span>
<span class="co">#&gt; Astro_6   23.359789   2.7812242   5.0533214 -15.138698  -4.6582773  -7.010287</span>
<span class="co">#&gt; Astro_7   14.123259   1.5707693  -9.9066753   6.465005   6.7825587 -17.416610</span>
<span class="co">#&gt; Astro_8   -4.820231 -33.5827710   2.1352066 -17.713664  11.1849868 -15.364835</span>
<span class="co">#&gt; Astro_9  -26.126305 -29.5169686  22.9350440  31.461582 -12.1708240 -49.017528</span>
<span class="co">#&gt; Astro_10  12.848309 -20.2983928 -11.1296495   2.761635  13.3926611 -19.443327</span>
<span class="co">#&gt;                PC19       PC20         PC21       PC22       PC23       PC24</span>
<span class="co">#&gt; Astro_1   48.951127  27.957638 -157.6949008  62.015088  22.845954 -49.453088</span>
<span class="co">#&gt; Astro_2   -1.059895  -4.180835    2.8746169  -5.340805  -2.389381   2.322775</span>
<span class="co">#&gt; Astro_3   34.988615  15.282764    1.0329751   1.875829  -6.054706  -4.717917</span>
<span class="co">#&gt; Astro_4   -9.011515  14.047718   15.7555590 -13.655194 -28.054581  -7.650544</span>
<span class="co">#&gt; Astro_5  113.917753  26.254411   81.3586089  15.482127 117.697578 -98.184054</span>
<span class="co">#&gt; Astro_6  -23.664528 -12.330412   19.1710814  38.481957 -10.453734  17.752131</span>
<span class="co">#&gt; Astro_7   -4.524567  -5.329433   11.2101649   6.540224 -22.047587  -2.479816</span>
<span class="co">#&gt; Astro_8   38.382421   2.743845  -51.0669738 -35.274721  -6.700870  55.521972</span>
<span class="co">#&gt; Astro_9  -92.248368 -51.977983    0.2634312 -53.721783  37.497943  12.935128</span>
<span class="co">#&gt; Astro_10  -7.777228 -17.139724   15.0351499 -31.343462  -6.175282   8.781373</span>
<span class="co">#&gt;                 PC25        PC26       PC27         PC28       PC29       PC30</span>
<span class="co">#&gt; Astro_1   17.2538885  -72.287672   6.810386 -141.3993517   1.494639 -31.976729</span>
<span class="co">#&gt; Astro_2   -3.9622365   -1.402800   1.480427   -2.2635311   1.855233  10.062486</span>
<span class="co">#&gt; Astro_3   11.2451862   22.094295 -38.189756   15.3222820  32.939979   7.241761</span>
<span class="co">#&gt; Astro_4   -1.4760143   21.880213  18.680734   19.1762489   6.719654  -7.760212</span>
<span class="co">#&gt; Astro_5   41.5705437   67.041285 -54.889418   50.2433915 -48.541976  12.393887</span>
<span class="co">#&gt; Astro_6    1.9500796   13.412457  24.733788    0.4438286  23.001485   7.388517</span>
<span class="co">#&gt; Astro_7   34.5581496   19.601552  16.804349   22.2831742  12.055649 -18.429589</span>
<span class="co">#&gt; Astro_8   -0.7497981    1.273784 -39.499127   39.0829660   2.163528 -33.172134</span>
<span class="co">#&gt; Astro_9    7.5346597 -167.052925  -2.212636   51.9978724  -6.118105  -9.300657</span>
<span class="co">#&gt; Astro_10 -44.8123324   11.880044  10.779472   14.8442184  11.505299   5.232621</span>
<span class="co">#&gt;                PC31        PC32        PC33       PC34       PC35       PC36</span>
<span class="co">#&gt; Astro_1   23.306122 -82.8079393  15.2812439 -28.739900  63.698870 -83.338917</span>
<span class="co">#&gt; Astro_2   -3.121833   4.1116351   9.8531243  13.666110  -5.366403  10.447920</span>
<span class="co">#&gt; Astro_3   14.295306 -22.7258385 -75.3443417  -5.857261  -2.276445  -6.442921</span>
<span class="co">#&gt; Astro_4  -22.462174  14.5164710  -7.3727610  18.148981  17.622218  17.921558</span>
<span class="co">#&gt; Astro_5  -18.957273  -0.6550348  -0.2811466 -42.906601  70.503229  23.890799</span>
<span class="co">#&gt; Astro_6   -9.050216  52.2247055  16.9102586   1.246012  14.652903  16.573884</span>
<span class="co">#&gt; Astro_7  -28.829829  -3.8874878 -33.4415256   1.963935 -65.089308  -3.095247</span>
<span class="co">#&gt; Astro_8   13.875232  13.2097018  54.9124768 -32.473082  45.214984  35.717249</span>
<span class="co">#&gt; Astro_9   57.382649 -17.6470662   3.4884289  -5.883856 -57.970842 -17.731259</span>
<span class="co">#&gt; Astro_10 -45.142652  50.3425279 -26.2687010  29.039705   5.767299  20.185158</span>
<span class="co">#&gt;                PC37      PC38       PC39       PC40       PC41        PC42</span>
<span class="co">#&gt; Astro_1   17.095536 -18.97547  48.645510  -3.963873   4.010915   3.3030398</span>
<span class="co">#&gt; Astro_2    2.328210  14.95864   5.789421  -4.229922  -8.089407  20.6751440</span>
<span class="co">#&gt; Astro_3  -17.301948  56.22902  -5.240622 -25.672355  13.667848   0.9663274</span>
<span class="co">#&gt; Astro_4  -26.724317 -33.55285  38.178072   8.005306 -44.736131 -20.8443088</span>
<span class="co">#&gt; Astro_5   -8.835019 -36.66229  30.898131 -41.718469  72.439273   7.7929683</span>
<span class="co">#&gt; Astro_6  -13.697482  31.28773  39.210126  44.833999  -6.949211  27.7550894</span>
<span class="co">#&gt; Astro_7  -19.159287  35.28345 -40.999401  13.844191  36.494412 -44.2561786</span>
<span class="co">#&gt; Astro_8   57.491447 -87.71583  19.956565 -30.358534 -12.603914 -19.3265170</span>
<span class="co">#&gt; Astro_9    6.956657  68.39481 -61.688944  27.830581 -13.560071  18.0942706</span>
<span class="co">#&gt; Astro_10  14.626552 -31.15094 -55.377152 -59.835162 -29.636377  19.0574114</span>
<span class="co">#&gt;                PC43       PC44       PC45        PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -3.651702  -9.296834  29.159998   8.7385990 -13.462191  19.892593</span>
<span class="co">#&gt; Astro_2   29.755365  -3.416570 -10.772615 -27.3320895   1.656046 -13.365692</span>
<span class="co">#&gt; Astro_3   -1.736975  38.483129  36.828579  14.8858141  56.434199  85.297133</span>
<span class="co">#&gt; Astro_4   28.858403   4.948306  10.161513  56.2709599  44.542867   2.542473</span>
<span class="co">#&gt; Astro_5  -14.225178   6.739684 -14.063304 -13.8848198 -47.162508   2.518002</span>
<span class="co">#&gt; Astro_6  -36.521773  15.460292 -57.600289  12.8257894 -34.252181  17.281463</span>
<span class="co">#&gt; Astro_7   -5.529226 -13.045228  10.246778 -70.0970196   3.745759  21.975879</span>
<span class="co">#&gt; Astro_8   52.655553  26.575209 -40.715868  43.4199206  47.815974 -46.954556</span>
<span class="co">#&gt; Astro_9  -15.585712  -9.745535  29.857023  24.6097641  37.878244 -48.889803</span>
<span class="co">#&gt; Astro_10  29.097879  24.074235   8.361039   0.4917049 -40.867814 -15.969362</span>
<span class="co">#&gt;                PC49       PC50</span>
<span class="co">#&gt; Astro_1   44.494862 -16.551337</span>
<span class="co">#&gt; Astro_2  -15.696573  21.731132</span>
<span class="co">#&gt; Astro_3   -1.153426   3.810867</span>
<span class="co">#&gt; Astro_4   36.643503  39.504956</span>
<span class="co">#&gt; Astro_5  -14.444272   1.753575</span>
<span class="co">#&gt; Astro_6  -13.328209 -14.353127</span>
<span class="co">#&gt; Astro_7    2.863665  44.651038</span>
<span class="co">#&gt; Astro_8   18.529651  40.688885</span>
<span class="co">#&gt; Astro_9   40.229183 -11.535861</span>
<span class="co">#&gt; Astro_10 -62.690558 -85.565060</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
