<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<div class="figure">
<img src="../figures/bandnorm_intro.png" alt=""><p class="caption">The introduction for BandNorm</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are two possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1  38.000000</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 105.000000</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 116.000000</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 133.000000</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 112.000000</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony converged after 9 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                  PC1       PC2        PC3       PC4         PC5         PC6</span>
<span class="co">#&gt; Astro_1  -14.6501131 35.794040 -26.929957 33.270183 -33.4263965 -14.2704124</span>
<span class="co">#&gt; Astro_2  -11.5493750 32.890940   6.022335 12.747598  -6.8500454   4.4846891</span>
<span class="co">#&gt; Astro_3  -17.9082522 -2.696272 -11.285644 11.141142  -3.6862742 -48.0467603</span>
<span class="co">#&gt; Astro_4   -4.0823762 30.183169 -23.291190 10.309232  -8.4422266 -12.4845300</span>
<span class="co">#&gt; Astro_5   16.1390904 16.817009 -26.294061  9.440128 -19.1067581 -11.5911731</span>
<span class="co">#&gt; Astro_6  -21.0654989 16.484312 -20.201085 -1.818684  -2.0594665   8.8280593</span>
<span class="co">#&gt; Astro_7  -42.9475014 15.403276 -21.558178 11.339661 -19.2460629 -32.3324311</span>
<span class="co">#&gt; Astro_8    0.8464861 42.795417   1.403291  7.676227   0.2831818  16.4428145</span>
<span class="co">#&gt; Astro_9   -9.0018860  8.421569 -15.209882 14.800875  -9.5577083   0.3454511</span>
<span class="co">#&gt; Astro_10 -16.6289712  6.869891 -13.377367  2.156844 -22.7214611  -9.0985568</span>
<span class="co">#&gt;                 PC7        PC8       PC9        PC10       PC11       PC12</span>
<span class="co">#&gt; Astro_1   -9.893100 -38.275330 40.390721   5.1422528  14.063077 -27.273650</span>
<span class="co">#&gt; Astro_2   -6.992088  -8.607405  4.610836  14.5242222  19.683410 -18.141533</span>
<span class="co">#&gt; Astro_3   -8.350202  12.640407 29.487530 -28.7159924 -11.621696   3.415619</span>
<span class="co">#&gt; Astro_4    7.568261  -0.646856  1.160359 -12.7661866  -3.005631 -11.686442</span>
<span class="co">#&gt; Astro_5   -4.628303  19.387042 43.690896  28.8382653  77.916945 -46.538177</span>
<span class="co">#&gt; Astro_6  -15.713098  -6.602876 -6.618219   0.9296112   2.832349 -38.269315</span>
<span class="co">#&gt; Astro_7    5.081892  -5.312322 20.034557 -43.3152336 -27.680622   2.033148</span>
<span class="co">#&gt; Astro_8  -19.700065  12.444542 -1.329663  -0.5531044  26.726897   1.821310</span>
<span class="co">#&gt; Astro_9   17.646133  34.632471 40.705946  13.5014833  45.501442 -69.275670</span>
<span class="co">#&gt; Astro_10   8.235005  -6.686641 24.304774 -17.4052485   8.677598 -24.189067</span>
<span class="co">#&gt;                 PC13       PC14       PC15       PC16        PC17       PC18</span>
<span class="co">#&gt; Astro_1   47.6585614 -49.297386   1.422643 -41.302332  71.2174526 -46.930191</span>
<span class="co">#&gt; Astro_2    6.1233578   7.679479  -7.921929   4.651489 -16.5765769 -11.320522</span>
<span class="co">#&gt; Astro_3   17.3895701   1.618277  10.506708  19.637418  33.8763856 -13.576092</span>
<span class="co">#&gt; Astro_4   -0.7170254   4.304638 -17.367347  -6.988495  -0.8474449   3.055825</span>
<span class="co">#&gt; Astro_5   31.5170290  23.430777 -21.159022   2.504391   9.4134301  18.823474</span>
<span class="co">#&gt; Astro_6   35.1268607   0.280321   4.033280 -23.508942   5.1458926  -6.029884</span>
<span class="co">#&gt; Astro_7   19.9729557  -6.940143  -2.773257   8.219398  32.8314915 -30.009543</span>
<span class="co">#&gt; Astro_8   -2.2116034 -47.740928   1.140185 -12.547012 -14.2979985   5.550314</span>
<span class="co">#&gt; Astro_9  -27.3510272 -17.344704  33.701714  42.900302 -19.2709876 -44.891560</span>
<span class="co">#&gt; Astro_10  25.0097923 -22.696931 -11.961295  -5.772675  23.4792874 -18.575840</span>
<span class="co">#&gt;                PC19        PC20        PC21       PC22       PC23        PC24</span>
<span class="co">#&gt; Astro_1   26.354310  15.3447029 -162.774726  58.502400   2.145823  -35.007123</span>
<span class="co">#&gt; Astro_2   14.618279   6.4442004    4.375259  -5.036139 -11.283573    4.385480</span>
<span class="co">#&gt; Astro_3   57.408634  19.8447105  -20.030937  -5.087167  10.756442   26.603278</span>
<span class="co">#&gt; Astro_4   -9.113338  18.2380014   13.068773  -2.887471 -22.413293  -15.900777</span>
<span class="co">#&gt; Astro_5   98.752527   4.3635566  113.855077  -8.552757 133.621093 -103.118981</span>
<span class="co">#&gt; Astro_6  -14.113550   0.6363645    2.314855  53.811966 -20.331774   15.358698</span>
<span class="co">#&gt; Astro_7   14.384259   2.6937196  -17.697729   2.158267  -9.642193   27.645593</span>
<span class="co">#&gt; Astro_8    5.302932 -22.7811478  -38.102588 -21.382442 -32.445043   34.079146</span>
<span class="co">#&gt; Astro_9  -74.598668 -47.5540582   19.997057 -58.160452  63.704528   -8.776339</span>
<span class="co">#&gt; Astro_10   2.182202  -3.4145137   -3.249810 -15.014111 -16.907016    6.711440</span>
<span class="co">#&gt;                PC25        PC26       PC27        PC28       PC29       PC30</span>
<span class="co">#&gt; Astro_1   11.398981  -77.629777  23.431427 -142.650139  19.902931 -37.106898</span>
<span class="co">#&gt; Astro_2   -1.441752   26.815617  -8.657797  -12.639073  -6.395343  14.130070</span>
<span class="co">#&gt; Astro_3    3.457160   12.388722 -33.298558   23.638493  22.569676  10.984663</span>
<span class="co">#&gt; Astro_4   -4.084076   10.104320  27.297939   16.738369   9.910400 -14.503708</span>
<span class="co">#&gt; Astro_5   45.720676   52.728120 -59.220034   91.897657 -51.432769  16.977010</span>
<span class="co">#&gt; Astro_6    5.501955   21.407782  28.429067  -21.125872  30.338044  -0.840262</span>
<span class="co">#&gt; Astro_7   29.548669    6.937027  23.086366   22.501923   5.178714 -19.884899</span>
<span class="co">#&gt; Astro_8    8.078732   -5.878556 -33.353586   23.443458  11.614629 -40.743229</span>
<span class="co">#&gt; Astro_9   21.062966 -167.668044 -19.315371   72.105035 -20.045361  -8.025357</span>
<span class="co">#&gt; Astro_10 -41.384532   20.272377  14.919187   -8.398508  19.272539  -3.377336</span>
<span class="co">#&gt;                PC31       PC32       PC33       PC34       PC35         PC36</span>
<span class="co">#&gt; Astro_1   13.940749 -66.516080  12.212057 -12.046288  58.174886 -67.70078683</span>
<span class="co">#&gt; Astro_2   -5.156106  -9.923566   5.308041   8.344931  -9.364407  -0.07797766</span>
<span class="co">#&gt; Astro_3   17.109007 -22.333696 -70.395349  -6.091890  11.743566   1.53427672</span>
<span class="co">#&gt; Astro_4  -21.324727  22.582113 -15.980537  14.452987  11.352268  22.09100935</span>
<span class="co">#&gt; Astro_5  -19.315350   5.393946  -4.062355 -40.288770  54.248420  30.09942800</span>
<span class="co">#&gt; Astro_6  -11.793116  46.043744  11.418334  -2.602233  21.629464  18.34668433</span>
<span class="co">#&gt; Astro_7  -26.709243  -2.366446 -25.573412   2.228727 -48.331837   6.02339783</span>
<span class="co">#&gt; Astro_8   16.427046  14.656218  44.214377 -21.430029  56.206216  35.01209914</span>
<span class="co">#&gt; Astro_9   63.697973 -34.314379  -2.700084 -23.906961 -61.252516 -25.63741360</span>
<span class="co">#&gt; Astro_10 -48.038519  44.068115 -31.752419  25.249644  13.372285  22.03897287</span>
<span class="co">#&gt;                PC37         PC38       PC39       PC40       PC41       PC42</span>
<span class="co">#&gt; Astro_1   13.868463   -0.6927935  42.376992  -5.701220 -10.856925   7.248710</span>
<span class="co">#&gt; Astro_2    4.475232    9.9682314   9.128378   1.423304   2.253510  13.710129</span>
<span class="co">#&gt; Astro_3  -13.151559   43.9942176  -1.671821 -24.439618   6.339248   7.423572</span>
<span class="co">#&gt; Astro_4  -23.419123  -17.0369511  37.234320   8.503275 -37.215341  -8.656405</span>
<span class="co">#&gt; Astro_5   -8.394536  -19.5274464  11.248576 -39.841049  68.380108  14.480922</span>
<span class="co">#&gt; Astro_6  -17.680420   21.5788231  43.486581  40.214256  -0.980155  16.713526</span>
<span class="co">#&gt; Astro_7  -16.940985   25.5687913 -35.389230  12.138103  26.226685 -37.756463</span>
<span class="co">#&gt; Astro_8   37.058120 -108.0574375  14.017434 -11.054379   8.619459 -17.433400</span>
<span class="co">#&gt; Astro_9    6.534253   57.0276154 -70.237030  26.741054   3.365223   9.530105</span>
<span class="co">#&gt; Astro_10  10.516028  -41.3697639 -50.498112 -64.670382 -23.635131   7.683832</span>
<span class="co">#&gt;                PC43       PC44       PC45       PC46       PC47       PC48</span>
<span class="co">#&gt; Astro_1   -3.878339 -10.314664  37.828601  11.684125 -11.916266  16.785764</span>
<span class="co">#&gt; Astro_2   37.327142  -2.381841 -10.006133 -32.252898   3.281723   0.364976</span>
<span class="co">#&gt; Astro_3   -3.907478  38.283763  28.895170  25.640104  47.706991  60.366670</span>
<span class="co">#&gt; Astro_4   18.227538   2.045541  15.931397  46.316045  28.189990   4.057235</span>
<span class="co">#&gt; Astro_5  -15.623888   6.325681  -7.468943  -6.191530 -36.765109  -6.238910</span>
<span class="co">#&gt; Astro_6  -38.284608   9.220413 -52.431195   6.481284 -34.515911  26.724122</span>
<span class="co">#&gt; Astro_7  -10.926565 -14.445863   4.925937 -59.342373 -13.075433  -2.045919</span>
<span class="co">#&gt; Astro_8   44.496551  28.977305 -55.128077  22.695489  26.817120 -36.864746</span>
<span class="co">#&gt; Astro_9  -18.460910 -15.417296  32.789926  22.713878  46.220920 -44.824053</span>
<span class="co">#&gt; Astro_10  27.297937  17.702324  13.518046  -6.137671 -41.460830  -6.196669</span>
<span class="co">#&gt;                 PC49        PC50</span>
<span class="co">#&gt; Astro_1   37.6346540 -19.3810854</span>
<span class="co">#&gt; Astro_2   -9.9394938  34.6292588</span>
<span class="co">#&gt; Astro_3   -3.0954477   0.3879654</span>
<span class="co">#&gt; Astro_4   37.0493223  26.2635340</span>
<span class="co">#&gt; Astro_5   -7.8278301   3.5619737</span>
<span class="co">#&gt; Astro_6   -7.7873799  -6.4565146</span>
<span class="co">#&gt; Astro_7   -0.6113016  37.5039600</span>
<span class="co">#&gt; Astro_8   20.4901664  19.6949220</span>
<span class="co">#&gt; Astro_9   59.1490729   0.9830698</span>
<span class="co">#&gt; Astro_10 -57.3367617 -77.6523854</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
