<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD_tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; Warning: package 'rmarkdown' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridis</span>
<span class="co">#&gt; Warning: package 'viridis' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridisLite</span>
<span class="co">#&gt; Warning: package 'viridisLite' was built under R version 4.0.5</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony converged after 9 iterations</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2         PC3        PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1  -21.209229 22.881296 -17.8533293 34.5588311 -34.373823 -26.951954</span>
<span class="co">#&gt; Astro_2  -12.865771 29.328250   0.9002136 11.5941812  -7.150899   3.522418</span>
<span class="co">#&gt; Astro_3   -7.329751  6.294433   2.6987324  6.3563955  10.665653 -15.275469</span>
<span class="co">#&gt; Astro_4  -12.462946 29.202916 -25.8234485 12.8369077 -14.141982 -13.323485</span>
<span class="co">#&gt; Astro_5    5.534366 10.413147 -19.5002178 14.7639248 -27.543240 -18.016304</span>
<span class="co">#&gt; Astro_6  -19.286209 12.851194 -18.9504378 -4.3111747   0.525343  11.578020</span>
<span class="co">#&gt; Astro_7  -27.910860 25.912293 -13.3435861  1.2505812  -3.609954  -2.777401</span>
<span class="co">#&gt; Astro_8  -10.515319 31.492580  -3.5129405  3.9499982  -1.195907   8.700791</span>
<span class="co">#&gt; Astro_9  -10.729415  4.404571 -20.1492413 13.7118300  -9.853733  -1.064233</span>
<span class="co">#&gt; Astro_10 -14.641335  3.410721 -12.2595392 -0.4164784 -20.028768  -6.028543</span>
<span class="co">#&gt;                 PC7         PC8       PC9       PC10       PC11       PC12</span>
<span class="co">#&gt; Astro_1    5.275359 -47.6900671 36.065488  -3.427872  35.544465 -21.740480</span>
<span class="co">#&gt; Astro_2   -4.067788 -12.6696542 11.237481  11.500097  14.624717 -20.128994</span>
<span class="co">#&gt; Astro_3  -19.116675  14.0991083  5.654399   5.562774   1.754219  -6.636466</span>
<span class="co">#&gt; Astro_4    7.192957  -1.4630334  5.267943 -13.582029  -8.704257 -15.976904</span>
<span class="co">#&gt; Astro_5    6.350169  -1.3472367 46.380318  17.310541  89.815766 -44.155456</span>
<span class="co">#&gt; Astro_6   -9.670464   2.4105485 -5.483572   3.498559   6.443768 -43.441853</span>
<span class="co">#&gt; Astro_7   -7.971600  -0.9219827 -3.374918  -6.204598 -14.920849  -9.409871</span>
<span class="co">#&gt; Astro_8  -19.096096   9.0313560 -2.653180  -1.239420  17.876600   6.367932</span>
<span class="co">#&gt; Astro_9   20.753710  29.9442682 47.108445  10.161529  40.740321 -70.807020</span>
<span class="co">#&gt; Astro_10  14.031073   2.7676357 25.513172 -14.540016  11.965576 -29.561793</span>
<span class="co">#&gt;                PC13        PC14       PC15        PC16       PC17        PC18</span>
<span class="co">#&gt; Astro_1   33.840918 -17.3479113  26.189447 -12.0878240  46.917959 -47.0455390</span>
<span class="co">#&gt; Astro_2    9.601984   8.7669675  -9.795218   3.5626399 -10.161567 -11.3069311</span>
<span class="co">#&gt; Astro_3   15.459518  -3.6322702   2.640329   6.0039945  14.287083  -8.5950265</span>
<span class="co">#&gt; Astro_4   -6.748154  -1.8500846 -18.572362 -11.8628626   1.814117  -0.6286640</span>
<span class="co">#&gt; Astro_5   23.175302  43.4397781   6.553054  20.5387115  -9.088165  11.4042236</span>
<span class="co">#&gt; Astro_6   22.256591   0.7314074   2.118300 -15.1938585  -3.006838  -7.2329158</span>
<span class="co">#&gt; Astro_7   18.500506  -8.1361369 -12.840909  -0.6689911  14.928993 -21.9666192</span>
<span class="co">#&gt; Astro_8   -1.086532 -33.0390176  -8.094720 -25.4109635   2.409790  -0.3252141</span>
<span class="co">#&gt; Astro_9  -23.675810 -15.2824916  32.386649  42.0016904 -12.786498 -44.8426840</span>
<span class="co">#&gt; Astro_10  12.180849 -22.7776021 -14.404331   2.1531063  15.613984 -19.7319398</span>
<span class="co">#&gt;                PC19       PC20        PC21       PC22       PC23         PC24</span>
<span class="co">#&gt; Astro_1   49.712652  32.970418 -153.760183  60.431801  25.205533  -53.2629534</span>
<span class="co">#&gt; Astro_2   11.743578   8.743340    7.025951   3.753170 -10.030888   -0.5698706</span>
<span class="co">#&gt; Astro_3   16.886712   7.651817  -19.602979   9.105116 -17.966499   14.1429326</span>
<span class="co">#&gt; Astro_4  -27.413542   8.085301   -2.702125  -8.855034 -35.389287    4.1091367</span>
<span class="co">#&gt; Astro_5   99.593848  12.966536   88.867232   5.860606 137.669979 -100.3862866</span>
<span class="co">#&gt; Astro_6  -22.961274  -7.499962   23.474647  36.812626  -7.465026   13.5353111</span>
<span class="co">#&gt; Astro_7  -17.282988  -5.762575  -11.495829  17.522184 -35.463073    7.4027588</span>
<span class="co">#&gt; Astro_8    5.532716  -9.651881  -40.225566 -15.519318 -31.579706   32.6066075</span>
<span class="co">#&gt; Astro_9  -76.419870 -44.346564   22.295971 -48.512374  65.095624  -14.2335535</span>
<span class="co">#&gt; Astro_10  -7.214651 -11.969263   18.251553 -32.348408  -4.225148    5.0299886</span>
<span class="co">#&gt;                  PC25         PC26       PC27         PC28       PC29</span>
<span class="co">#&gt; Astro_1   18.29753518  -76.2419109  14.942948 -136.4641350  -1.858504</span>
<span class="co">#&gt; Astro_2   -8.19411266   31.3289170   3.425520  -20.4420848  -2.154758</span>
<span class="co">#&gt; Astro_3   -0.08690666    4.1300413 -18.985167   -2.1231924  41.275210</span>
<span class="co">#&gt; Astro_4   -6.91938569    0.4020721  33.598165    4.0130890  12.764316</span>
<span class="co">#&gt; Astro_5   43.24855717   35.7471998 -54.569849   66.3718325 -57.347102</span>
<span class="co">#&gt; Astro_6    3.10594271    9.0648906  32.705291    5.8261982  19.419671</span>
<span class="co">#&gt; Astro_7   30.37642968    6.4926655  30.252221   -0.2290738  21.130082</span>
<span class="co">#&gt; Astro_8    9.55598999   24.3156713 -32.433311    6.1666292  17.068131</span>
<span class="co">#&gt; Astro_9   14.49314367 -162.0912012  -7.507147   63.2429047 -15.840415</span>
<span class="co">#&gt; Astro_10 -43.91426115    7.8322478  19.377435   18.9803431   8.581522</span>
<span class="co">#&gt;                PC30       PC31       PC32        PC33       PC34       PC35</span>
<span class="co">#&gt; Astro_1  -35.890994  28.001060 -84.781000   9.8617939 -35.151643  52.134920</span>
<span class="co">#&gt; Astro_2   15.544269 -14.086019   3.419242  -0.6937072  16.752313  -6.086073</span>
<span class="co">#&gt; Astro_3    9.833544  14.448511 -16.211218 -65.8944291   3.552218  -3.317627</span>
<span class="co">#&gt; Astro_4   -7.682384 -21.352844  19.059386   4.0239187  24.339877  17.135381</span>
<span class="co">#&gt; Astro_5   18.562695 -10.449967  -3.029641  11.4524982 -47.187896  62.330171</span>
<span class="co">#&gt; Astro_6    3.570080  -4.251068  50.123392  11.5597503  -5.338238   3.053448</span>
<span class="co">#&gt; Astro_7  -25.014760 -31.314010   2.330715 -27.9647961   7.717958 -61.606056</span>
<span class="co">#&gt; Astro_8  -34.964381   2.686789  11.157650  39.1152712 -17.641716  68.376857</span>
<span class="co">#&gt; Astro_9   -6.634259  54.546776 -21.463413  -9.0574139 -15.898577 -57.366039</span>
<span class="co">#&gt; Astro_10   1.065363 -40.623704  48.552222 -31.6871021  22.839107  -5.476938</span>
<span class="co">#&gt;                PC36        PC37       PC38       PC39       PC40       PC41</span>
<span class="co">#&gt; Astro_1  -85.621787  15.8312268  -13.43988  50.791164 -10.614668   8.625062</span>
<span class="co">#&gt; Astro_2    8.073203  -4.5486281   26.67995  12.034571  -2.305350  -6.663039</span>
<span class="co">#&gt; Astro_3   -8.681998 -10.5765094   56.81035  -1.907075 -22.257600   5.097017</span>
<span class="co">#&gt; Astro_4   14.387643 -22.4526483  -31.31612  40.345090   8.878987 -53.198586</span>
<span class="co">#&gt; Astro_5   13.856074  -0.9409281  -35.30491  23.236645 -38.322917  71.650933</span>
<span class="co">#&gt; Astro_6   14.186760 -14.9475679   36.88475  41.148382  38.221390  -2.281636</span>
<span class="co">#&gt; Astro_7   -1.893929 -18.5888466   39.36624 -34.527436  11.352181  28.378012</span>
<span class="co">#&gt; Astro_8   38.147227  23.8238352 -120.10335  17.852754 -18.611237   3.009555</span>
<span class="co">#&gt; Astro_9  -17.591554  -2.7529495   72.90740 -67.025205  22.901238  -4.934871</span>
<span class="co">#&gt; Astro_10  18.151228  13.2033148  -25.52110 -53.049486 -66.622017 -25.255920</span>
<span class="co">#&gt;                PC42       PC43         PC44       PC45       PC46       PC47</span>
<span class="co">#&gt; Astro_1    1.075291   2.482969 -14.28618128  30.244406  -1.616667   3.405948</span>
<span class="co">#&gt; Astro_2   25.458186  25.010055  -2.07154618  -8.763516 -32.322790 -13.392106</span>
<span class="co">#&gt; Astro_3    8.936136   2.003220  34.96238486  35.801377  13.870422  54.945538</span>
<span class="co">#&gt; Astro_4  -13.170964  28.438346  -0.04542305  11.428859  56.336719  37.620996</span>
<span class="co">#&gt; Astro_5   18.216748 -15.581942   1.98071920 -14.241538 -10.340019 -48.294426</span>
<span class="co">#&gt; Astro_6   25.755477 -30.632022  10.40534150 -56.490817   2.586434 -17.607459</span>
<span class="co">#&gt; Astro_7  -39.448483  -7.468561 -17.78599614  14.048097 -72.683620  -6.553973</span>
<span class="co">#&gt; Astro_8  -29.415725  41.988426  25.60939744 -55.196292  27.849885  30.300124</span>
<span class="co">#&gt; Astro_9   20.709195 -31.093410 -15.28951440  34.066989  22.403566  29.207059</span>
<span class="co">#&gt; Astro_10  16.931399  34.970798  18.97587492   9.559153  -9.917507 -24.457453</span>
<span class="co">#&gt;                 PC48       PC49       PC50</span>
<span class="co">#&gt; Astro_1   15.2142797  47.890422  -9.967947</span>
<span class="co">#&gt; Astro_2    0.2409696 -25.624909  20.866141</span>
<span class="co">#&gt; Astro_3   71.0075318 -10.090061  -4.616696</span>
<span class="co">#&gt; Astro_4  -11.1277897  29.644472  29.185383</span>
<span class="co">#&gt; Astro_5  -16.7944734 -11.469022  -6.127383</span>
<span class="co">#&gt; Astro_6   12.4398780  -9.836293  -7.930831</span>
<span class="co">#&gt; Astro_7   14.5135575  -5.445455  35.395411</span>
<span class="co">#&gt; Astro_8  -29.1231518  13.976441  31.773761</span>
<span class="co">#&gt; Astro_9  -44.1981697  43.669191 -12.398811</span>
<span class="co">#&gt; Astro_10 -20.5927442 -59.568311 -79.249020</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
