<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BandNorm-tutorial • BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BandNorm-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD_tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BandNorm-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BandNorm-tutorial</h1>
            
      
      
      <div class="hidden name"><code>BandNorm-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The advent of single-cell sequencing technologies in profiling 3D genome organization led to development of single-cell high-throughput chromatin conformation (scHi-C) assays. Data from these assays enhance our ability to study dynamic chromatin architecture and the impact of spatial genome interactions on cell regulation at an unprecedented resolution. At the individual cell resolution, heterogeneity driven by the stochastic nature of chromatin fiber, various nuclear processes, and unwanted variation due to sequencing depths and batch effects poses major analytical challenges for inferring single cell-level 3D genome organizations.</p>
<p>To explicitly capture chromatin conformation features and distinguish cells based on their 3D genome organizations, we develop a simple and fast band normalization approach, <code>BandNorm</code>. <code>BandNorm</code> first removes genomic distance bias within a cell, and sequencing depth normalizes between cells. Consequently, <code>BandNorm</code> adds back a common band-dependent contact decay profile for the contact matrices across cells. The former step is achieved by dividing the interaction frequencies of each band within a cell with the cell’s band mean. The latter step is implemented by multiplying each scaled band by the average band mean across cells.</p>
<p><img src="../figures/bandnorm_intro.png"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To use <code>BandNorm</code>, the following packages are necessary:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ggplot2'</span>, <span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'Rtsne'</span>, <span class="st">'umap'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"immunogenomics/harmony"</span><span class="op">)</span></code></pre></div>
<p>BandNorm can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'sshen82/BandNorm'</span>, build_vignettes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Also, if the user doesn’t have default R path writing permission, it is possible to install the package to the personal library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_libpaths.html">with_libpaths</a></span><span class="op">(</span><span class="st">"personalLibraryPath"</span>, <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"sshen82/BandNorm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>There are three possible ways to input the data:</p>
<ol style="list-style-type: decimal">
<li>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</li>
</ol>
<pre><code>chr1    0       chr1    1000000 9
chr1    1000000 chr1    1000000 200
chr1    0       chr1    2000000 2
chr1    1000000 chr1    2000000 4
chr1    2000000 chr1    2000000 220
chr1    1000000 chr1    3000000 1
chr1    2000000 chr1    3000000 11
chr1    3000000 chr1    3000000 197
chr1    1000000 chr1    4000000 1
chr1    2000000 chr1    4000000 2</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>An R data.frame in the form of [chr1, binA, binB, count, diag, cell_name]. The column names should be c(“chrom”, “binA”, “binB”, “count”, “diag”, “cell”). Below we print the example data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Loading required package: strawr</span>
<span class="co">#&gt; Warning: package 'strawr' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: rmarkdown</span>
<span class="co">#&gt; Warning: package 'rmarkdown' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridis</span>
<span class="co">#&gt; Warning: package 'viridis' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridisLite</span>
<span class="co">#&gt; Warning: package 'viridisLite' was built under R version 4.0.5</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hic_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;    chrom  binA  binB count  diag    cell</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00    38 0e+00 Astro_1</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06   105 0e+00 Astro_1</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06   116 0e+00 Astro_1</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06     2 1e+06 Astro_1</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06   133 0e+00 Astro_1</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06     1 1e+06 Astro_1</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06   112 0e+00 Astro_1</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06     1 2e+06 Astro_1</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>We also support using Juicer .hic format, and you can use bandnorm_juicer function to run it. We will have a thorough example for this in the future.</li>
</ol>
<p>For the example dataset, we also have batch information and cell-type information. Note that the cell_name in those external information should be consistent with the hic_df file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">batch</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  batch        </span>
<span class="co">#&gt;  [1,] "Astro_1"  "181218_21yr"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "181218_21yr"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "181218_21yr"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "181218_21yr"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "181218_21yr"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "181218_21yr"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "181218_21yr"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "181218_21yr"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "181218_21yr"</span>
<span class="co">#&gt; [10,] "Astro_10" "181218_21yr"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       cell_name  cluster</span>
<span class="co">#&gt;  [1,] "Astro_1"  "Astro"</span>
<span class="co">#&gt;  [2,] "Astro_2"  "Astro"</span>
<span class="co">#&gt;  [3,] "Astro_3"  "Astro"</span>
<span class="co">#&gt;  [4,] "Astro_4"  "Astro"</span>
<span class="co">#&gt;  [5,] "Astro_5"  "Astro"</span>
<span class="co">#&gt;  [6,] "Astro_6"  "Astro"</span>
<span class="co">#&gt;  [7,] "Astro_7"  "Astro"</span>
<span class="co">#&gt;  [8,] "Astro_8"  "Astro"</span>
<span class="co">#&gt;  [9,] "Astro_9"  "Astro"</span>
<span class="co">#&gt; [10,] "Astro_10" "Astro"</span></code></pre></div>
</div>
<div id="download-existing-single-cell-hi-c-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-existing-single-cell-hi-c-data" class="anchor"></a>Download existing single-cell Hi-C data</h2>
<p>You can also use <code>download_schic</code> function to download real data. The available data includes <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008173">Kim2020</a>, <a href="https://www.nature.com/articles/s41592-019-0502-z">Li2019</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5330809/">Ramani2017</a>, and <a href="https://www.nature.com/articles/s41592-019-0547-z">Lee2019</a>. You can specify one of the cell-types in those cell lines. There are also summary files available for them, and the summary includes batch, cell-type, depth and sparsity information. You can go to <a href="http://pages.stat.wisc.edu/~sshen82/bandnorm/">this website</a> for more information. We won’t run it here since it will mess up your computer.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/download_schic.html">download_schic</a></span><span class="op">(</span><span class="st">"Li2019"</span>, cell_type <span class="op">=</span> <span class="st">"2i"</span>, cell_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>, summary_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-bandnorm" class="section level2">
<h2 class="hasAnchor">
<a href="#use-bandnorm" class="anchor"></a>Use BandNorm</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 400 cells of Astro, ODC, MG, Sst, four cell types and chromosome 1 from <a href="https://www.nature.com/articles/s41592-019-0547-z">Ecker2019</a> for test run. After obtaining the hic_df file, you can use the <code>bandnorm</code> function to normalize the data. The result consists of 6 columns: chromosome, binA, binB, diag (which is binB - binA), cell (which is the name of the cell), and BandNorm. You can use the save option to save the normalized file, and if so, you need to give a path for output.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandnorm_result</span> <span class="op">=</span> <span class="fu"><a href="../reference/bandnorm.html">bandnorm</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">hic_df</span>, save <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">bandnorm_result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;    chrom  binA  binB  diag    cell   BandNorm</span>
<span class="co">#&gt; 1   chr1 0e+00 0e+00 0e+00 Astro_1 120.131557</span>
<span class="co">#&gt; 2   chr1 0e+00 1e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 3   chr1 1e+06 1e+06 0e+00 Astro_1 331.942461</span>
<span class="co">#&gt; 4   chr1 1e+06 2e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 5   chr1 2e+06 2e+06 0e+00 Astro_1 366.717385</span>
<span class="co">#&gt; 6   chr1 2e+06 3e+06 1e+06 Astro_1  16.102209</span>
<span class="co">#&gt; 7   chr1 3e+06 3e+06 0e+00 Astro_1 420.460450</span>
<span class="co">#&gt; 8   chr1 3e+06 4e+06 1e+06 Astro_1   8.051104</span>
<span class="co">#&gt; 9   chr1 4e+06 4e+06 0e+00 Astro_1 354.071958</span>
<span class="co">#&gt; 10  chr1 3e+06 5e+06 2e+06 Astro_1   8.392744</span>
<span class="co"># bandnorm_result = bandnorm(hic_df = hic_df, save = TRUE, save_path = getwd())</span></code></pre></div>
<p>Then, you can use <code>create_embedding</code> function to obtain a PCA embedding for the data. You can choose to use Harmony to remove the batch effect.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_embedding.html">create_embedding</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">bandnorm_result</span>, do_harmony <span class="op">=</span> <span class="cn">TRUE</span>, batch <span class="op">=</span> <span class="va">batch</span><span class="op">)</span>
<span class="co">#&gt; [1] "The number of features is 26330"</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony 4/10</span>
<span class="co">#&gt; Harmony 5/10</span>
<span class="co">#&gt; Harmony 6/10</span>
<span class="co">#&gt; Harmony 7/10</span>
<span class="co">#&gt; Harmony 8/10</span>
<span class="co">#&gt; Harmony 9/10</span>
<span class="co">#&gt; Harmony 10/10</span>
<span class="va">embedding</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;                 PC1       PC2        PC3        PC4        PC5        PC6</span>
<span class="co">#&gt; Astro_1  -22.529701 29.919691 -20.428575 32.1949589 -37.575305 -19.379780</span>
<span class="co">#&gt; Astro_2  -10.301145 32.862935   3.350233 14.9808294 -10.700752   4.250852</span>
<span class="co">#&gt; Astro_3   -4.703623  6.786751   2.476166  0.2657549  14.912671 -14.366302</span>
<span class="co">#&gt; Astro_4   -8.918750 26.699081 -25.765253 11.0752057 -11.815217 -18.998084</span>
<span class="co">#&gt; Astro_5   11.590558 25.563502 -15.222894 17.5473435 -28.994855 -33.667267</span>
<span class="co">#&gt; Astro_6  -19.791087 13.083690 -21.267348 -0.7930665  -3.790735   8.999173</span>
<span class="co">#&gt; Astro_7  -30.353866 19.285986 -11.084831  8.1710666  -5.023204  -9.188444</span>
<span class="co">#&gt; Astro_8   -8.880290 33.510113  -2.924422  7.6929131  -5.747726   9.874657</span>
<span class="co">#&gt; Astro_9   -5.005009 26.798847 -12.123772 18.9937397 -15.694366  -6.067186</span>
<span class="co">#&gt; Astro_10 -15.134113  3.813363 -14.551682  3.1299376 -24.393120  -8.603493</span>
<span class="co">#&gt;                 PC7        PC8        PC9       PC10       PC11       PC12</span>
<span class="co">#&gt; Astro_1    6.458602 -46.882023 38.7504531  -7.626622 32.0083475 -25.276847</span>
<span class="co">#&gt; Astro_2   -5.438821 -12.743877 10.8641452  11.922280 23.5781816 -30.150702</span>
<span class="co">#&gt; Astro_3  -13.984058  25.493606  0.9458139   3.802326  8.0730609  -6.993781</span>
<span class="co">#&gt; Astro_4   10.530769   2.464007  3.1278263 -13.864330  0.7679487 -14.650912</span>
<span class="co">#&gt; Astro_5  -30.651370 -43.181097  5.6943393 -10.807100 90.5010664 -52.433387</span>
<span class="co">#&gt; Astro_6   -9.282068  -4.486121 -0.1934689   2.364112  7.4510784 -42.292238</span>
<span class="co">#&gt; Astro_7   -1.013412  -4.028229  1.5766239  -9.921416 -1.5696722 -15.877149</span>
<span class="co">#&gt; Astro_8  -16.725932  10.671548  3.5465213   1.015077 28.4052792  -7.780347</span>
<span class="co">#&gt; Astro_9  -18.677168  -9.436774  7.5974314 -15.096990 39.6389396 -86.254182</span>
<span class="co">#&gt; Astro_10  14.146469  -4.437926 30.5671555 -15.851628 12.8841777 -28.514043</span>
<span class="co">#&gt;                 PC13         PC14        PC15         PC16       PC17</span>
<span class="co">#&gt; Astro_1   29.7871760 -14.03269563  22.4903216 -11.89009165  41.628181</span>
<span class="co">#&gt; Astro_2    0.4141777   6.21025722  -5.3086165   9.83630187 -16.779398</span>
<span class="co">#&gt; Astro_3    6.0483724  11.85226169  -3.5001379  17.63692407   3.345467</span>
<span class="co">#&gt; Astro_4   -8.7359644   9.52007734 -18.6797051  -2.84284797  -2.035488</span>
<span class="co">#&gt; Astro_5   22.0716484  48.55813248  -1.9161044  15.97590974  18.480319</span>
<span class="co">#&gt; Astro_6   27.3867252   0.60898479   6.6110806 -17.25068463   2.000931</span>
<span class="co">#&gt; Astro_7   21.8554648  -0.05097078  -4.5850498   5.93895062  11.318766</span>
<span class="co">#&gt; Astro_8   -8.0592549 -34.94100383  -0.9429693 -17.41458486  -6.830620</span>
<span class="co">#&gt; Astro_9  -35.8966078 -24.41068469  25.3160653  34.63747519   7.000637</span>
<span class="co">#&gt; Astro_10  17.3041471 -22.98543000  -9.9786622  -0.02696858  20.793093</span>
<span class="co">#&gt;                 PC18       PC19       PC20         PC21       PC22        PC23</span>
<span class="co">#&gt; Astro_1  -45.6220736  36.707874  17.873814 -141.7689593  54.644011  14.5179120</span>
<span class="co">#&gt; Astro_2  -16.8659361   3.038191   1.176259    5.0450348  -7.910147  -1.8126838</span>
<span class="co">#&gt; Astro_3    0.2697172  26.821915   2.019619   25.4606066  -8.617872  -9.9047287</span>
<span class="co">#&gt; Astro_4    2.8021940  -7.454029  15.549901   17.4760863 -13.339986 -19.4152074</span>
<span class="co">#&gt; Astro_5   29.0575766 116.756384  29.511012   74.4688967  18.250999 126.7316101</span>
<span class="co">#&gt; Astro_6  -13.0847494 -17.300515  -2.883770    2.2909895  47.168000  -3.4283134</span>
<span class="co">#&gt; Astro_7  -23.8959061  -9.889128   2.124611    0.8401251  20.550757 -22.5065995</span>
<span class="co">#&gt; Astro_8   -7.6143673  -7.411180 -18.016980  -37.6377802 -25.238952 -23.4390635</span>
<span class="co">#&gt; Astro_9  -35.6057337 -89.512015 -48.633371   -7.1264310 -50.702962  46.0687683</span>
<span class="co">#&gt; Astro_10 -25.5138002  -1.760108  -7.425337   -3.1682509 -21.912548  -0.4675542</span>
<span class="co">#&gt;                PC24        PC25         PC26        PC27        PC28       PC29</span>
<span class="co">#&gt; Astro_1  -44.074336  17.6885751  -67.3336045  15.0078509 -125.100117   3.600031</span>
<span class="co">#&gt; Astro_2    5.078126  -8.9506843    6.7741262  -0.4546433   -4.175400  -7.513889</span>
<span class="co">#&gt; Astro_3    2.519785   8.5629702   31.3574878 -31.9900280   40.101862  30.953583</span>
<span class="co">#&gt; Astro_4  -16.322886   0.7245706   13.8217022  21.9322372   24.710150   3.010023</span>
<span class="co">#&gt; Astro_5  -76.012293  37.9164424   68.3591338 -57.6107660   60.882958 -46.771381</span>
<span class="co">#&gt; Astro_6    6.841335   3.3628883    1.8453346  31.9804666  -11.118046  22.603891</span>
<span class="co">#&gt; Astro_7   -6.225793  28.6642384    5.3568129  29.8482530   10.018715  18.729924</span>
<span class="co">#&gt; Astro_8   36.286450   7.0038688   -4.0511955 -34.2223279   22.914826  13.179697</span>
<span class="co">#&gt; Astro_9   35.846337   3.6442627 -165.4590883  -4.8063002   62.235580  -4.018587</span>
<span class="co">#&gt; Astro_10  -1.237639 -43.7914496    0.6414033  18.7550745    1.873991  11.948384</span>
<span class="co">#&gt;                PC30       PC31       PC32       PC33       PC34       PC35</span>
<span class="co">#&gt; Astro_1  -32.158110   5.042545 -56.776083  22.628783 -17.714043  51.005149</span>
<span class="co">#&gt; Astro_2   15.639317  -1.457761  -3.441088   6.201641  12.153262 -11.595463</span>
<span class="co">#&gt; Astro_3   13.075929  -2.949684   1.488805 -70.968091   4.538318 -18.236622</span>
<span class="co">#&gt; Astro_4  -11.211401 -22.155449  14.297772  -8.142230  12.641897  12.147478</span>
<span class="co">#&gt; Astro_5   13.415900 -13.933353  -4.711585  -1.486070 -46.035770  67.527995</span>
<span class="co">#&gt; Astro_6   -1.022839  -3.810666  41.290892  13.602443  -8.930443  14.285264</span>
<span class="co">#&gt; Astro_7  -22.782815 -29.764525   1.791390 -40.087603   6.189136 -56.701343</span>
<span class="co">#&gt; Astro_8  -33.562446  13.490445  10.108195  43.939740 -18.006447  66.595883</span>
<span class="co">#&gt; Astro_9   -8.281454  62.320588 -21.523872   2.268917  -8.727296 -60.799006</span>
<span class="co">#&gt; Astro_10  -3.514149 -40.248681  39.880609 -29.542120  19.443569   5.805777</span>
<span class="co">#&gt;                 PC36       PC37       PC38       PC39       PC40        PC41</span>
<span class="co">#&gt; Astro_1  -69.4262652  26.288819  -15.90950  41.740125  -2.554066  -2.7335249</span>
<span class="co">#&gt; Astro_2   -1.6925239   5.331197   27.27409   8.423797  -1.607595  -6.6678398</span>
<span class="co">#&gt; Astro_3    5.5860665  -5.762101   57.03883 -14.169663 -20.685463  10.0529821</span>
<span class="co">#&gt; Astro_4   14.2635142 -24.011775  -28.27633  37.875896   5.303825 -41.9081148</span>
<span class="co">#&gt; Astro_5   30.2986208  -8.147428  -35.76902  27.479330 -43.769482  68.2261398</span>
<span class="co">#&gt; Astro_6    5.5525134 -15.031977   37.97109  46.654762  38.156125  -2.1616256</span>
<span class="co">#&gt; Astro_7    0.6096938 -23.364645   42.90639 -40.687411   5.511780  31.7321687</span>
<span class="co">#&gt; Astro_8   32.7001838  32.084228 -118.73059  11.131009 -19.095841   0.5911464</span>
<span class="co">#&gt; Astro_9  -10.9956581   7.562468   69.25402 -65.033339  25.776105 -18.0565995</span>
<span class="co">#&gt; Astro_10   9.6943407  13.153470  -24.47659 -47.533941 -66.642515 -25.3266188</span>
<span class="co">#&gt;                 PC42       PC43        PC44       PC45       PC46        PC47</span>
<span class="co">#&gt; Astro_1    1.4360557  -7.421172 -10.6472387  11.232254   3.924602 -25.4635095</span>
<span class="co">#&gt; Astro_2   28.5169867  35.063323  -0.2764091  -5.449398 -26.600556   0.6182511</span>
<span class="co">#&gt; Astro_3   -0.6583107  -1.546360  38.8723128  16.109059  10.174859  52.8651955</span>
<span class="co">#&gt; Astro_4  -19.4152513  28.227347   2.0239637  11.322492  52.215480  46.5853909</span>
<span class="co">#&gt; Astro_5    7.2451402 -12.921605   6.3510862 -16.224477 -10.205672 -45.6440372</span>
<span class="co">#&gt; Astro_6   27.3436267 -34.051537   9.9925122 -47.902733   6.368458 -23.8307610</span>
<span class="co">#&gt; Astro_7  -36.8625452 -15.013416 -16.6719048  21.548301 -67.032268  -8.1360737</span>
<span class="co">#&gt; Astro_8  -22.6824536  46.352427  27.5293018 -49.812160  37.143088  35.6965045</span>
<span class="co">#&gt; Astro_9   17.5254763 -14.264078 -10.0886548  27.644936  28.382010  39.2033808</span>
<span class="co">#&gt; Astro_10  18.5467293  31.557911  18.5940894  18.058345  -6.067512 -30.8551111</span>
<span class="co">#&gt;                PC48       PC49       PC50</span>
<span class="co">#&gt; Astro_1    1.582327  24.369637 -16.482827</span>
<span class="co">#&gt; Astro_2  -12.649302 -12.944595  20.317202</span>
<span class="co">#&gt; Astro_3   66.622452 -19.572025   7.914659</span>
<span class="co">#&gt; Astro_4   -3.025927  38.696105  39.743034</span>
<span class="co">#&gt; Astro_5    3.656793 -14.531549   2.144960</span>
<span class="co">#&gt; Astro_6   18.375873  -4.864853 -13.030812</span>
<span class="co">#&gt; Astro_7   14.226905   4.888972  33.098099</span>
<span class="co">#&gt; Astro_8  -46.219452  25.881059  25.702394</span>
<span class="co">#&gt; Astro_9  -47.652236  39.898849 -11.229672</span>
<span class="co">#&gt; Astro_10 -14.705111 -54.788716 -84.443361</span></code></pre></div>
<p>Finally, using the embedding, you can use UMAP or tSNE to get a projection of all the cells on the 2D plane. If you have the cell-type information or batch information, it is possible to color them on the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">cell_type</span>, label <span class="op">=</span> <span class="st">"Cell Type"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_embedding.html">plot_embedding</a></span><span class="op">(</span><span class="va">embedding</span>, <span class="st">"UMAP"</span>, cell_info <span class="op">=</span> <span class="va">batch</span>, label <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></code></pre></div>
<p><img src="BandNorm-tutorial_files/figure-html/plot_embedding-2.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
