<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGAD-tutorial â€¢ BandNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scGAD-tutorial">
<meta property="og:description" content="BandNorm">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BandNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BandNorm-tutorial.html">BandNorm-tutorial</a>
    </li>
    <li>
      <a href="../articles/scGAD-tutorial.html">scGAD-tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="scGAD-tutorial_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>scGAD-tutorial</h1>
            
      
      
      <div class="hidden name"><code>scGAD-tutorial.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Recent advancements in single-cell technologies enabled the profiling of 3D genome structures in a single-cell fashion. Quantitative tools are needed to fully leverage the unprecedented resolution of single-cell high-throughput chromatin conformation (scHi-C) data and integrate it with other single-cell data modalities. We developed single-cell gene-body associating domain (scGAD) scores available in R package to explore scHi-C data in units of genes. Proper normalization is investigated to eliminate technical biases.</p>
<div class="figure">
<img src="../figures/scgadMethod.jpg" alt=""><p class="caption">The introduction for scGAD</p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The installation is the same as <code>BandNorm</code>, since scGAD is included in this package.</p>
</div>
<div id="format-of-input" class="section level2">
<h2 class="hasAnchor">
<a href="#format-of-input" class="anchor"></a>Format of input</h2>
<p>We currently allows two ways input the data:</p>
<p>A path containing all cells in the form of [chr1, binA, chr2, binB, count], the cell looks like below:</p>
<pre><code>chr1    0   chr1    10000 9
chr1    10000   chr1    10000 20
chr1    0   chr1    20000 2
chr1    10000 chr1    20000 4
chr1    20000 chr1    20000 22
chr1    10000 chr1    30000 1
chr1    20000 chr1    30000 11
chr1    30000 chr1    30000 197
chr1    10000 chr1    40000 1
chr1    20000 chr1    40000 2</code></pre>
<p>Another way is to input a data.frame containing all interactions and the column is like [chr1, binA, binB, count, cell_name]. This way is not recommended, since single-cell Hi-C data is huge. However, it will be fast if your computer can handle the data in this way.</p>
<p>The data should be 10kb resolution. We will allow more formats to be used here, and we will try to allow higher resolution.</p>
</div>
<div id="use-scgad" class="section level2">
<h2 class="hasAnchor">
<a href="#use-scgad" class="anchor"></a>Use scGAD</h2>
<p>We provide a demo scHi-C data named hic_df, sampled 350 cells of Mature Oligodendrocyte, Microglia Etc. and Hippocampal Granule Cell, three cell types from <a href="https://www.cell.com/cell/fulltext/S0092-8674(20)31754-2?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420317542%3Fshowall%3Dtrue">Tan2021</a> for test run. Note that the cells are different from real cells, since I modify it for the propose of fast computation. The data presented here is just to show how the data looks like, and how our framework runs. After obtaining the scgad_df file, you can use the <code>scGAD</code> function to normalize the data and obtain the scGAD score. In scGAD function, except from the data, we also need annotation file, where you can get mm9, mm10, hg19 and hg38 using data(Annotations). The result is similar to scRNAseq, where what you can get is a matrix that rows are genes, and columns are cells.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BandNorm</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: package 'data.table' was built under R version 4.0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'data.table'</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     between, first, last</span>
<span class="co">#&gt; Loading required package: Rtsne</span>
<span class="co">#&gt; Warning: package 'Rtsne' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: umap</span>
<span class="co">#&gt; Warning: package 'umap' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: progress</span>
<span class="co">#&gt; Warning: package 'progress' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: harmony</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Warning: package 'Rcpp' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: FreeHiCLite</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: gmodels</span>
<span class="co">#&gt; Warning: package 'gmodels' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: doParallel</span>
<span class="co">#&gt; Warning: package 'doParallel' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Warning: package 'foreach' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Warning: package 'iterators' was built under R version 4.0.3</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; Warning: package 'matrixStats' was built under R version 4.0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     count</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.0.5</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jeroen/curl">curl</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'curl' was built under R version 4.0.5</span>
<span class="co">#&gt; Using libcurl 7.64.1 with Schannel</span>
<span class="va">h</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/curl/man/handle.html">new_handle</a></span><span class="op">(</span>dirlistonly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">con</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/curl/man/curl.html">curl</a></span><span class="op">(</span><span class="st">"http://ftp.cs.wisc.edu/pub/users/kelesgroup/siqi/scGAD/scGADExample.rda"</span>, <span class="st">"r"</span>, <span class="va">h</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="va">con</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/connections.html">close</a></span><span class="op">(</span><span class="va">con</span><span class="op">)</span>

<span class="va">gad_score</span> <span class="op">=</span> <span class="fu"><a href="../reference/scGAD.html">scGAD</a></span><span class="op">(</span>hic_df <span class="op">=</span> <span class="va">scgad_df</span>, genes <span class="op">=</span> <span class="va">geneANNO</span>, depthNorm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Then, you can use PCA to look at the result, and cells are clearly separated.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="va">summary</span> <span class="op">=</span> <span class="va">summary</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">gad_score</span><span class="op">)</span>, <span class="va">summary</span><span class="op">$</span><span class="va">cell</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">gadPCA</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html">prcomp</a></span><span class="op">(</span><span class="va">gad_score</span><span class="op">)</span><span class="op">$</span><span class="va">rotation</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>
<span class="va">gadPCA</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">gadPCA</span>, cellTypes <span class="op">=</span> <span class="va">summary</span><span class="op">$</span><span class="va">`cell-type cluster`</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">gadPCA</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, col <span class="op">=</span> <span class="va">cellTypes</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Hippocampal Granule Cell"</span>, <span class="st">"Mature Oligodendrocyte"</span>, <span class="st">"Microglia Etc."</span><span class="op">)</span>,
                   values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#1B4F72"</span>, <span class="st">"#F39C12"</span>, <span class="st">"#000000"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="scGAD-tutorial_files/figure-html/PCA%20Result-1.png" width="700"></p>
<p>Finally, with scRNA-seq data on hand, we can do projection from scGAD to scRNA-seq. For scRNA-seq, we have 1076 cells from Mature Oligodendrocyte, Microglia Etc. and Hippocampal Granule Cell.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/">viridis</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'viridis' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: viridisLite</span>
<span class="co">#&gt; Warning: package 'viridisLite' was built under R version 4.0.5</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'gridExtra'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'ggpubr' was built under R version 4.0.3</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>

<span class="va">DataList</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scGAD <span class="op">=</span> <span class="va">gad_score</span>, scRNAseq <span class="op">=</span> <span class="va">RNA</span><span class="op">)</span>
<span class="va">cellTypeList</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scGAD <span class="op">=</span> <span class="va">summary</span><span class="op">$</span><span class="va">`cell-type cluster`</span>, scRNAseq <span class="op">=</span> <span class="va">cellTypeRNA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">=</span> <span class="va">summary</span><span class="op">$</span><span class="va">cell</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">RNA</span><span class="op">)</span>

<span class="va">combinedAssay</span> <span class="op">=</span> <span class="fu"><a href="../reference/runProjection.html">runProjection</a></span><span class="op">(</span><span class="va">DataList</span>, doNorm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">FALSE</span>, <span class="cn">FALSE</span><span class="op">)</span>, <span class="va">cellTypeList</span><span class="op">)</span>
<span class="co">#&gt; Warning: The following arguments are not used: row.names</span>
<span class="co">#&gt; Warning: The following arguments are not used: row.names</span>
<span class="co">#&gt; Scaling features for provided objects</span>
<span class="co">#&gt; Finding all pairwise anchors</span>
<span class="co">#&gt; Running CCA</span>
<span class="co">#&gt; Merging objects</span>
<span class="co">#&gt; Finding neighborhoods</span>
<span class="co">#&gt; Finding anchors</span>
<span class="co">#&gt;  Found 1722 anchors</span>
<span class="co">#&gt; Filtering anchors</span>
<span class="co">#&gt;  Retained 541 anchors</span>
<span class="co">#&gt; Merging dataset 1 into 2</span>
<span class="co">#&gt; Extracting anchors for merged samples</span>
<span class="co">#&gt; Finding integration vectors</span>
<span class="co">#&gt; Finding integration vector weights</span>
<span class="co">#&gt; Integrating data</span>
<span class="co">#&gt; Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span>
<span class="co">#&gt; To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span>
<span class="co">#&gt; This message will be shown once per session</span>
<span class="co">#&gt; 19:02:57 UMAP embedding parameters a = 0.9922 b = 1.112</span>
<span class="co">#&gt; 19:02:57 Read 1426 rows and found 5 numeric columns</span>
<span class="co">#&gt; 19:02:57 Using Annoy for neighbor search, n_neighbors = 30</span>
<span class="co">#&gt; 19:02:57 Building Annoy index with metric = cosine, n_trees = 50</span>
<span class="co">#&gt; 0%   10   20   30   40   50   60   70   80   90   100%</span>
<span class="co">#&gt; [----|----|----|----|----|----|----|----|----|----|</span>
<span class="co">#&gt; **************************************************|</span>
<span class="co">#&gt; 19:02:57 Writing NN index file to temp file C:\Users\solei\AppData\Local\Temp\RtmpcVH4qX\file36081bb04829</span>
<span class="co">#&gt; 19:02:57 Searching Annoy index using 1 thread, search_k = 3000</span>
<span class="co">#&gt; 19:02:57 Annoy recall = 100%</span>
<span class="co">#&gt; 19:02:58 Commencing smooth kNN distance calibration using 1 thread</span>
<span class="co">#&gt; 19:02:58 Initializing from normalized Laplacian + noise</span>
<span class="co">#&gt; 19:02:58 Commencing optimization for 500 epochs, with 55294 positive edges</span>
<span class="co">#&gt; 19:03:01 Optimization finished</span>

<span class="va">p_celltype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">combinedAssay</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, repel <span class="op">=</span> <span class="cn">TRUE</span>, pt.size <span class="op">=</span> <span class="fl">1.3</span>, shape.by <span class="op">=</span> <span class="st">"method"</span>, label.size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"UMAP 1"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"UMAP 2"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Hippocampal Granule Cell"</span>, <span class="st">"Mature Oligodendrocyte"</span>, <span class="st">"Microglia Etc."</span><span class="op">)</span>,
                   values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#1B4F72"</span>, <span class="st">"#F39C12"</span>, <span class="st">"#000000"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/rremove.html">rremove</a></span><span class="op">(</span><span class="st">"legend"</span><span class="op">)</span>


<span class="va">pRNA</span> <span class="op">=</span> <span class="va">combinedAssay</span><span class="op">@</span><span class="va">reductions</span><span class="op">$</span><span class="va">umap</span><span class="op">@</span><span class="va">cell.embeddings</span> <span class="op">%&gt;%</span> <span class="va">data.frame</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>celltype <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"scGAD"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"scRNA-seq"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">label</span> <span class="op">==</span> <span class="st">"scRNA-seq"</span><span class="op">)</span> <span class="op">%&gt;%</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, color <span class="op">=</span> <span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/theme_pubr.html">theme_pubr</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Hippocampal Granule Cell"</span>, <span class="st">"Mature Oligodendrocyte"</span>, <span class="st">"Microglia Etc."</span><span class="op">)</span>,
                   values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#1B4F72"</span>, <span class="st">"#F39C12"</span>, <span class="st">"#000000"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"UMAP 1"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"UMAP 2"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/rremove.html">rremove</a></span><span class="op">(</span><span class="st">"legend"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Single-cell Transcriptomics"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>


<span class="va">pGAD</span> <span class="op">=</span> <span class="va">combinedAssay</span><span class="op">@</span><span class="va">reductions</span><span class="op">$</span><span class="va">umap</span><span class="op">@</span><span class="va">cell.embeddings</span> <span class="op">%&gt;%</span> <span class="va">data.frame</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>celltype <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"scGAD"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"scRNA-seq"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">cellTypeList</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">label</span> <span class="op">==</span> <span class="st">"scGAD"</span><span class="op">)</span> <span class="op">%&gt;%</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, color <span class="op">=</span> <span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/theme_pubr.html">theme_pubr</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Hippocampal Granule Cell"</span>, <span class="st">"Mature Oligodendrocyte"</span>, <span class="st">"Microglia Etc."</span><span class="op">)</span>,
                   values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#1B4F72"</span>, <span class="st">"#F39C12"</span>, <span class="st">"#000000"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"UMAP 1"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"UMAP 2"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/rremove.html">rremove</a></span><span class="op">(</span><span class="st">"legend"</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Single-cell 3D Genomics"</span><span class="op">)</span> <span class="op">+</span>
 <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>


<span class="va">lay</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">p_celltype</span>, <span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">arrangeGrob</a></span><span class="op">(</span><span class="va">pRNA</span>, <span class="va">pGAD</span>, ncol <span class="op">=</span> <span class="fl">1</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>, layout_matrix <span class="op">=</span> <span class="va">lay</span><span class="op">)</span></code></pre></div>
<p><img src="scGAD-tutorial_files/figure-html/Projection-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Siqi Shen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
